<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 Data Wrangling | Applications of R in Healthcare</title>
  <meta name="description" content="This is a book describing the use of R in Healthcare. It is aimed at beginners of R language. The majority of the examples are taken from works in Neurology. Where possible data from other disease such as heart disease, cancer and vaccine are used. The ideas and principles can be applied to other aspects of Healthcare. The output format for this example is bookdown::gitbook." />
  <meta name="generator" content="bookdown 0.21 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 Data Wrangling | Applications of R in Healthcare" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This is a book describing the use of R in Healthcare. It is aimed at beginners of R language. The majority of the examples are taken from works in Neurology. Where possible data from other disease such as heart disease, cancer and vaccine are used. The ideas and principles can be applied to other aspects of Healthcare. The output format for this example is bookdown::gitbook." />
  <meta name="github-repo" content="GNtem2/HealthcareRbook" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Data Wrangling | Applications of R in Healthcare" />
  
  <meta name="twitter:description" content="This is a book describing the use of R in Healthcare. It is aimed at beginners of R language. The majority of the examples are taken from works in Neurology. Where possible data from other disease such as heart disease, cancer and vaccine are used. The ideas and principles can be applied to other aspects of Healthcare. The output format for this example is bookdown::gitbook." />
  

<meta name="author" content="Thanh Phan" />


<meta name="date" content="2023-03-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="intro.html"/>
<link rel="next" href="statistics.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Applications-of-R-in-Healthcare</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preamble</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Introduction to R</a><ul>
<li class="chapter" data-level="2.1" data-path="intro.html"><a href="intro.html#plot-using-base-r"><i class="fa fa-check"></i><b>2.1</b> Plot using base R</a></li>
<li class="chapter" data-level="2.2" data-path="intro.html"><a href="intro.html#ggplot2"><i class="fa fa-check"></i><b>2.2</b> ggplot2</a><ul>
<li class="chapter" data-level="2.2.1" data-path="intro.html"><a href="intro.html#histogram"><i class="fa fa-check"></i><b>2.2.1</b> Histogram</a></li>
<li class="chapter" data-level="2.2.2" data-path="intro.html"><a href="intro.html#bar-plot"><i class="fa fa-check"></i><b>2.2.2</b> Bar plot</a></li>
<li class="chapter" data-level="2.2.3" data-path="intro.html"><a href="intro.html#pie-chart"><i class="fa fa-check"></i><b>2.2.3</b> Pie chart</a></li>
<li class="chapter" data-level="2.2.4" data-path="intro.html"><a href="intro.html#scatter-plot"><i class="fa fa-check"></i><b>2.2.4</b> Scatter plot</a></li>
<li class="chapter" data-level="2.2.5" data-path="intro.html"><a href="intro.html#arrange-plot-in-grids"><i class="fa fa-check"></i><b>2.2.5</b> arrange plot in grids</a></li>
<li class="chapter" data-level="2.2.6" data-path="intro.html"><a href="intro.html#line-plot"><i class="fa fa-check"></i><b>2.2.6</b> Line plot</a></li>
<li class="chapter" data-level="2.2.7" data-path="intro.html"><a href="intro.html#facet-wrap"><i class="fa fa-check"></i><b>2.2.7</b> Facet wrap</a></li>
<li class="chapter" data-level="2.2.8" data-path="intro.html"><a href="intro.html#polygons"><i class="fa fa-check"></i><b>2.2.8</b> Polygons</a></li>
<li class="chapter" data-level="2.2.9" data-path="intro.html"><a href="intro.html#gantt-chart"><i class="fa fa-check"></i><b>2.2.9</b> Gantt chart</a></li>
<li class="chapter" data-level="2.2.10" data-path="intro.html"><a href="intro.html#heatmap"><i class="fa fa-check"></i><b>2.2.10</b> Heatmap</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="intro.html"><a href="intro.html#ggplot2-extra"><i class="fa fa-check"></i><b>2.3</b> ggplot2 extra</a><ul>
<li class="chapter" data-level="2.3.1" data-path="intro.html"><a href="intro.html#alluvial-and-sankey-diagram"><i class="fa fa-check"></i><b>2.3.1</b> Alluvial and Sankey diagram</a></li>
<li class="chapter" data-level="2.3.2" data-path="intro.html"><a href="intro.html#survival-plot"><i class="fa fa-check"></i><b>2.3.2</b> Survival plot</a></li>
<li class="chapter" data-level="2.3.3" data-path="intro.html"><a href="intro.html#ggraph-and-tidygraph"><i class="fa fa-check"></i><b>2.3.3</b> ggraph and tidygraph</a></li>
<li class="chapter" data-level="2.3.4" data-path="intro.html"><a href="intro.html#ggparty-decision-tree"><i class="fa fa-check"></i><b>2.3.4</b> ggparty-decision tree</a></li>
<li class="chapter" data-level="2.3.5" data-path="intro.html"><a href="intro.html#ggroc--roc-curve"><i class="fa fa-check"></i><b>2.3.5</b> ggROC- ROC curve</a></li>
<li class="chapter" data-level="2.3.6" data-path="intro.html"><a href="intro.html#map"><i class="fa fa-check"></i><b>2.3.6</b> Map</a></li>
<li class="chapter" data-level="2.3.7" data-path="intro.html"><a href="intro.html#ggwordcloud"><i class="fa fa-check"></i><b>2.3.7</b> ggwordcloud</a></li>
<li class="chapter" data-level="2.3.8" data-path="intro.html"><a href="intro.html#gganimate"><i class="fa fa-check"></i><b>2.3.8</b> gganimate</a></li>
<li class="chapter" data-level="2.3.9" data-path="intro.html"><a href="intro.html#ggneuro"><i class="fa fa-check"></i><b>2.3.9</b> ggneuro</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="intro.html"><a href="intro.html#plotly"><i class="fa fa-check"></i><b>2.4</b> plotly</a><ul>
<li class="chapter" data-level="2.4.1" data-path="intro.html"><a href="intro.html#scatter-plot-1"><i class="fa fa-check"></i><b>2.4.1</b> Scatter plot</a></li>
<li class="chapter" data-level="2.4.2" data-path="intro.html"><a href="intro.html#bar-plot-1"><i class="fa fa-check"></i><b>2.4.2</b> Bar plot</a></li>
<li class="chapter" data-level="2.4.3" data-path="intro.html"><a href="intro.html#heatmap-1"><i class="fa fa-check"></i><b>2.4.3</b> Heatmap</a></li>
<li class="chapter" data-level="2.4.4" data-path="intro.html"><a href="intro.html#map-1"><i class="fa fa-check"></i><b>2.4.4</b> map</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="data-wrangling.html"><a href="data-wrangling.html"><i class="fa fa-check"></i><b>3</b> Data Wrangling</a><ul>
<li class="chapter" data-level="3.1" data-path="data-wrangling.html"><a href="data-wrangling.html#data"><i class="fa fa-check"></i><b>3.1</b> Data</a><ul>
<li class="chapter" data-level="3.1.1" data-path="data-wrangling.html"><a href="data-wrangling.html#vector-arrays-matrix"><i class="fa fa-check"></i><b>3.1.1</b> Vector, Arrays, Matrix</a></li>
<li class="chapter" data-level="3.1.2" data-path="data-wrangling.html"><a href="data-wrangling.html#apply-lapply-sapply"><i class="fa fa-check"></i><b>3.1.2</b> apply, lapply, sapply</a></li>
<li class="chapter" data-level="3.1.3" data-path="data-wrangling.html"><a href="data-wrangling.html#simple-function"><i class="fa fa-check"></i><b>3.1.3</b> Simple function</a></li>
<li class="chapter" data-level="3.1.4" data-path="data-wrangling.html"><a href="data-wrangling.html#for-loop"><i class="fa fa-check"></i><b>3.1.4</b> for loop</a></li>
<li class="chapter" data-level="3.1.5" data-path="data-wrangling.html"><a href="data-wrangling.html#functional"><i class="fa fa-check"></i><b>3.1.5</b> Functional</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="data-wrangling.html"><a href="data-wrangling.html#data-storage"><i class="fa fa-check"></i><b>3.2</b> Data storage</a><ul>
<li class="chapter" data-level="3.2.1" data-path="data-wrangling.html"><a href="data-wrangling.html#data-frame"><i class="fa fa-check"></i><b>3.2.1</b> Data frame</a></li>
<li class="chapter" data-level="3.2.2" data-path="data-wrangling.html"><a href="data-wrangling.html#excel-data"><i class="fa fa-check"></i><b>3.2.2</b> Excel data</a></li>
<li class="chapter" data-level="3.2.3" data-path="data-wrangling.html"><a href="data-wrangling.html#foreign-data"><i class="fa fa-check"></i><b>3.2.3</b> Foreign data</a></li>
<li class="chapter" data-level="3.2.4" data-path="data-wrangling.html"><a href="data-wrangling.html#json-format"><i class="fa fa-check"></i><b>3.2.4</b> json format</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="data-wrangling.html"><a href="data-wrangling.html#tidy-data"><i class="fa fa-check"></i><b>3.3</b> Tidy data</a><ul>
<li class="chapter" data-level="3.3.1" data-path="data-wrangling.html"><a href="data-wrangling.html#factors"><i class="fa fa-check"></i><b>3.3.1</b> Factors</a></li>
<li class="chapter" data-level="3.3.2" data-path="data-wrangling.html"><a href="data-wrangling.html#multiple-files"><i class="fa fa-check"></i><b>3.3.2</b> Multiple files</a></li>
<li class="chapter" data-level="3.3.3" data-path="data-wrangling.html"><a href="data-wrangling.html#pivot"><i class="fa fa-check"></i><b>3.3.3</b> Pivot</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="data-wrangling.html"><a href="data-wrangling.html#regular-expressions"><i class="fa fa-check"></i><b>3.4</b> Regular Expressions</a><ul>
<li class="chapter" data-level="3.4.1" data-path="data-wrangling.html"><a href="data-wrangling.html#base-r"><i class="fa fa-check"></i><b>3.4.1</b> base R</a></li>
<li class="chapter" data-level="3.4.2" data-path="data-wrangling.html"><a href="data-wrangling.html#stringr"><i class="fa fa-check"></i><b>3.4.2</b> stringr</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="data-wrangling.html"><a href="data-wrangling.html#pdf-to-xcel"><i class="fa fa-check"></i><b>3.5</b> PDF to xcel</a><ul>
<li class="chapter" data-level="3.5.1" data-path="data-wrangling.html"><a href="data-wrangling.html#scanned-text-or-picture"><i class="fa fa-check"></i><b>3.5.1</b> Scanned text or picture</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="data-wrangling.html"><a href="data-wrangling.html#web-scraping"><i class="fa fa-check"></i><b>3.6</b> Web scraping</a></li>
<li class="chapter" data-level="3.7" data-path="data-wrangling.html"><a href="data-wrangling.html#manipulating-medical-images"><i class="fa fa-check"></i><b>3.7</b> Manipulating Medical Images</a><ul>
<li class="chapter" data-level="3.7.1" data-path="data-wrangling.html"><a href="data-wrangling.html#dicom-and-nifti-format"><i class="fa fa-check"></i><b>3.7.1</b> DICOM and nifti format</a></li>
<li class="chapter" data-level="3.7.2" data-path="data-wrangling.html"><a href="data-wrangling.html#manipulating-array-of-medical-images"><i class="fa fa-check"></i><b>3.7.2</b> Manipulating array of medical images</a></li>
<li class="chapter" data-level="3.7.3" data-path="data-wrangling.html"><a href="data-wrangling.html#combining-arrays"><i class="fa fa-check"></i><b>3.7.3</b> Combining arrays</a></li>
<li class="chapter" data-level="3.7.4" data-path="data-wrangling.html"><a href="data-wrangling.html#math-operation-on-multidimensional-array"><i class="fa fa-check"></i><b>3.7.4</b> Math operation on multidimensional array</a></li>
<li class="chapter" data-level="3.7.5" data-path="data-wrangling.html"><a href="data-wrangling.html#math-operation-on-list"><i class="fa fa-check"></i><b>3.7.5</b> Math operation on list</a></li>
<li class="chapter" data-level="3.7.6" data-path="data-wrangling.html"><a href="data-wrangling.html#vectorising-nifti-object"><i class="fa fa-check"></i><b>3.7.6</b> Vectorising nifti object</a></li>
<li class="chapter" data-level="3.7.7" data-path="data-wrangling.html"><a href="data-wrangling.html#tar-file"><i class="fa fa-check"></i><b>3.7.7</b> tar file</a></li>
<li class="chapter" data-level="3.7.8" data-path="data-wrangling.html"><a href="data-wrangling.html#image-registration"><i class="fa fa-check"></i><b>3.7.8</b> Image registration</a></li>
<li class="chapter" data-level="3.7.9" data-path="data-wrangling.html"><a href="data-wrangling.html#rescaling"><i class="fa fa-check"></i><b>3.7.9</b> Rescaling</a></li>
<li class="chapter" data-level="3.7.10" data-path="data-wrangling.html"><a href="data-wrangling.html#mni-template"><i class="fa fa-check"></i><b>3.7.10</b> MNI template</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="statistics.html"><a href="statistics.html"><i class="fa fa-check"></i><b>4</b> Statistics</a><ul>
<li class="chapter" data-level="4.1" data-path="statistics.html"><a href="statistics.html#univariable-analyses"><i class="fa fa-check"></i><b>4.1</b> Univariable analyses</a><ul>
<li class="chapter" data-level="4.1.1" data-path="statistics.html"><a href="statistics.html#parametric-tests"><i class="fa fa-check"></i><b>4.1.1</b> Parametric tests</a></li>
<li class="chapter" data-level="4.1.2" data-path="statistics.html"><a href="statistics.html#non-parametric-tests"><i class="fa fa-check"></i><b>4.1.2</b> Non-parametric tests</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="statistics.html"><a href="statistics.html#regression"><i class="fa fa-check"></i><b>4.2</b> Regression</a><ul>
<li class="chapter" data-level="4.2.1" data-path="statistics.html"><a href="statistics.html#brief-review-of-matrix"><i class="fa fa-check"></i><b>4.2.1</b> Brief review of matrix</a></li>
<li class="chapter" data-level="4.2.2" data-path="statistics.html"><a href="statistics.html#linear-least-square-regression"><i class="fa fa-check"></i><b>4.2.2</b> Linear (least square) regression</a></li>
<li class="chapter" data-level="4.2.3" data-path="statistics.html"><a href="statistics.html#logistic-regression"><i class="fa fa-check"></i><b>4.2.3</b> Logistic regression</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="statistics.html"><a href="statistics.html#special-types-of-regression"><i class="fa fa-check"></i><b>4.3</b> Special types of regression</a><ul>
<li class="chapter" data-level="4.3.1" data-path="statistics.html"><a href="statistics.html#ordinal-regression"><i class="fa fa-check"></i><b>4.3.1</b> Ordinal regression</a></li>
<li class="chapter" data-level="4.3.2" data-path="statistics.html"><a href="statistics.html#survival-analysis"><i class="fa fa-check"></i><b>4.3.2</b> Survival analysis</a></li>
<li class="chapter" data-level="4.3.3" data-path="statistics.html"><a href="statistics.html#quantile-regression"><i class="fa fa-check"></i><b>4.3.3</b> Quantile regression</a></li>
<li class="chapter" data-level="4.3.4" data-path="statistics.html"><a href="statistics.html#poisson-regression"><i class="fa fa-check"></i><b>4.3.4</b> Poisson regression</a></li>
<li class="chapter" data-level="4.3.5" data-path="statistics.html"><a href="statistics.html#conditional-logistic-regression"><i class="fa fa-check"></i><b>4.3.5</b> Conditional logistic regression</a></li>
<li class="chapter" data-level="4.3.6" data-path="statistics.html"><a href="statistics.html#multinomial-modelling"><i class="fa fa-check"></i><b>4.3.6</b> Multinomial modelling</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="statistics.html"><a href="statistics.html#sample-size-estimation"><i class="fa fa-check"></i><b>4.4</b> Sample size estimation</a><ul>
<li class="chapter" data-level="4.4.1" data-path="statistics.html"><a href="statistics.html#proportion"><i class="fa fa-check"></i><b>4.4.1</b> Proportion</a></li>
<li class="chapter" data-level="4.4.2" data-path="statistics.html"><a href="statistics.html#logistic-regression-1"><i class="fa fa-check"></i><b>4.4.2</b> Logistic regression</a></li>
<li class="chapter" data-level="4.4.3" data-path="statistics.html"><a href="statistics.html#survival-studies"><i class="fa fa-check"></i><b>4.4.3</b> Survival studies</a></li>
<li class="chapter" data-level="4.4.4" data-path="statistics.html"><a href="statistics.html#multiple-regression"><i class="fa fa-check"></i><b>4.4.4</b> Multiple regression</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="statistics.html"><a href="statistics.html#interpreting-clinical-trials"><i class="fa fa-check"></i><b>4.5</b> Interpreting clinical trials</a><ul>
<li class="chapter" data-level="4.5.1" data-path="statistics.html"><a href="statistics.html#nnt-from-arr"><i class="fa fa-check"></i><b>4.5.1</b> NNT from ARR</a></li>
<li class="chapter" data-level="4.5.2" data-path="statistics.html"><a href="statistics.html#nnt-from-odds-ratio"><i class="fa fa-check"></i><b>4.5.2</b> NNT from odds ratio</a></li>
<li class="chapter" data-level="4.5.3" data-path="statistics.html"><a href="statistics.html#nnt-from-hazard-ratio"><i class="fa fa-check"></i><b>4.5.3</b> NNT from hazard ratio</a></li>
<li class="chapter" data-level="4.5.4" data-path="statistics.html"><a href="statistics.html#nnt-from-metaananlysis"><i class="fa fa-check"></i><b>4.5.4</b> NNT from metaananlysis</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="statistics.html"><a href="statistics.html#metaanalysis"><i class="fa fa-check"></i><b>4.6</b> Metaanalysis</a><ul>
<li class="chapter" data-level="4.6.1" data-path="statistics.html"><a href="statistics.html#prisma"><i class="fa fa-check"></i><b>4.6.1</b> PRISMA</a></li>
<li class="chapter" data-level="4.6.2" data-path="statistics.html"><a href="statistics.html#conversion-of-mean-and-median"><i class="fa fa-check"></i><b>4.6.2</b> Conversion of mean and median</a></li>
<li class="chapter" data-level="4.6.3" data-path="statistics.html"><a href="statistics.html#inconsistency-i2"><i class="fa fa-check"></i><b>4.6.3</b> Inconsistency I2</a></li>
<li class="chapter" data-level="4.6.4" data-path="statistics.html"><a href="statistics.html#metaanalysis-of-proportion"><i class="fa fa-check"></i><b>4.6.4</b> Metaanalysis of proportion</a></li>
<li class="chapter" data-level="4.6.5" data-path="statistics.html"><a href="statistics.html#bivariate-metaanalysis"><i class="fa fa-check"></i><b>4.6.5</b> Bivariate Metaanalysis</a></li>
<li class="chapter" data-level="4.6.6" data-path="statistics.html"><a href="statistics.html#metaanalysis-of-clinical-trial."><i class="fa fa-check"></i><b>4.6.6</b> Metaanalysis of clinical trial.</a></li>
<li class="chapter" data-level="4.6.7" data-path="statistics.html"><a href="statistics.html#metaregression"><i class="fa fa-check"></i><b>4.6.7</b> Metaregression</a></li>
</ul></li>
<li class="chapter" data-level="4.7" data-path="statistics.html"><a href="statistics.html#data-simulation"><i class="fa fa-check"></i><b>4.7</b> Data simulation</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html"><i class="fa fa-check"></i><b>5</b> Multivariate Analysis</a><ul>
<li class="chapter" data-level="5.1" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#multivariate-regression"><i class="fa fa-check"></i><b>5.1</b> Multivariate regression</a><ul>
<li class="chapter" data-level="5.1.1" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#penalised-regression"><i class="fa fa-check"></i><b>5.1.1</b> Penalised regression</a></li>
<li class="chapter" data-level="5.1.2" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#mars"><i class="fa fa-check"></i><b>5.1.2</b> MARS</a></li>
<li class="chapter" data-level="5.1.3" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#mixed-modelling"><i class="fa fa-check"></i><b>5.1.3</b> Mixed modelling</a></li>
<li class="chapter" data-level="5.1.4" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#trajectory-modelling"><i class="fa fa-check"></i><b>5.1.4</b> Trajectory modelling</a></li>
<li class="chapter" data-level="5.1.5" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#generalized-estimating-equation-gee"><i class="fa fa-check"></i><b>5.1.5</b> Generalized estimating equation (GEE)</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#principal-component-analysis"><i class="fa fa-check"></i><b>5.2</b> Principal component analysis</a></li>
<li class="chapter" data-level="5.3" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#independent-component-analysis"><i class="fa fa-check"></i><b>5.3</b> Independent component analysis</a></li>
<li class="chapter" data-level="5.4" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#partial-least-squares"><i class="fa fa-check"></i><b>5.4</b> Partial least squares</a></li>
<li class="chapter" data-level="5.5" data-path="multivariate-analysis.html"><a href="multivariate-analysis.html#causal-inference"><i class="fa fa-check"></i><b>5.5</b> Causal inference</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="machine-learning.html"><a href="machine-learning.html"><i class="fa fa-check"></i><b>6</b> Machine learning</a><ul>
<li class="chapter" data-level="6.1" data-path="machine-learning.html"><a href="machine-learning.html#decision-tree-analysis"><i class="fa fa-check"></i><b>6.1</b> Decision tree analysis</a><ul>
<li class="chapter" data-level="6.1.1" data-path="machine-learning.html"><a href="machine-learning.html#information-theory-driven"><i class="fa fa-check"></i><b>6.1.1</b> Information theory driven</a></li>
<li class="chapter" data-level="6.1.2" data-path="machine-learning.html"><a href="machine-learning.html#conditional-decision-tree"><i class="fa fa-check"></i><b>6.1.2</b> Conditional decision tree</a></li>
<li class="chapter" data-level="6.1.3" data-path="machine-learning.html"><a href="machine-learning.html#criticisms-of-decision-tree"><i class="fa fa-check"></i><b>6.1.3</b> criticisms of decision tree</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="machine-learning.html"><a href="machine-learning.html#random-forest"><i class="fa fa-check"></i><b>6.2</b> Random Forest</a></li>
<li class="chapter" data-level="6.3" data-path="machine-learning.html"><a href="machine-learning.html#gradient-boost-machine"><i class="fa fa-check"></i><b>6.3</b> Gradient Boost Machine</a><ul>
<li class="chapter" data-level="6.3.1" data-path="machine-learning.html"><a href="machine-learning.html#extreme-gradient-boost-machine"><i class="fa fa-check"></i><b>6.3.1</b> Extreme gradient boost machine</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="machine-learning.html"><a href="machine-learning.html#knn"><i class="fa fa-check"></i><b>6.4</b> KNN</a></li>
<li class="chapter" data-level="6.5" data-path="machine-learning.html"><a href="machine-learning.html#support-vector-machine"><i class="fa fa-check"></i><b>6.5</b> Support vector machine</a><ul>
<li class="chapter" data-level="6.5.1" data-path="machine-learning.html"><a href="machine-learning.html#survival-analysis-using-random-forest"><i class="fa fa-check"></i><b>6.5.1</b> Survival analysis using random forest</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="machine-learning.html"><a href="machine-learning.html#non-negative-matrix-factorisation"><i class="fa fa-check"></i><b>6.6</b> Non-negative matrix factorisation</a></li>
<li class="chapter" data-level="6.7" data-path="machine-learning.html"><a href="machine-learning.html#formal-concept-analysis"><i class="fa fa-check"></i><b>6.7</b> Formal concept analysis</a></li>
<li class="chapter" data-level="6.8" data-path="machine-learning.html"><a href="machine-learning.html#evolutionary-algorithm"><i class="fa fa-check"></i><b>6.8</b> Evolutionary Algorithm</a><ul>
<li class="chapter" data-level="6.8.1" data-path="machine-learning.html"><a href="machine-learning.html#simulated-annealing"><i class="fa fa-check"></i><b>6.8.1</b> Simulated Annealing</a></li>
<li class="chapter" data-level="6.8.2" data-path="machine-learning.html"><a href="machine-learning.html#genetic-algorithm"><i class="fa fa-check"></i><b>6.8.2</b> Genetic Algorithm</a></li>
</ul></li>
<li class="chapter" data-level="6.9" data-path="machine-learning.html"><a href="machine-learning.html#manifold-learning"><i class="fa fa-check"></i><b>6.9</b> Manifold learning</a><ul>
<li class="chapter" data-level="6.9.1" data-path="machine-learning.html"><a href="machine-learning.html#t-stochastic-neighbourhood-embedding"><i class="fa fa-check"></i><b>6.9.1</b> T-Stochastic Neighbourhood Embedding</a></li>
<li class="chapter" data-level="6.9.2" data-path="machine-learning.html"><a href="machine-learning.html#self-organising-map"><i class="fa fa-check"></i><b>6.9.2</b> Self organising map</a></li>
</ul></li>
<li class="chapter" data-level="6.10" data-path="machine-learning.html"><a href="machine-learning.html#deep-learning"><i class="fa fa-check"></i><b>6.10</b> Deep learning</a><ul>
<li class="chapter" data-level="6.10.1" data-path="machine-learning.html"><a href="machine-learning.html#multiplayer-perceptron"><i class="fa fa-check"></i><b>6.10.1</b> Multiplayer Perceptron</a></li>
<li class="chapter" data-level="6.10.2" data-path="machine-learning.html"><a href="machine-learning.html#cnn"><i class="fa fa-check"></i><b>6.10.2</b> CNN</a></li>
<li class="chapter" data-level="6.10.3" data-path="machine-learning.html"><a href="machine-learning.html#rnn"><i class="fa fa-check"></i><b>6.10.3</b> RNN</a></li>
<li class="chapter" data-level="6.10.4" data-path="machine-learning.html"><a href="machine-learning.html#reinforcement-learning"><i class="fa fa-check"></i><b>6.10.4</b> Reinforcement learning</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="machine-learning-part-2.html"><a href="machine-learning-part-2.html"><i class="fa fa-check"></i><b>7</b> Machine Learning Part 2</a><ul>
<li class="chapter" data-level="7.1" data-path="machine-learning-part-2.html"><a href="machine-learning-part-2.html#bag-of-words"><i class="fa fa-check"></i><b>7.1</b> Bag of words</a><ul>
<li class="chapter" data-level="7.1.1" data-path="machine-learning-part-2.html"><a href="machine-learning-part-2.html#tfidf"><i class="fa fa-check"></i><b>7.1.1</b> TFIDF</a></li>
<li class="chapter" data-level="7.1.2" data-path="machine-learning-part-2.html"><a href="machine-learning-part-2.html#extracting-data-from-web"><i class="fa fa-check"></i><b>7.1.2</b> Extracting data from web</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="machine-learning-part-2.html"><a href="machine-learning-part-2.html#wordcloud"><i class="fa fa-check"></i><b>7.2</b> Wordcloud</a></li>
<li class="chapter" data-level="7.3" data-path="machine-learning-part-2.html"><a href="machine-learning-part-2.html#bigram-analysis"><i class="fa fa-check"></i><b>7.3</b> Bigram analysis</a></li>
<li class="chapter" data-level="7.4" data-path="machine-learning-part-2.html"><a href="machine-learning-part-2.html#trigram"><i class="fa fa-check"></i><b>7.4</b> Trigram</a></li>
<li class="chapter" data-level="7.5" data-path="machine-learning-part-2.html"><a href="machine-learning-part-2.html#topic-modeling-or-thematic-analysis"><i class="fa fa-check"></i><b>7.5</b> Topic modeling or thematic analysis</a><ul>
<li class="chapter" data-level="7.5.1" data-path="machine-learning-part-2.html"><a href="machine-learning-part-2.html#probabilistic-topic-model"><i class="fa fa-check"></i><b>7.5.1</b> Probabilistic topic model</a></li>
<li class="chapter" data-level="7.5.2" data-path="machine-learning-part-2.html"><a href="machine-learning-part-2.html#nmf"><i class="fa fa-check"></i><b>7.5.2</b> NMF</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="bayesian-analysis.html"><a href="bayesian-analysis.html"><i class="fa fa-check"></i><b>8</b> Bayesian Analysis</a><ul>
<li class="chapter" data-level="8.1" data-path="bayesian-analysis.html"><a href="bayesian-analysis.html#baysian-belief"><i class="fa fa-check"></i><b>8.1</b> Baysian belief</a><ul>
<li class="chapter" data-level="8.1.1" data-path="bayesian-analysis.html"><a href="bayesian-analysis.html#conditional-probability"><i class="fa fa-check"></i><b>8.1.1</b> Conditional probability</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="bayesian-analysis.html"><a href="bayesian-analysis.html#markov-model"><i class="fa fa-check"></i><b>8.2</b> Markov model</a></li>
<li class="chapter" data-level="8.3" data-path="bayesian-analysis.html"><a href="bayesian-analysis.html#inla-stan-and-bugs"><i class="fa fa-check"></i><b>8.3</b> INLA, Stan and BUGS</a><ul>
<li class="chapter" data-level="8.3.1" data-path="bayesian-analysis.html"><a href="bayesian-analysis.html#linear-regression"><i class="fa fa-check"></i><b>8.3.1</b> Linear regression</a></li>
<li class="chapter" data-level="8.3.2" data-path="bayesian-analysis.html"><a href="bayesian-analysis.html#logistic-regression-2"><i class="fa fa-check"></i><b>8.3.2</b> Logistic regression</a></li>
<li class="chapter" data-level="8.3.3" data-path="bayesian-analysis.html"><a href="bayesian-analysis.html#mixed-model"><i class="fa fa-check"></i><b>8.3.3</b> Mixed model</a></li>
<li class="chapter" data-level="8.3.4" data-path="bayesian-analysis.html"><a href="bayesian-analysis.html#bayesian-metaanalysis"><i class="fa fa-check"></i><b>8.3.4</b> Bayesian Metaanalysis</a></li>
<li class="chapter" data-level="8.3.5" data-path="bayesian-analysis.html"><a href="bayesian-analysis.html#cost"><i class="fa fa-check"></i><b>8.3.5</b> Cost</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="operational-research.html"><a href="operational-research.html"><i class="fa fa-check"></i><b>9</b> Operational Research</a><ul>
<li class="chapter" data-level="9.1" data-path="operational-research.html"><a href="operational-research.html#queueing-theory"><i class="fa fa-check"></i><b>9.1</b> Queueing theory</a></li>
<li class="chapter" data-level="9.2" data-path="operational-research.html"><a href="operational-research.html#discrete-event-simulations"><i class="fa fa-check"></i><b>9.2</b> Discrete Event Simulations</a><ul>
<li class="chapter" data-level="9.2.1" data-path="operational-research.html"><a href="operational-research.html#simulate-capacity-of-system"><i class="fa fa-check"></i><b>9.2.1</b> Simulate capacity of system</a></li>
<li class="chapter" data-level="9.2.2" data-path="operational-research.html"><a href="operational-research.html#queuing-network"><i class="fa fa-check"></i><b>9.2.2</b> Queuing network</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="operational-research.html"><a href="operational-research.html#linear-programming"><i class="fa fa-check"></i><b>9.3</b> Linear Programming</a></li>
<li class="chapter" data-level="9.4" data-path="operational-research.html"><a href="operational-research.html#forecasting"><i class="fa fa-check"></i><b>9.4</b> Forecasting</a><ul>
<li class="chapter" data-level="9.4.1" data-path="operational-research.html"><a href="operational-research.html#bed-requirement"><i class="fa fa-check"></i><b>9.4.1</b> Bed requirement</a></li>
<li class="chapter" data-level="9.4.2" data-path="operational-research.html"><a href="operational-research.html#length-of-stay"><i class="fa fa-check"></i><b>9.4.2</b> Length of stay</a></li>
<li class="chapter" data-level="9.4.3" data-path="operational-research.html"><a href="operational-research.html#customer-churns"><i class="fa fa-check"></i><b>9.4.3</b> Customer churns</a></li>
</ul></li>
<li class="chapter" data-level="9.5" data-path="operational-research.html"><a href="operational-research.html#process-mapping"><i class="fa fa-check"></i><b>9.5</b> Process mapping</a></li>
<li class="chapter" data-level="9.6" data-path="operational-research.html"><a href="operational-research.html#supply-chains"><i class="fa fa-check"></i><b>9.6</b> Supply chains</a></li>
<li class="chapter" data-level="9.7" data-path="operational-research.html"><a href="operational-research.html#health-economics"><i class="fa fa-check"></i><b>9.7</b> Health economics</a><ul>
<li class="chapter" data-level="9.7.1" data-path="operational-research.html"><a href="operational-research.html#cost-1"><i class="fa fa-check"></i><b>9.7.1</b> Cost</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="graph-theory.html"><a href="graph-theory.html"><i class="fa fa-check"></i><b>10</b> Graph Theory</a><ul>
<li class="chapter" data-level="10.1" data-path="graph-theory.html"><a href="graph-theory.html#special-graphs"><i class="fa fa-check"></i><b>10.1</b> Special graphs</a><ul>
<li class="chapter" data-level="10.1.1" data-path="graph-theory.html"><a href="graph-theory.html#laplacian-matrix"><i class="fa fa-check"></i><b>10.1.1</b> Laplacian matrix</a></li>
<li class="chapter" data-level="10.1.2" data-path="graph-theory.html"><a href="graph-theory.html#bimodal-bipartite-graph"><i class="fa fa-check"></i><b>10.1.2</b> Bimodal (bipartite) graph</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="graph-theory.html"><a href="graph-theory.html#centrality-measures"><i class="fa fa-check"></i><b>10.2</b> Centrality Measures</a><ul>
<li class="chapter" data-level="10.2.1" data-path="graph-theory.html"><a href="graph-theory.html#local-centrality-measures"><i class="fa fa-check"></i><b>10.2.1</b> Local centrality measures</a></li>
<li class="chapter" data-level="10.2.2" data-path="graph-theory.html"><a href="graph-theory.html#global-centrality-measures"><i class="fa fa-check"></i><b>10.2.2</b> Global centrality measures</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="graph-theory.html"><a href="graph-theory.html#community"><i class="fa fa-check"></i><b>10.3</b> Community</a></li>
<li class="chapter" data-level="10.4" data-path="graph-theory.html"><a href="graph-theory.html#visualising-graph"><i class="fa fa-check"></i><b>10.4</b> Visualising graph</a><ul>
<li class="chapter" data-level="10.4.1" data-path="graph-theory.html"><a href="graph-theory.html#visnetwork"><i class="fa fa-check"></i><b>10.4.1</b> Visnetwork</a></li>
<li class="chapter" data-level="10.4.2" data-path="graph-theory.html"><a href="graph-theory.html#large-graph"><i class="fa fa-check"></i><b>10.4.2</b> Large graph</a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="graph-theory.html"><a href="graph-theory.html#social-media-and-network-analysis"><i class="fa fa-check"></i><b>10.5</b> Social Media and Network Analysis</a><ul>
<li class="chapter" data-level="10.5.1" data-path="graph-theory.html"><a href="graph-theory.html#twitter"><i class="fa fa-check"></i><b>10.5.1</b> Twitter</a></li>
<li class="chapter" data-level="10.5.2" data-path="graph-theory.html"><a href="graph-theory.html#youtube"><i class="fa fa-check"></i><b>10.5.2</b> Youtube</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="geospatial-analysis.html"><a href="geospatial-analysis.html"><i class="fa fa-check"></i><b>11</b> Geospatial analysis</a><ul>
<li class="chapter" data-level="11.1" data-path="geospatial-analysis.html"><a href="geospatial-analysis.html#geocoding"><i class="fa fa-check"></i><b>11.1</b> Geocoding</a><ul>
<li class="chapter" data-level="11.1.1" data-path="geospatial-analysis.html"><a href="geospatial-analysis.html#openstreetmap"><i class="fa fa-check"></i><b>11.1.1</b> OpenStreetMap</a></li>
<li class="chapter" data-level="11.1.2" data-path="geospatial-analysis.html"><a href="geospatial-analysis.html#google-maps-api"><i class="fa fa-check"></i><b>11.1.2</b> Google Maps API</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="geospatial-analysis.html"><a href="geospatial-analysis.html#sp-and-sf-objects"><i class="fa fa-check"></i><b>11.2</b> Sp and sf objects</a></li>
<li class="chapter" data-level="11.3" data-path="geospatial-analysis.html"><a href="geospatial-analysis.html#thematic-map-1"><i class="fa fa-check"></i><b>11.3</b> Thematic map</a><ul>
<li class="chapter" data-level="11.3.1" data-path="geospatial-analysis.html"><a href="geospatial-analysis.html#calculate-distance-to-hospital-openstreetmap"><i class="fa fa-check"></i><b>11.3.1</b> Calculate distance to Hospital-OpenStreetMap</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="geospatial-analysis.html"><a href="geospatial-analysis.html#spatial-regression"><i class="fa fa-check"></i><b>11.4</b> Spatial regression</a><ul>
<li class="chapter" data-level="11.4.1" data-path="geospatial-analysis.html"><a href="geospatial-analysis.html#new-york-covid-19-mortality"><i class="fa fa-check"></i><b>11.4.1</b> New York COVID-19 mortality</a></li>
<li class="chapter" data-level="11.4.2" data-path="geospatial-analysis.html"><a href="geospatial-analysis.html#danish-stroke-registry"><i class="fa fa-check"></i><b>11.4.2</b> Danish Stroke Registry</a></li>
<li class="chapter" data-level="11.4.3" data-path="geospatial-analysis.html"><a href="geospatial-analysis.html#inla"><i class="fa fa-check"></i><b>11.4.3</b> INLA</a></li>
<li class="chapter" data-level="11.4.4" data-path="geospatial-analysis.html"><a href="geospatial-analysis.html#stan"><i class="fa fa-check"></i><b>11.4.4</b> Stan</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="geospatial-analysis.html"><a href="geospatial-analysis.html#machine-learning-1"><i class="fa fa-check"></i><b>11.5</b> Machine learning</a></li>
<li class="chapter" data-level="11.6" data-path="geospatial-analysis.html"><a href="geospatial-analysis.html#spatio-temporal-regression"><i class="fa fa-check"></i><b>11.6</b> Spatio-temporal regression</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="app.html"><a href="app.html"><i class="fa fa-check"></i><b>12</b> App</a><ul>
<li class="chapter" data-level="12.1" data-path="app.html"><a href="app.html#brief-introduction-to-shiny-app"><i class="fa fa-check"></i><b>12.1</b> Brief introduction to Shiny app</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="appendix.html"><a href="appendix.html"><i class="fa fa-check"></i><b>13</b> Appendix</a><ul>
<li class="chapter" data-level="13.1" data-path="appendix.html"><a href="appendix.html#brief-introduction-to-matrix"><i class="fa fa-check"></i><b>13.1</b> Brief introduction to Matrix</a></li>
<li class="chapter" data-level="13.2" data-path="appendix.html"><a href="appendix.html#regression-1"><i class="fa fa-check"></i><b>13.2</b> Regression</a><ul>
<li class="chapter" data-level="13.2.1" data-path="appendix.html"><a href="appendix.html#linear-regression-1"><i class="fa fa-check"></i><b>13.2.1</b> Linear regression</a></li>
<li class="chapter" data-level="13.2.2" data-path="appendix.html"><a href="appendix.html#logistic-regression-3"><i class="fa fa-check"></i><b>13.2.2</b> Logistic regression</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Applications of R in Healthcare</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data-wrangling" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> Data Wrangling</h1>
<p>This section deals with the nitty gritty of data analysis. There’s no nice plots like the previous chapter. In fact, this data wrangling is the major aspect of data science.</p>
<div id="data" class="section level2">
<h2><span class="header-section-number">3.1</span> Data</h2>
<p>This section on atomic vector, arrays, matrix and list is often considered
boring and ignored.</p>
<div id="vector-arrays-matrix" class="section level3">
<h3><span class="header-section-number">3.1.1</span> Vector, Arrays, Matrix</h3>
<div id="vector-and-list" class="section level4">
<h4><span class="header-section-number">3.1.1.1</span> Vector and list</h4>
<p>All elements of an atomic vector and arrays are the same. A vector has one dimensional and a matrix has 2 or more dimension. List can contain different types of data. A complex example of a structured list is the <em>json</em> format shown below. In base R, <em>c</em> is a function for creating a vector or list. The function <em>list</em> can also be used to create list. It is best to avoid using
<em>c</em> when assigning a name to a dataframe or vector <span class="citation">(Wickham <a href="#ref-Wickham2019" role="doc-biblioref">2019</a>)</span>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="data-wrangling.html#cb1-1"></a>a&lt;-<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>)</span>
<span id="cb1-2"><a href="data-wrangling.html#cb1-2"></a><span class="kw">is.vector</span>(a)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="data-wrangling.html#cb3-1"></a><span class="kw">class</span>(a) <span class="co">#numeric vector</span></span></code></pre></div>
<pre><code>## [1] &quot;numeric&quot;</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="data-wrangling.html#cb5-1"></a>b&lt;-<span class="kw">c</span>(<span class="st">&quot;apple&quot;</span>,<span class="st">&quot;orange&quot;</span>,<span class="st">&quot;banana&quot;</span>)</span>
<span id="cb5-2"><a href="data-wrangling.html#cb5-2"></a><span class="kw">is.vector</span>(b) <span class="co">#character vector</span></span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="data-wrangling.html#cb7-1"></a><span class="kw">class</span>(b)</span></code></pre></div>
<pre><code>## [1] &quot;character&quot;</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="data-wrangling.html#cb9-1"></a>d&lt;-<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="st">&quot;banana&quot;</span>)</span>
<span id="cb9-2"><a href="data-wrangling.html#cb9-2"></a><span class="kw">is.list</span>(d) <span class="co">#character vector</span></span></code></pre></div>
<pre><code>## [1] FALSE</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="data-wrangling.html#cb11-1"></a><span class="kw">class</span>(d) <span class="co">#FALSE</span></span></code></pre></div>
<pre><code>## [1] &quot;character&quot;</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="data-wrangling.html#cb13-1"></a>e&lt;-<span class="kw">list</span>(a,b,<span class="kw">c</span>(<span class="ot">TRUE</span>,<span class="ot">FALSE</span>,<span class="ot">FALSE</span>))</span>
<span id="cb13-2"><a href="data-wrangling.html#cb13-2"></a><span class="kw">is.list</span>(e) <span class="co">#TRUE</span></span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
</div>
<div id="matrix-and-arrays" class="section level4">
<h4><span class="header-section-number">3.1.1.2</span> Matrix and Arrays</h4>
<p>Arrays are arranged in rows and columns of a single data type. It contains a description of the number of dimension. Array can also be accessed by its indexing. Later in this chapter, we will illustrate the importance of arrays for manipulating MRI scans. A volumetric mri scan, there are 3 dimenions <em>[,,]</em>. The first column is the sagittal sequence, second column is the coronal sequence and the third column is the axial sequence. In the example below, this knowledge of arrays can be used to reverse the ordering of information on MRI scan (flip on it axis between left and right)</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="data-wrangling.html#cb15-1"></a><span class="co">#vector</span></span>
<span id="cb15-2"><a href="data-wrangling.html#cb15-2"></a>vector1&lt;-<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>)</span>
<span id="cb15-3"><a href="data-wrangling.html#cb15-3"></a>vector2&lt;-<span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">7</span>,<span class="dv">8</span>,<span class="dv">9</span>)</span>
<span id="cb15-4"><a href="data-wrangling.html#cb15-4"></a>array1&lt;-<span class="kw">array</span>(<span class="kw">c</span>(vector1,vector2),<span class="dt">dim =</span> <span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">3</span>,<span class="dv">2</span>))</span>
<span id="cb15-5"><a href="data-wrangling.html#cb15-5"></a>array1</span></code></pre></div>
<pre><code>## , , 1
## 
##      [,1] [,2] [,3]
## [1,]    1    4    7
## [2,]    2    5    8
## [3,]    3    6    9
## 
## , , 2
## 
##      [,1] [,2] [,3]
## [1,]    1    4    7
## [2,]    2    5    8
## [3,]    3    6    9</code></pre>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="data-wrangling.html#cb17-1"></a>array2&lt;-<span class="kw">array</span>(<span class="kw">c</span>(vector1,vector2),<span class="dt">dim =</span> <span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>))</span>
<span id="cb17-2"><a href="data-wrangling.html#cb17-2"></a>array2</span></code></pre></div>
<pre><code>## , , 1
## 
##      [,1] [,2]
## [1,]    1    3
## [2,]    2    4
## 
## , , 2
## 
##      [,1] [,2]
## [1,]    5    7
## [2,]    6    8
## 
## , , 3
## 
##      [,1] [,2]
## [1,]    9    2
## [2,]    1    3</code></pre>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="data-wrangling.html#cb19-1"></a><span class="co">#check if array or matrix</span></span>
<span id="cb19-2"><a href="data-wrangling.html#cb19-2"></a><span class="kw">is.matrix</span>(array1)</span></code></pre></div>
<pre><code>## [1] FALSE</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="data-wrangling.html#cb21-1"></a><span class="kw">is.array</span>(array1)</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p>Arrays can be accessed by indexing its structure.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="data-wrangling.html#cb23-1"></a>array1[,,<span class="dv">1</span>]</span></code></pre></div>
<pre><code>##      [,1] [,2] [,3]
## [1,]    1    4    7
## [2,]    2    5    8
## [3,]    3    6    9</code></pre>
<p>The second array of array1</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="data-wrangling.html#cb25-1"></a>array1[,,<span class="dv">2</span>]</span></code></pre></div>
<pre><code>##      [,1] [,2] [,3]
## [1,]    1    4    7
## [2,]    2    5    8
## [3,]    3    6    9</code></pre>
<p>First row of array1</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="data-wrangling.html#cb27-1"></a>array1[<span class="dv">1</span>,,]</span></code></pre></div>
<pre><code>##      [,1] [,2]
## [1,]    1    1
## [2,]    4    4
## [3,]    7    7</code></pre>
<p>First column of array1</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="data-wrangling.html#cb29-1"></a>array1[,<span class="dv">1</span>,]</span></code></pre></div>
<pre><code>##      [,1] [,2]
## [1,]    1    1
## [2,]    2    2
## [3,]    3    3</code></pre>
</div>
</div>
<div id="apply-lapply-sapply" class="section level3">
<h3><span class="header-section-number">3.1.2</span> apply, lapply, sapply</h3>
<div id="apply" class="section level4">
<h4><span class="header-section-number">3.1.2.1</span> apply</h4>
<p>The <em>apply</em> function works on data in array format.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="data-wrangling.html#cb31-1"></a><span class="co">#sum data across rows ie c(1)</span></span>
<span id="cb31-2"><a href="data-wrangling.html#cb31-2"></a><span class="kw">apply</span>(array1,<span class="kw">c</span>(<span class="dv">1</span>),sum)</span></code></pre></div>
<pre><code>## [1] 24 30 36</code></pre>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="data-wrangling.html#cb33-1"></a><span class="co">#sum data across columns(2)</span></span>
<span id="cb33-2"><a href="data-wrangling.html#cb33-2"></a><span class="kw">apply</span>(array1,<span class="kw">c</span>(<span class="dv">2</span>),sum)</span></code></pre></div>
<pre><code>## [1] 12 30 48</code></pre>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="data-wrangling.html#cb35-1"></a><span class="co">#sum data across rows and columns c(1,2)</span></span>
<span id="cb35-2"><a href="data-wrangling.html#cb35-2"></a><span class="kw">apply</span>(array1,<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>),sum)</span></code></pre></div>
<pre><code>##      [,1] [,2] [,3]
## [1,]    2    8   14
## [2,]    4   10   16
## [3,]    6   12   18</code></pre>
</div>
<div id="lapply" class="section level4">
<h4><span class="header-section-number">3.1.2.2</span> lapply</h4>
<p>The call <em>lapply</em> applies a function to a list. In the section below of medical images the <em>lapply</em> function will be used to creates a list of nifti files and which can be opened painlessly with additional call to <em>readNIfTI</em>. Here a more simple example is used.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="data-wrangling.html#cb37-1"></a>a&lt;-<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">6</span>,<span class="dv">7</span>,<span class="dv">8</span>)</span>
<span id="cb37-2"><a href="data-wrangling.html#cb37-2"></a><span class="kw">lapply</span>(a, <span class="cf">function</span>(x) x<span class="op">^</span><span class="dv">3</span>)</span></code></pre></div>
<pre><code>## [[1]]
## [1] 1
## 
## [[2]]
## [1] 8
## 
## [[3]]
## [1] 27
## 
## [[4]]
## [1] 64
## 
## [[5]]
## [1] 125
## 
## [[6]]
## [1] 216
## 
## [[7]]
## [1] 343
## 
## [[8]]
## [1] 512</code></pre>
</div>
<div id="sapply" class="section level4">
<h4><span class="header-section-number">3.1.2.3</span> sapply</h4>
<p>The call <em>sapply</em> applies a function to a vector, matrix or list. It returns the results in the form of a matrix.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="data-wrangling.html#cb39-1"></a>a&lt;-<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="dv">4</span>)</span>
<span id="cb39-2"><a href="data-wrangling.html#cb39-2"></a><span class="kw">sapply</span>(a, <span class="cf">function</span>(x) x<span class="op">^</span><span class="dv">2</span>)</span></code></pre></div>
<pre><code>## [1]  1  4  9 16</code></pre>
</div>
<div id="tapply" class="section level4">
<h4><span class="header-section-number">3.1.2.4</span> tapply</h4>
<p>The <em>tapply</em> function applies a function to a subset of the data.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="data-wrangling.html#cb41-1"></a>dat.array1&lt;-<span class="kw">as.data.frame.array</span>(array1)</span>
<span id="cb41-2"><a href="data-wrangling.html#cb41-2"></a>dat.array1</span></code></pre></div>
<pre><code>##   V1 V2 V3 V4 V5 V6
## 1  1  4  7  1  4  7
## 2  2  5  8  2  5  8
## 3  3  6  9  3  6  9</code></pre>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="data-wrangling.html#cb43-1"></a>dat.array1<span class="op">$</span>V6&lt;-<span class="st"> </span>dat.array1<span class="op">$</span>V6 <span class="op">%%</span><span class="st"> </span><span class="dv">2</span></span>
<span id="cb43-2"><a href="data-wrangling.html#cb43-2"></a>dat.array1<span class="op">$</span>V6</span></code></pre></div>
<pre><code>## [1] 1 0 1</code></pre>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="data-wrangling.html#cb45-1"></a><span class="kw">tapply</span>(dat.array1<span class="op">$</span>V1, dat.array1<span class="op">$</span>V6, sum) </span></code></pre></div>
<pre><code>## 0 1 
## 2 4</code></pre>
<p>The <em>rapply</em> function or recursive apply is applied to a list, contingent on the second argument. In this example, the first function is to multiple elements of the list by 2 contingent on the element being numeric.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="data-wrangling.html#cb47-1"></a>a&lt;-<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="fl">3.4</span>,D,<span class="dv">5</span>, <span class="st">&quot;NA&quot;</span>)</span>
<span id="cb47-2"><a href="data-wrangling.html#cb47-2"></a></span>
<span id="cb47-3"><a href="data-wrangling.html#cb47-3"></a><span class="kw">rapply</span>(a, <span class="cf">function</span>(x) x<span class="op">*</span><span class="dv">2</span>, <span class="dt">class=</span><span class="st">&quot;numeric&quot;</span>)</span></code></pre></div>
<pre><code>## [1]  2.0  4.0  6.8 10.0</code></pre>
</div>
</div>
<div id="simple-function" class="section level3">
<h3><span class="header-section-number">3.1.3</span> Simple function</h3>
<p>a function is written by creating name of function, calling on <em>function</em> (x)
and brackets to define function.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="data-wrangling.html#cb49-1"></a><span class="co"># function sun</span></span>
<span id="cb49-2"><a href="data-wrangling.html#cb49-2"></a>ArithSum&lt;-<span class="cf">function</span> (x) {</span>
<span id="cb49-3"><a href="data-wrangling.html#cb49-3"></a>  <span class="kw">sum</span>(x)</span>
<span id="cb49-4"><a href="data-wrangling.html#cb49-4"></a>}</span>
<span id="cb49-5"><a href="data-wrangling.html#cb49-5"></a></span>
<span id="cb49-6"><a href="data-wrangling.html#cb49-6"></a>vector1&lt;-<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>)</span>
<span id="cb49-7"><a href="data-wrangling.html#cb49-7"></a><span class="kw">ArithSum</span>(vector1)</span></code></pre></div>
<pre><code>## [1] 6</code></pre>
</div>
<div id="for-loop" class="section level3">
<h3><span class="header-section-number">3.1.4</span> for loop</h3>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="data-wrangling.html#cb51-1"></a>vector2&lt;-<span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">6</span>)</span>
<span id="cb51-2"><a href="data-wrangling.html#cb51-2"></a></span>
<span id="cb51-3"><a href="data-wrangling.html#cb51-3"></a><span class="cf">for</span> (i <span class="cf">in</span> vector2){</span>
<span id="cb51-4"><a href="data-wrangling.html#cb51-4"></a>  <span class="kw">print</span>(i)</span>
<span id="cb51-5"><a href="data-wrangling.html#cb51-5"></a>}</span></code></pre></div>
<pre><code>## [1] 4
## [1] 5
## [1] 6</code></pre>
<p>Perform math operation using for loop</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="data-wrangling.html#cb53-1"></a><span class="cf">for</span> (i <span class="cf">in</span> vector2){</span>
<span id="cb53-2"><a href="data-wrangling.html#cb53-2"></a>  m=<span class="st"> </span><span class="kw">sum</span>(vector2)<span class="op">/</span><span class="kw">length</span>(vector2)</span>
<span id="cb53-3"><a href="data-wrangling.html#cb53-3"></a>}</span>
<span id="cb53-4"><a href="data-wrangling.html#cb53-4"></a></span>
<span id="cb53-5"><a href="data-wrangling.html#cb53-5"></a>m</span></code></pre></div>
<pre><code>## [1] 5</code></pre>
<p>The <em>next</em> command can be used to modify the loop. This simple example removes even number</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="data-wrangling.html#cb55-1"></a><span class="cf">for</span>(i <span class="cf">in</span> vector2){</span>
<span id="cb55-2"><a href="data-wrangling.html#cb55-2"></a>  <span class="cf">if</span>(<span class="op">!</span>i <span class="op">%%</span><span class="st"> </span><span class="dv">2</span>){</span>
<span id="cb55-3"><a href="data-wrangling.html#cb55-3"></a>    <span class="cf">next</span></span>
<span id="cb55-4"><a href="data-wrangling.html#cb55-4"></a>  }</span>
<span id="cb55-5"><a href="data-wrangling.html#cb55-5"></a>  <span class="kw">print</span>(i)</span>
<span id="cb55-6"><a href="data-wrangling.html#cb55-6"></a>}</span></code></pre></div>
<pre><code>## [1] 5</code></pre>
<p>Perform a math operation along the columns.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="data-wrangling.html#cb57-1"></a>A=<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>)</span>
<span id="cb57-2"><a href="data-wrangling.html#cb57-2"></a>B=<span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">6</span>)</span>
<span id="cb57-3"><a href="data-wrangling.html#cb57-3"></a>df=<span class="kw">data.frame</span>(A,B)</span>
<span id="cb57-4"><a href="data-wrangling.html#cb57-4"></a></span>
<span id="cb57-5"><a href="data-wrangling.html#cb57-5"></a>output_col &lt;-<span class="st"> </span><span class="kw">vector</span>(<span class="st">&quot;double&quot;</span>, <span class="kw">ncol</span>(df))  <span class="co"># 1. output</span></span>
<span id="cb57-6"><a href="data-wrangling.html#cb57-6"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq_along</span>(df)) {            <span class="co"># 2. sequence</span></span>
<span id="cb57-7"><a href="data-wrangling.html#cb57-7"></a>  output_col[[i]] &lt;-<span class="st"> </span><span class="kw">mean</span>(df[[i]])      <span class="co"># 3. body</span></span>
<span id="cb57-8"><a href="data-wrangling.html#cb57-8"></a>}</span>
<span id="cb57-9"><a href="data-wrangling.html#cb57-9"></a>output_col</span></code></pre></div>
<pre><code>## [1] 2 5</code></pre>
<p>Perform a math operation along the rows.</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="data-wrangling.html#cb59-1"></a>output_row &lt;-<span class="st"> </span><span class="kw">vector</span>(<span class="st">&quot;double&quot;</span>, <span class="kw">nrow</span>(df))  <span class="co"># 1. output</span></span>
<span id="cb59-2"><a href="data-wrangling.html#cb59-2"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq_along</span>(df)) {            <span class="co"># 2. sequence</span></span>
<span id="cb59-3"><a href="data-wrangling.html#cb59-3"></a>  output_row[[i]] &lt;-<span class="st"> </span><span class="kw">mean</span>(df[[i]])      <span class="co"># 3. body</span></span>
<span id="cb59-4"><a href="data-wrangling.html#cb59-4"></a>}</span>
<span id="cb59-5"><a href="data-wrangling.html#cb59-5"></a>output_row</span></code></pre></div>
<pre><code>## [1] 2 5 0</code></pre>
</div>
<div id="functional" class="section level3">
<h3><span class="header-section-number">3.1.5</span> Functional</h3>
<p>A functional is a function embedded in a function. Later, we will revisit functional to perform analysis on a list of <em>nifti</em> files.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="data-wrangling.html#cb61-1"></a>Fou&lt;-<span class="kw">lapply</span>(df, <span class="cf">function</span>(A) <span class="kw">mean</span>(A)<span class="op">/</span><span class="kw">sd</span>(A))</span>
<span id="cb61-2"><a href="data-wrangling.html#cb61-2"></a></span>
<span id="cb61-3"><a href="data-wrangling.html#cb61-3"></a><span class="co">#returns a list</span></span>
<span id="cb61-4"><a href="data-wrangling.html#cb61-4"></a>Fou</span></code></pre></div>
<pre><code>## $A
## [1] 2
## 
## $B
## [1] 5</code></pre>
<p>The <em>unlist</em> function returns a matrix.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="data-wrangling.html#cb63-1"></a>Fou2&lt;-<span class="kw">unlist</span>(<span class="kw">lapply</span>(df, <span class="cf">function</span>(A) <span class="kw">mean</span>(A)<span class="op">/</span><span class="kw">sd</span>(A)))</span>
<span id="cb63-2"><a href="data-wrangling.html#cb63-2"></a></span>
<span id="cb63-3"><a href="data-wrangling.html#cb63-3"></a><span class="co">#returns a matrix</span></span>
<span id="cb63-4"><a href="data-wrangling.html#cb63-4"></a>Fou2</span></code></pre></div>
<pre><code>## A B 
## 2 5</code></pre>
<div id="mapply" class="section level4">
<h4><span class="header-section-number">3.1.5.1</span> Mapply</h4>
<p><em>Mapply</em> applies the function over elements of the data.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="data-wrangling.html#cb65-1"></a>MeanFou&lt;-<span class="kw">lapply</span>(df, mean)</span>
<span id="cb65-2"><a href="data-wrangling.html#cb65-2"></a></span>
<span id="cb65-3"><a href="data-wrangling.html#cb65-3"></a>MeanFou[]&lt;-<span class="kw">Map</span>(<span class="st">&#39;/&#39;</span>, df, MeanFou)</span></code></pre></div>
<p>The equivalent code with <em>lapply</em> is provided below.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="data-wrangling.html#cb66-1"></a>MeanFou2&lt;-<span class="kw">lapply</span>(df, <span class="cf">function</span>(M) M<span class="op">/</span><span class="kw">mean</span>(M))</span>
<span id="cb66-2"><a href="data-wrangling.html#cb66-2"></a></span>
<span id="cb66-3"><a href="data-wrangling.html#cb66-3"></a>MeanFou2</span></code></pre></div>
<pre><code>## $A
## [1] 0.5 1.0 1.5
## 
## $B
## [1] 0.8 1.0 1.2</code></pre>
</div>
</div>
</div>
<div id="data-storage" class="section level2">
<h2><span class="header-section-number">3.2</span> Data storage</h2>
<p>Often one assumes that opening Rstudio is sufficient to locate the file and run the analysis. One way of doing this at the console is to click on <em>Session</em> tab, then <em>Set Working Directory</em> to location of file. Another way of doing this seemlessly is to use the library <em>here</em>. It is easy to find the files in your directory using the <em>list.files()</em> call.</p>
<p>To list all files in a directory then</p>
<p>To list files matching a pattern</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="data-wrangling.html#cb68-1"></a><span class="co">#list files matching pattern</span></span>
<span id="cb68-2"><a href="data-wrangling.html#cb68-2"></a><span class="kw">list.files</span>(<span class="dt">pattern=</span><span class="st">&quot;.Rmd&quot;</span>)</span></code></pre></div>
<pre><code>##  [1] &quot;01-intro.Rmd&quot;                        &quot;02-Data-Wrangling.Rmd&quot;              
##  [3] &quot;03-Statistics.Rmd&quot;                   &quot;04-multivariate-analysis.Rmd&quot;       
##  [5] &quot;05-machinelearning.Rmd&quot;              &quot;06-machinelearningpt2.Rmd&quot;          
##  [7] &quot;07-Bayesian-analysis.Rmd&quot;            &quot;08-operational-research.Rmd&quot;        
##  [9] &quot;09-graph-theory.Rmd&quot;                 &quot;10-geospatial-analysis.Rmd&quot;         
## [11] &quot;11-App.Rmd&quot;                          &quot;12-Appendix.Rmd&quot;                    
## [13] &quot;13-references.Rmd&quot;                   &quot;Applications-of-R-in-Healthcare.Rmd&quot;
## [15] &quot;index.Rmd&quot;</code></pre>
<div id="data-frame" class="section level3">
<h3><span class="header-section-number">3.2.1</span> Data frame</h3>
<p>Data frame is a convenient way of formatting data in table format. It is worth checking the structure of data. Some libraries prefer to work with data in data frame while others prefer matrix or array structure.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="data-wrangling.html#cb70-1"></a>a&lt;-<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>)</span>
<span id="cb70-2"><a href="data-wrangling.html#cb70-2"></a>b&lt;-<span class="kw">c</span>(<span class="st">&quot;apple&quot;</span>,<span class="st">&quot;orange&quot;</span>,<span class="st">&quot;banana&quot;</span>)</span>
<span id="cb70-3"><a href="data-wrangling.html#cb70-3"></a>e&lt;-<span class="kw">data.frame</span>(a,b)</span>
<span id="cb70-4"><a href="data-wrangling.html#cb70-4"></a><span class="kw">rownames</span>(e)</span></code></pre></div>
<pre><code>## [1] &quot;1&quot; &quot;2&quot; &quot;3&quot;</code></pre>
</div>
<div id="excel-data" class="section level3">
<h3><span class="header-section-number">3.2.2</span> Excel data</h3>
<p>Excel data are stored as csv, xls and xlsx. Csv files can be open in base R
using <em>read.csv</em> function or using <em>readr</em> library and <em>read_csv function</em>. I would urge you to get used to manipulating data in R as the codes serve as a way to keep track with the change in data. The original xcel data should not be touched outside of R. A problem with excel data is that its autocorrect function change provenance of genomic data eg SEPT1, MARCH1. SEPT1 is now relabeled as SEPTIN1.</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="data-wrangling.html#cb72-1"></a>A&lt;-<span class="kw">read.csv</span>(<span class="st">&quot;File.csv&quot;</span>)</span>
<span id="cb72-2"><a href="data-wrangling.html#cb72-2"></a></span>
<span id="cb72-3"><a href="data-wrangling.html#cb72-3"></a>B&lt;-readr<span class="op">::</span><span class="kw">read_csv</span> (<span class="st">&quot;File.csv&quot;</span>)</span></code></pre></div>
<p>The <em>readxl</em> library can be used to open files ending in xls and xlsx.</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="data-wrangling.html#cb73-1"></a>C&lt;-readxl<span class="op">::</span><span class="kw">read_xlsx</span>(<span class="st">&quot;File.xlsx&quot;</span>,<span class="dt">skip=</span><span class="dv">1</span>) <span class="co">#skip first row </span></span>
<span id="cb73-2"><a href="data-wrangling.html#cb73-2"></a></span>
<span id="cb73-3"><a href="data-wrangling.html#cb73-3"></a>D&lt;-readxl<span class="op">::</span><span class="kw">read_xlsx</span>(<span class="st">&quot;File.xlsx&quot;</span>,<span class="dt">sheet=</span><span class="dv">2</span>) <span class="co">#read data from sheet 2</span></span></code></pre></div>
<div id="date-and-time" class="section level4">
<h4><span class="header-section-number">3.2.2.1</span> Date and time</h4>
<p>Date and time can be handle in base R. The library <em>lubridate</em> is useful for parsing date data. It is possible to get an overview of the functions of the library by typing <em>help(package=lubridate)</em>. Errors with parsing can occur if there are characters in the column containing date data.</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="data-wrangling.html#cb74-1"></a><span class="kw">library</span>(dplyr)</span></code></pre></div>
<pre><code>## Warning: package &#39;dplyr&#39; was built under R version 4.0.5</code></pre>
<pre><code>## 
## Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:stats&#39;:
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="data-wrangling.html#cb79-1"></a>dfdate&lt;-<span class="kw">data.frame</span>(<span class="st">&quot;DateofEvent&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;12/03/2005&quot;</span>,<span class="st">&quot;12/04/2006&quot;</span>,<span class="ot">NA</span>),</span>
<span id="cb79-2"><a href="data-wrangling.html#cb79-2"></a>                   <span class="st">&quot;Result&quot;</span>=<span class="kw">c</span>(<span class="dv">4</span>,<span class="dv">5</span>,<span class="dv">6</span>))</span>
<span id="cb79-3"><a href="data-wrangling.html#cb79-3"></a><span class="kw">class</span>(dfdate<span class="op">$</span>DateofEvent)</span></code></pre></div>
<pre><code>## [1] &quot;character&quot;</code></pre>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="data-wrangling.html#cb81-1"></a>dfdate<span class="op">$</span>DateofEvent</span></code></pre></div>
<pre><code>## [1] &quot;12/03/2005&quot; &quot;12/04/2006&quot; NA</code></pre>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="data-wrangling.html#cb83-1"></a><span class="co">#remove NA using filter</span></span>
<span id="cb83-2"><a href="data-wrangling.html#cb83-2"></a>dfdate <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(DateofEvent))</span></code></pre></div>
<pre><code>##   DateofEvent Result
## 1  12/03/2005      4
## 2  12/04/2006      5</code></pre>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="data-wrangling.html#cb85-1"></a><span class="co">#re assigning date data type</span></span>
<span id="cb85-2"><a href="data-wrangling.html#cb85-2"></a>dfdate<span class="op">$</span>DateofEvent2&lt;-<span class="kw">as.POSIXct</span>(dfdate<span class="op">$</span>DateofEvent)</span>
<span id="cb85-3"><a href="data-wrangling.html#cb85-3"></a><span class="kw">class</span>(dfdate<span class="op">$</span>DateofEvent2)</span></code></pre></div>
<pre><code>## [1] &quot;POSIXct&quot; &quot;POSIXt&quot;</code></pre>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="data-wrangling.html#cb87-1"></a>dfdate<span class="op">$</span>DateofEvent2</span></code></pre></div>
<pre><code>## [1] &quot;0012-03-20 LMT&quot; &quot;0012-04-20 LMT&quot; NA</code></pre>
<p>Problem can occur when date and time are located in separate columns. The first issue is that time data is assigned a default date 1899-12-30. This error occurs as the base date in MS Office is 1899-12-30. This issue can be compounded when there are 2 time data eg a patient has stroke onset at 22:00 and is scanned at 02:00. The data become 1899-12-31 22:00 and 1899-12-31 02:00. In this case it implies that scanning occurs before stroke onset. This could have been solved at the data collection stage by having 2 separate date colums. There are several solutions inclusing <em>ifelse</em> but care must be taken with this argument. Note
that <em>ifelse</em> argument convert date time data to numeric class. This can be resolved by embedding <em>ifelse</em> statement within <em>as.Date</em> argument. This argument requires <em>origin</em> argument. The logic argument may fail when the default date differ eg 1904-02-08 02:00. In this case 1904-02-08 02:00 is greater than 1899-12-31 22:00.</p>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="data-wrangling.html#cb89-1"></a><span class="kw">library</span>(lubridate)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;lubridate&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:base&#39;:
## 
##     date, intersect, setdiff, union</code></pre>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="data-wrangling.html#cb92-1"></a><span class="kw">library</span>(tidyverse)</span></code></pre></div>
<pre><code>## -- Attaching packages --------------------------------------- tidyverse 1.3.0 --</code></pre>
<pre><code>## v ggplot2 3.3.5     v purrr   0.3.4
## v tibble  3.1.6     v stringr 1.4.0
## v tidyr   1.1.4     v forcats 0.5.0
## v readr   1.4.0</code></pre>
<pre><code>## Warning: package &#39;ggplot2&#39; was built under R version 4.0.5</code></pre>
<pre><code>## Warning: package &#39;tibble&#39; was built under R version 4.0.5</code></pre>
<pre><code>## Warning: package &#39;tidyr&#39; was built under R version 4.0.5</code></pre>
<pre><code>## -- Conflicts ------------------------------------------ tidyverse_conflicts() --
## x lubridate::as.difftime() masks base::as.difftime()
## x lubridate::date()        masks base::date()
## x dplyr::filter()          masks stats::filter()
## x lubridate::intersect()   masks base::intersect()
## x dplyr::lag()             masks stats::lag()
## x lubridate::setdiff()     masks base::setdiff()
## x lubridate::union()       masks base::union()</code></pre>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="data-wrangling.html#cb99-1"></a>df&lt;-<span class="kw">data.frame</span>(<span class="st">&quot;Onset_date&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;2014-03-06&quot;</span>,<span class="st">&quot;2013-06-09&quot;</span>), <span class="st">&quot;Onset_Time&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;1899-12-31 08:03:00&quot;</span>,<span class="st">&quot;1899-12-31 22:00:00&quot;</span>), <span class="st">&quot;Scan_Time&quot;</span>=<span class="kw">c</span>(<span class="st">&quot;1904-02-08 10:00:00&quot;</span>,<span class="st">&quot;1899-12-31 02:00:00&quot;</span>)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb99-2"><a href="data-wrangling.html#cb99-2"></a><span class="st">  </span><span class="kw">mutate</span>(</span>
<span id="cb99-3"><a href="data-wrangling.html#cb99-3"></a>  <span class="co">#use update argument to reassign year, month and day</span></span>
<span id="cb99-4"><a href="data-wrangling.html#cb99-4"></a>  <span class="dt">Scantime=</span><span class="kw">update</span>(<span class="kw">ymd_hms</span>(Scan_Time), <span class="dt">year=</span><span class="kw">year</span>(<span class="kw">ymd</span>(Onset_date)), <span class="dt">month=</span><span class="kw">month</span>(<span class="kw">ymd</span>(Onset_date)), <span class="dt">mday=</span><span class="kw">day</span>(<span class="kw">ymd</span>(Onset_date))),</span>
<span id="cb99-5"><a href="data-wrangling.html#cb99-5"></a> </span>
<span id="cb99-6"><a href="data-wrangling.html#cb99-6"></a>  <span class="dt">Onsettime=</span><span class="kw">update</span>(<span class="kw">ymd_hms</span>(Onset_Time), <span class="dt">year=</span><span class="kw">year</span>(<span class="kw">ymd</span>(Onset_date)), <span class="dt">month=</span><span class="kw">month</span>(<span class="kw">ymd</span>(Onset_date)), <span class="dt">mday=</span><span class="kw">day</span>(<span class="kw">ymd</span>(Onset_date))),</span>
<span id="cb99-7"><a href="data-wrangling.html#cb99-7"></a>  </span>
<span id="cb99-8"><a href="data-wrangling.html#cb99-8"></a>  <span class="dt">Scan_date=</span><span class="kw">ifelse</span>(Onsettime<span class="op">&gt;</span>Scantime,<span class="dv">1</span>,<span class="dv">0</span>),</span>
<span id="cb99-9"><a href="data-wrangling.html#cb99-9"></a>  <span class="dt">Scantime=</span>Scantime<span class="op">+</span>Scan_date<span class="op">*</span><span class="kw">hms</span>(<span class="st">&quot;24:00:00&quot;</span>),</span>
<span id="cb99-10"><a href="data-wrangling.html#cb99-10"></a>  </span>
<span id="cb99-11"><a href="data-wrangling.html#cb99-11"></a> <span class="dt">DiffHour=</span>Scantime<span class="op">-</span>Onsettime <span class="co">#minutes</span></span>
<span id="cb99-12"><a href="data-wrangling.html#cb99-12"></a>)</span>
<span id="cb99-13"><a href="data-wrangling.html#cb99-13"></a></span>
<span id="cb99-14"><a href="data-wrangling.html#cb99-14"></a>df <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="op">-</span>Scan_date) <span class="op">%&gt;%</span><span class="st"> </span>gt<span class="op">::</span><span class="kw">gt</span>()</span></code></pre></div>
<div id="orpwgnixds" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#orpwgnixds .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#orpwgnixds .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#orpwgnixds .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#orpwgnixds .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#orpwgnixds .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#orpwgnixds .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#orpwgnixds .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#orpwgnixds .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#orpwgnixds .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#orpwgnixds .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#orpwgnixds .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#orpwgnixds .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#orpwgnixds .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#orpwgnixds .gt_from_md > :first-child {
  margin-top: 0;
}

#orpwgnixds .gt_from_md > :last-child {
  margin-bottom: 0;
}

#orpwgnixds .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#orpwgnixds .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#orpwgnixds .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#orpwgnixds .gt_row_group_first td {
  border-top-width: 2px;
}

#orpwgnixds .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#orpwgnixds .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#orpwgnixds .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#orpwgnixds .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#orpwgnixds .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#orpwgnixds .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#orpwgnixds .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#orpwgnixds .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#orpwgnixds .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#orpwgnixds .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-left: 4px;
  padding-right: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#orpwgnixds .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#orpwgnixds .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#orpwgnixds .gt_left {
  text-align: left;
}

#orpwgnixds .gt_center {
  text-align: center;
}

#orpwgnixds .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#orpwgnixds .gt_font_normal {
  font-weight: normal;
}

#orpwgnixds .gt_font_bold {
  font-weight: bold;
}

#orpwgnixds .gt_font_italic {
  font-style: italic;
}

#orpwgnixds .gt_super {
  font-size: 65%;
}

#orpwgnixds .gt_footnote_marks {
  font-style: italic;
  font-weight: normal;
  font-size: 75%;
  vertical-align: 0.4em;
}

#orpwgnixds .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#orpwgnixds .gt_slash_mark {
  font-size: 0.7em;
  line-height: 0.7em;
  vertical-align: 0.15em;
}

#orpwgnixds .gt_fraction_numerator {
  font-size: 0.6em;
  line-height: 0.6em;
  vertical-align: 0.45em;
}

#orpwgnixds .gt_fraction_denominator {
  font-size: 0.6em;
  line-height: 0.6em;
  vertical-align: -0.05em;
}
</style>
<table class="gt_table">
  
  <thead class="gt_col_headings">
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">Onset_date</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">Onset_Time</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">Scan_Time</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">Scantime</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">Onsettime</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1">DiffHour</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td class="gt_row gt_left">2014-03-06</td>
<td class="gt_row gt_left">1899-12-31 08:03:00</td>
<td class="gt_row gt_left">1904-02-08 10:00:00</td>
<td class="gt_row gt_left">2014-03-06 10:00:00</td>
<td class="gt_row gt_left">2014-03-06 08:03:00</td>
<td class="gt_row gt_center">1.95</td></tr>
    <tr><td class="gt_row gt_left">2013-06-09</td>
<td class="gt_row gt_left">1899-12-31 22:00:00</td>
<td class="gt_row gt_left">1899-12-31 02:00:00</td>
<td class="gt_row gt_left">2013-06-10 02:00:00</td>
<td class="gt_row gt_left">2013-06-09 22:00:00</td>
<td class="gt_row gt_center">4</td></tr>
  </tbody>
  
  
</table>
</div>
<p>One way of storing data in R format is to save the file as .Rda. This format
will ensure that no one can accidentally rewrite or delete a number. For very large data, it’s quicker to save as .Rda file than as csv file.</p>
</div>
</div>
<div id="foreign-data" class="section level3">
<h3><span class="header-section-number">3.2.3</span> Foreign data</h3>
<p>The foreign library is traditionally use to handle data from <em>SPSS</em> (.sav), <em>Stata</em> (.dta) and <em>SAS</em> (.sas). One should look at the ending in the file to determine the necessary library.</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="data-wrangling.html#cb100-1"></a><span class="kw">library</span>(foreign)</span>
<span id="cb100-2"><a href="data-wrangling.html#cb100-2"></a><span class="co">#write and read Stata</span></span>
<span id="cb100-3"><a href="data-wrangling.html#cb100-3"></a><span class="kw">write.dta</span>(dfdate,<span class="dt">file=</span><span class="st">&quot;./Data-Use/dfdate_temp.dta&quot;</span>)</span>
<span id="cb100-4"><a href="data-wrangling.html#cb100-4"></a>a&lt;-<span class="kw">read.dta</span>(<span class="st">&quot;./Data-Use/dfdate_temp.dta&quot;</span>)</span>
<span id="cb100-5"><a href="data-wrangling.html#cb100-5"></a>a</span></code></pre></div>
<p>The <em>foreign</em> library can handle older <em>SAS</em> files but not the current version. The current version of <em>SAS</em> file requires <em>sas7bdat</em>. The clue is the file ends in .sas7bdat.</p>
</div>
<div id="json-format" class="section level3">
<h3><span class="header-section-number">3.2.4</span> json format</h3>
<p>Json is short for JavaScript object Notification. These files have a
hierarchical structured format. The json file is in text format amd can also be examined using <em>Notepad</em>. These files can be read using the <em>RJSONIO</em> or <em>rjson</em>
libraries in R.</p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="data-wrangling.html#cb101-1"></a><span class="kw">library</span>(RJSONIO)</span>
<span id="cb101-2"><a href="data-wrangling.html#cb101-2"></a>j&lt;-<span class="kw">fromJSON</span>(<span class="st">&quot;./Data-Use/0411.geojson&quot;</span>)</span>
<span id="cb101-3"><a href="data-wrangling.html#cb101-3"></a>j&lt;-<span class="kw">lapply</span>(j, <span class="cf">function</span>(x) {</span>
<span id="cb101-4"><a href="data-wrangling.html#cb101-4"></a>  x[<span class="kw">sapply</span>(x,is.null)]&lt;-<span class="ot">NA</span></span>
<span id="cb101-5"><a href="data-wrangling.html#cb101-5"></a>  <span class="kw">unlist</span>(x)</span>
<span id="cb101-6"><a href="data-wrangling.html#cb101-6"></a>})</span>
<span id="cb101-7"><a href="data-wrangling.html#cb101-7"></a>k&lt;-<span class="kw">as.data.frame</span>(<span class="kw">do.call</span>(<span class="st">&quot;cbind&quot;</span>,j)) <span class="co">#list to data frame</span></span></code></pre></div>
</div>
</div>
<div id="tidy-data" class="section level2">
<h2><span class="header-section-number">3.3</span> Tidy data</h2>
<p>Attention to collection of data is important as it shows the way for performing analysis. In general each row represents on variable and each column represents
an attribute of that variables. Sometimes there is a temptation to embed 2 types of attributes into a column.</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="data-wrangling.html#cb102-1"></a>df2&lt;-<span class="kw">data.frame</span>(<span class="dt">Sex=</span><span class="kw">c</span>(<span class="st">&quot;Male&quot;</span>,<span class="st">&quot;Female&quot;</span>), <span class="dt">Test=</span><span class="kw">c</span>(<span class="st">&quot;positive 5 negative 5&quot;</span>,</span>
<span id="cb102-2"><a href="data-wrangling.html#cb102-2"></a>                        <span class="st">&quot; negative 0 negative 10&quot;</span>))</span>
<span id="cb102-3"><a href="data-wrangling.html#cb102-3"></a>df2</span></code></pre></div>
<pre><code>##      Sex                    Test
## 1   Male   positive 5 negative 5
## 2 Female  negative 0 negative 10</code></pre>
<p>The above example should be entered this way. This change allows one to group variables by Test status: ‘positive’ or ‘negative’. One can easily perform a
t-test here (not recommend in this case as the data contains only 2 rows).</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="data-wrangling.html#cb104-1"></a>df2&lt;-<span class="kw">data.frame</span>(<span class="dt">Sex=</span><span class="kw">c</span>(<span class="st">&quot;Male&quot;</span>,<span class="st">&quot;Female&quot;</span>), <span class="st">`</span><span class="dt">Test Positive</span><span class="st">`</span> =<span class="kw">c</span>(<span class="dv">5</span>,<span class="dv">0</span>), <span class="st">`</span><span class="dt">Test Negative</span><span class="st">`</span>=<span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">10</span>))</span>
<span id="cb104-2"><a href="data-wrangling.html#cb104-2"></a>df2</span></code></pre></div>
<pre><code>##      Sex Test.Positive Test.Negative
## 1   Male             5             5
## 2 Female             0            10</code></pre>
<p>The below example is illustrate how to collapse columns when using base R.</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="data-wrangling.html#cb106-1"></a>dfa&lt;-<span class="kw">data.frame</span>(<span class="dt">City=</span><span class="kw">c</span>(<span class="st">&quot;Melbourne&quot;</span>,<span class="st">&quot;Sydney&quot;</span>,<span class="st">&quot;Adelaide&quot;</span>),</span>
<span id="cb106-2"><a href="data-wrangling.html#cb106-2"></a>                <span class="dt">State=</span><span class="kw">c</span>(<span class="st">&quot;Victoria&quot;</span>,<span class="st">&quot;NSW&quot;</span>,<span class="st">&quot;South Australia&quot;</span>))</span>
<span id="cb106-3"><a href="data-wrangling.html#cb106-3"></a><span class="co">#collapsing City and State columns and generate new column address</span></span>
<span id="cb106-4"><a href="data-wrangling.html#cb106-4"></a>dfa<span class="op">$</span>addresses&lt;-<span class="kw">paste0</span>(dfa<span class="op">$</span>City,<span class="st">&quot;,&quot;</span>, dfa<span class="op">$</span>State) <span class="co">#separate by comma</span></span>
<span id="cb106-5"><a href="data-wrangling.html#cb106-5"></a>dfa<span class="op">$</span>addresses2&lt;-<span class="kw">paste0</span>(dfa<span class="op">$</span>City,<span class="st">&quot;,&quot;</span>, dfa<span class="op">$</span>State,<span class="st">&quot;, Australia&quot;</span>)</span>
<span id="cb106-6"><a href="data-wrangling.html#cb106-6"></a>dfa</span></code></pre></div>
<pre><code>##        City           State                addresses
## 1 Melbourne        Victoria       Melbourne,Victoria
## 2    Sydney             NSW               Sydney,NSW
## 3  Adelaide South Australia Adelaide,South Australia
##                            addresses2
## 1       Melbourne,Victoria, Australia
## 2               Sydney,NSW, Australia
## 3 Adelaide,South Australia, Australia</code></pre>
<p>This example is same as above but uses verbs from <em>tidyr</em>. This is useful for collapsing address for geocoding.</p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="data-wrangling.html#cb108-1"></a><span class="kw">library</span>(tidyr)</span>
<span id="cb108-2"><a href="data-wrangling.html#cb108-2"></a>dfa1&lt;-dfa <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">unite</span> (<span class="st">&quot;new_address&quot;</span>,City<span class="op">:</span>State,<span class="dt">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb108-3"><a href="data-wrangling.html#cb108-3"></a>dfa1</span></code></pre></div>
<pre><code>##                new_address                addresses
## 1       Melbourne,Victoria       Melbourne,Victoria
## 2               Sydney,NSW               Sydney,NSW
## 3 Adelaide,South Australia Adelaide,South Australia
##                            addresses2
## 1       Melbourne,Victoria, Australia
## 2               Sydney,NSW, Australia
## 3 Adelaide,South Australia, Australia</code></pre>
<p>Using the data above, let’s split the column address</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="data-wrangling.html#cb110-1"></a><span class="kw">library</span>(tidyr)</span>
<span id="cb110-2"><a href="data-wrangling.html#cb110-2"></a>dfa2&lt;-dfa1 <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">separate</span>(addresses, <span class="kw">c</span>(<span class="st">&quot;City2&quot;</span>, <span class="st">&quot;State2&quot;</span>))</span></code></pre></div>
<pre><code>## Warning: Expected 2 pieces. Additional pieces discarded in 1 rows [3].</code></pre>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="data-wrangling.html#cb112-1"></a>dfa2</span></code></pre></div>
<pre><code>##                new_address     City2   State2
## 1       Melbourne,Victoria Melbourne Victoria
## 2               Sydney,NSW    Sydney      NSW
## 3 Adelaide,South Australia  Adelaide    South
##                            addresses2
## 1       Melbourne,Victoria, Australia
## 2               Sydney,NSW, Australia
## 3 Adelaide,South Australia, Australia</code></pre>
<div id="factors" class="section level3">
<h3><span class="header-section-number">3.3.1</span> Factors</h3>
<p>There are several types of factors in R: ordered and not ordered. It is
important to pay attention to how factors are coded. Sometimes, male is represented as 1 and female as 0. Sometimes, female is represented as 2. This discussion may seems trivial but a paper had been retracted in a high impact factor journal Jama because of miscoding of the trial assignment 1 and 2 rather than the assignment of 0 and 1. This error led to reversing the results with logistic regression <span class="citation">(Aboumatar and Wise <a href="#ref-pmid31593277" role="doc-biblioref">2019</a>)</span>. This error led to report that an outpatient management program for chronic obstructive pulmonary disease resulted in fewer admissions. Below is an example which can occur when data is transformed into factor and back to number. Note that the coding goes from 0 and 1 to 2 and 1.</p>
<p>In certain analyses, the libraries prefer to use the dependent or outcome variable as binary coding in numeric format eg logistic regression and random forest. The library <em>e1071</em> for performing support vector machine prefers the outcome variable as factor.</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="data-wrangling.html#cb114-1"></a><span class="kw">library</span>(Stat2Data)</span>
<span id="cb114-2"><a href="data-wrangling.html#cb114-2"></a><span class="kw">data</span>(<span class="st">&quot;Leukemia&quot;</span>) <span class="co">#treatment of leukemia</span></span>
<span id="cb114-3"><a href="data-wrangling.html#cb114-3"></a></span>
<span id="cb114-4"><a href="data-wrangling.html#cb114-4"></a>Leukemia <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">glimpse</span>()</span></code></pre></div>
<pre><code>## Rows: 51
## Columns: 9
## $ Age    &lt;int&gt; 20, 25, 26, 26, 27, 27, 28, 28, 31, 33, 33, 33, 34, 36, 37, 40,~
## $ Smear  &lt;int&gt; 78, 64, 61, 64, 95, 80, 88, 70, 72, 58, 92, 42, 26, 55, 71, 91,~
## $ Infil  &lt;int&gt; 39, 61, 55, 64, 95, 64, 88, 70, 72, 58, 92, 38, 26, 55, 71, 91,~
## $ Index  &lt;int&gt; 7, 16, 12, 16, 6, 8, 20, 14, 5, 7, 5, 12, 7, 14, 15, 9, 12, 4, ~
## $ Blasts &lt;dbl&gt; 0.6, 35.0, 7.5, 21.0, 7.5, 0.6, 4.8, 10.0, 2.3, 5.7, 2.6, 2.5, ~
## $ Temp   &lt;int&gt; 990, 1030, 982, 1000, 980, 1010, 986, 1010, 988, 986, 980, 984,~
## $ Resp   &lt;int&gt; 1, 1, 1, 1, 1, 0, 1, 1, 1, 0, 1, 1, 0, 1, 0, 1, 1, 0, 1, 1, 0, ~
## $ Time   &lt;int&gt; 18, 31, 31, 31, 36, 1, 9, 39, 20, 4, 45, 36, 12, 8, 1, 15, 24, ~
## $ Status &lt;int&gt; 0, 1, 0, 0, 0, 0, 0, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, ~</code></pre>
<p>The variable Resp is now a factor with levels 0 and 1</p>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="data-wrangling.html#cb116-1"></a>Leukemia<span class="op">$</span>Resp</span></code></pre></div>
<pre><code>##  [1] 1 1 1 1 1 0 1 1 1 0 1 1 0 1 0 1 1 0 1 1 0 0 0 0 1 1 0 0 1 0 1 0 0 0 0 0 0 0
## [39] 0 1 0 0 0 1 0 0 1 0 1 1 0</code></pre>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="data-wrangling.html#cb118-1"></a>Leukemia<span class="op">$</span>Response.factor&lt;-<span class="kw">as.factor</span>(Leukemia<span class="op">$</span>Resp)</span>
<span id="cb118-2"><a href="data-wrangling.html#cb118-2"></a><span class="kw">head</span>(Leukemia<span class="op">$</span>Response.factor)</span></code></pre></div>
<pre><code>## [1] 1 1 1 1 1 0
## Levels: 0 1</code></pre>
<p>Note in the conversion back to numeric ‘dummy’ values, the data takes the form 1 and 2. This has changed the dummy values of 0 and 1. It is important to examine the data before running analysis.</p>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="data-wrangling.html#cb120-1"></a>Leukemia<span class="op">$</span>Response.numeric&lt;-<span class="kw">as.numeric</span>(Leukemia<span class="op">$</span>Response.factor)</span>
<span id="cb120-2"><a href="data-wrangling.html#cb120-2"></a>Leukemia<span class="op">$</span>Response.numeric</span></code></pre></div>
<pre><code>##  [1] 2 2 2 2 2 1 2 2 2 1 2 2 1 2 1 2 2 1 2 2 1 1 1 1 2 2 1 1 2 1 2 1 1 1 1 1 1 1
## [39] 1 2 1 1 1 2 1 1 2 1 2 2 1</code></pre>
<p>For variables which are characters but considered as factors, it is necessary to convert to class character before converting to dummy values.</p>
<div class="sourceCode" id="cb122"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb122-1"><a href="data-wrangling.html#cb122-1"></a><span class="kw">data</span>(<span class="st">&quot;BreastCancer&quot;</span>, <span class="dt">package=</span><span class="st">&quot;mlbench&quot;</span>)</span>
<span id="cb122-2"><a href="data-wrangling.html#cb122-2"></a>BreastCancer <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">glimpse</span>()</span></code></pre></div>
<pre><code>## Rows: 699
## Columns: 11
## $ Id              &lt;chr&gt; &quot;1000025&quot;, &quot;1002945&quot;, &quot;1015425&quot;, &quot;1016277&quot;, &quot;1017023&quot;,~
## $ Cl.thickness    &lt;ord&gt; 5, 5, 3, 6, 4, 8, 1, 2, 2, 4, 1, 2, 5, 1, 8, 7, 4, 4, ~
## $ Cell.size       &lt;ord&gt; 1, 4, 1, 8, 1, 10, 1, 1, 1, 2, 1, 1, 3, 1, 7, 4, 1, 1,~
## $ Cell.shape      &lt;ord&gt; 1, 4, 1, 8, 1, 10, 1, 2, 1, 1, 1, 1, 3, 1, 5, 6, 1, 1,~
## $ Marg.adhesion   &lt;ord&gt; 1, 5, 1, 1, 3, 8, 1, 1, 1, 1, 1, 1, 3, 1, 10, 4, 1, 1,~
## $ Epith.c.size    &lt;ord&gt; 2, 7, 2, 3, 2, 7, 2, 2, 2, 2, 1, 2, 2, 2, 7, 6, 2, 2, ~
## $ Bare.nuclei     &lt;fct&gt; 1, 10, 2, 4, 1, 10, 10, 1, 1, 1, 1, 1, 3, 3, 9, 1, 1, ~
## $ Bl.cromatin     &lt;fct&gt; 3, 3, 3, 3, 3, 9, 3, 3, 1, 2, 3, 2, 4, 3, 5, 4, 2, 3, ~
## $ Normal.nucleoli &lt;fct&gt; 1, 2, 1, 7, 1, 7, 1, 1, 1, 1, 1, 1, 4, 1, 5, 3, 1, 1, ~
## $ Mitoses         &lt;fct&gt; 1, 1, 1, 1, 1, 1, 1, 1, 5, 1, 1, 1, 1, 1, 4, 1, 1, 1, ~
## $ Class           &lt;fct&gt; benign, benign, benign, benign, benign, malignant, ben~</code></pre>
<p>The steps for conversion are illustrated below. Conversion of multiple columns
of factors and ordered factors can be done in one step using <em>lapply</em> function. This will be described much further below.</p>
<div class="sourceCode" id="cb124"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb124-1"><a href="data-wrangling.html#cb124-1"></a>BreastCancer<span class="op">$</span>Class&lt;-<span class="kw">as.character</span>(BreastCancer<span class="op">$</span>Class)</span>
<span id="cb124-2"><a href="data-wrangling.html#cb124-2"></a>BreastCancer<span class="op">$</span>Class[BreastCancer<span class="op">$</span>Class<span class="op">==</span><span class="st">&quot;benign&quot;</span>]&lt;-<span class="dv">0</span></span>
<span id="cb124-3"><a href="data-wrangling.html#cb124-3"></a>BreastCancer<span class="op">$</span>Class[BreastCancer<span class="op">$</span>Class<span class="op">==</span><span class="st">&quot;malignant&quot;</span>]&lt;-<span class="dv">1</span></span>
<span id="cb124-4"><a href="data-wrangling.html#cb124-4"></a>BreastCancer<span class="op">$</span>Class&lt;-<span class="kw">as.numeric</span>(BreastCancer<span class="op">$</span>Class)</span>
<span id="cb124-5"><a href="data-wrangling.html#cb124-5"></a><span class="kw">head</span>(BreastCancer<span class="op">$</span>Class)</span></code></pre></div>
<pre><code>## [1] 0 0 0 0 0 1</code></pre>
<p>This illustration describes conversion of a continuous variable into orderly factors.</p>
<div class="sourceCode" id="cb126"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb126-1"><a href="data-wrangling.html#cb126-1"></a><span class="kw">library</span>(Stat2Data)</span>
<span id="cb126-2"><a href="data-wrangling.html#cb126-2"></a><span class="kw">data</span>(<span class="st">&quot;Leukemia&quot;</span>) <span class="co">#treatment of leukemia</span></span>
<span id="cb126-3"><a href="data-wrangling.html#cb126-3"></a><span class="co">#partition Age into 8 ordered factors</span></span>
<span id="cb126-4"><a href="data-wrangling.html#cb126-4"></a>Leukemia<span class="op">$</span>AgeCat&lt;-ggplot2<span class="op">::</span><span class="kw">cut_interval</span>(Leukemia<span class="op">$</span>Age, <span class="dt">n=</span><span class="dv">8</span>, <span class="dt">ordered_result=</span><span class="ot">TRUE</span>)</span>
<span id="cb126-5"><a href="data-wrangling.html#cb126-5"></a><span class="kw">class</span>(Leukemia<span class="op">$</span>AgeCat)</span></code></pre></div>
<pre><code>## [1] &quot;ordered&quot; &quot;factor&quot;</code></pre>
</div>
<div id="multiple-files" class="section level3">
<h3><span class="header-section-number">3.3.2</span> Multiple files</h3>
<p>Merging of files can be done using <em>dplyr</em> to perform <em>inner_join</em>,
<em>outer_join</em>, <em>left_join</em> and <em>right_join</em>. Note that this can also be done in base R or using syntax of <em>data.table</em>. These files can be joined using <em>%&gt;%</em> operator.</p>
<div class="sourceCode" id="cb128"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb128-1"><a href="data-wrangling.html#cb128-1"></a>DF1&lt;-<span class="kw">data.frame</span>(<span class="dt">ID=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>),<span class="dt">Age=</span><span class="kw">c</span>(<span class="dv">20</span>,<span class="dv">30</span>,<span class="dv">40</span>))</span>
<span id="cb128-2"><a href="data-wrangling.html#cb128-2"></a>DF2&lt;-<span class="kw">data.frame</span>(<span class="dt">ID=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>),<span class="dt">Sex=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>))</span>
<span id="cb128-3"><a href="data-wrangling.html#cb128-3"></a>DF3&lt;-<span class="kw">data.frame</span>(<span class="dt">ID=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), <span class="dt">Diabetes=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb128-4"><a href="data-wrangling.html#cb128-4"></a></span>
<span id="cb128-5"><a href="data-wrangling.html#cb128-5"></a>DF &lt;-DF1 <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(DF2, <span class="dt">by=</span><span class="st">&quot;ID&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">left_join</span>(DF3, <span class="dt">by=</span><span class="st">&quot;ID&quot;</span>)</span></code></pre></div>
</div>
<div id="pivot" class="section level3">
<h3><span class="header-section-number">3.3.3</span> Pivot</h3>
<p>A variety of different expressions are used to describe data format such as wide and long formats. In some case the distinction between such formats is not
clear. The verbs for performing these operations are <em>pivot_wide</em>, <em>pivot_long</em>. Again <em>data.table</em> uses different verbs such as <em>cast</em> and <em>melt</em>. In general, most regression analyses are performed with data in wide format. In this case
each row represents a unique ID. Longitudinal analyses are performed with data
in long format. In this format, there are several rows with the same ID. In the next Chapter on Statistics, an example of data generated in wide format and coverted to long format using <em>plyr</em>. Here we will demonstrate the use of
<em>tidyr</em> to pivot loner or wider.</p>
<p>The best way to think about how data should be presented is that data is
analyzed according to columns not rows. The data below is extracted from CDC
COVID website. Details are given below under Web scraping on how this task was performed.</p>
<div class="sourceCode" id="cb129"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb129-1"><a href="data-wrangling.html#cb129-1"></a><span class="kw">library</span>(dplyr)</span>
<span id="cb129-2"><a href="data-wrangling.html#cb129-2"></a><span class="kw">library</span>(tidyr)</span>
<span id="cb129-3"><a href="data-wrangling.html#cb129-3"></a><span class="kw">library</span>(stringr)</span>
<span id="cb129-4"><a href="data-wrangling.html#cb129-4"></a>usa&lt;-<span class="kw">read.csv</span>(<span class="st">&quot;./Data-Use/Covid_bystate_Table130420.csv&quot;</span>)</span>
<span id="cb129-5"><a href="data-wrangling.html#cb129-5"></a><span class="co"># for demonstration we will select 3 columns of interest</span></span>
<span id="cb129-6"><a href="data-wrangling.html#cb129-6"></a>usa_long &lt;-usa <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb129-7"><a href="data-wrangling.html#cb129-7"></a><span class="st">  </span><span class="kw">select</span>(Jurisdiction,NumberCases31.<span class="fl">03.20</span>,NumberCases07.<span class="fl">04.20</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">pivot_longer</span>(<span class="op">-</span>Jurisdiction,<span class="dt">names_to =</span> <span class="st">&quot;Date&quot;</span>,<span class="dt">values_to =</span> <span class="st">&quot;Number.Cases&quot;</span>)  </span>
<span id="cb129-8"><a href="data-wrangling.html#cb129-8"></a>usa_long<span class="op">$</span>Date &lt;-<span class="st"> </span><span class="kw">str_replace</span>(usa_long<span class="op">$</span>Date,<span class="st">&quot;NumberCases&quot;</span>,<span class="st">&quot;&quot;</span>)</span>
<span id="cb129-9"><a href="data-wrangling.html#cb129-9"></a></span>
<span id="cb129-10"><a href="data-wrangling.html#cb129-10"></a><span class="co">#data in wide format</span></span>
<span id="cb129-11"><a href="data-wrangling.html#cb129-11"></a><span class="kw">head</span>(usa <span class="op">%&gt;%</span><span class="kw">select</span>(Jurisdiction,NumberCases31.<span class="fl">03.20</span>,NumberCases07.<span class="fl">04.20</span>),<span class="dv">6</span>) </span></code></pre></div>
<pre><code>##   Jurisdiction NumberCases31.03.20 NumberCases07.04.20
## 1      Alabama                 999                2197
## 2       Alaska                 133                 213
## 3      Arizona                1289                2575
## 4     Arkansas                 560                 993
## 5   California                8131               15865
## 6     Colorado                2966                5429</code></pre>
<div class="sourceCode" id="cb131"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb131-1"><a href="data-wrangling.html#cb131-1"></a><span class="co">#data in long format</span></span>
<span id="cb131-2"><a href="data-wrangling.html#cb131-2"></a><span class="kw">head</span>(usa_long,<span class="dv">6</span>) </span></code></pre></div>
<pre><code>## # A tibble: 6 x 3
##   Jurisdiction Date     Number.Cases
##   &lt;chr&gt;        &lt;chr&gt;           &lt;int&gt;
## 1 Alabama      31.03.20          999
## 2 Alabama      07.04.20         2197
## 3 Alaska       31.03.20          133
## 4 Alaska       07.04.20          213
## 5 Arizona      31.03.20         1289
## 6 Arizona      07.04.20         2575</code></pre>
</div>
</div>
<div id="regular-expressions" class="section level2">
<h2><span class="header-section-number">3.4</span> Regular Expressions</h2>
<p>Here is a short tutorial on regular expression. We will begin using base R. This section is based on experience trying to clean a data frame containing many
words used to describe one disease or one drug.</p>
<div id="base-r" class="section level3">
<h3><span class="header-section-number">3.4.1</span> base R</h3>
<div class="sourceCode" id="cb133"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb133-1"><a href="data-wrangling.html#cb133-1"></a><span class="co">#create example dataframe</span></span>
<span id="cb133-2"><a href="data-wrangling.html#cb133-2"></a>df&lt;-<span class="kw">data.frame</span>(</span>
<span id="cb133-3"><a href="data-wrangling.html#cb133-3"></a>  <span class="dt">drug=</span><span class="kw">c</span>(<span class="st">&quot;valium 1mg&quot;</span>,<span class="st">&quot;verapamil sr&quot;</span>,<span class="st">&quot;betaloc zoc&quot;</span>,<span class="st">&quot;tramadol&quot;</span>,<span class="st">&quot;valium (diazepam)&quot;</span>),</span>
<span id="cb133-4"><a href="data-wrangling.html#cb133-4"></a>  <span class="dt">infection=</span><span class="kw">c</span>(<span class="st">&quot;pneumonia&quot;</span>,<span class="st">&quot;aspiration pneumonia&quot;</span>,<span class="st">&quot;tracheobronchitis&quot;</span>,<span class="st">&quot;respiratory tract infection&quot;</span>,<span class="st">&quot;respiratory.tract.infection&quot;</span>))</span>
<span id="cb133-5"><a href="data-wrangling.html#cb133-5"></a>df</span></code></pre></div>
<pre><code>##                drug                   infection
## 1        valium 1mg                   pneumonia
## 2      verapamil sr        aspiration pneumonia
## 3       betaloc zoc           tracheobronchitis
## 4          tramadol respiratory tract infection
## 5 valium (diazepam) respiratory.tract.infection</code></pre>
<p>Now that we have a data frame, we can use pattern matching to replace part of phrase. This step can be done simply using <em>gsub</em> command. First create a list
so that the computer searches the phrases in the list.</p>
<div class="sourceCode" id="cb135"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb135-1"><a href="data-wrangling.html#cb135-1"></a><span class="co">#create list to remove phrase</span></span>
<span id="cb135-2"><a href="data-wrangling.html#cb135-2"></a>redun=<span class="kw">c</span>(<span class="st">&quot;1mg&quot;</span>, <span class="st">&quot;zoc&quot;</span>, <span class="st">&quot;sr&quot;</span>)</span>
<span id="cb135-3"><a href="data-wrangling.html#cb135-3"></a>pat=<span class="kw">paste0</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">b(&quot;</span>,<span class="kw">paste0</span>(redun,<span class="dt">collapse =</span> <span class="st">&quot;|&quot;</span>),<span class="st">&quot;)</span><span class="ch">\\</span><span class="st">b&quot;</span>)</span>
<span id="cb135-4"><a href="data-wrangling.html#cb135-4"></a>df<span class="op">$</span>drug1&lt;-<span class="kw">gsub</span>(pat,<span class="st">&quot;&quot;</span>,df<span class="op">$</span>drug)</span>
<span id="cb135-5"><a href="data-wrangling.html#cb135-5"></a>df<span class="op">$</span>drug1</span></code></pre></div>
<pre><code>## [1] &quot;valium &quot;           &quot;verapamil &quot;        &quot;betaloc &quot;         
## [4] &quot;tramadol&quot;          &quot;valium (diazepam)&quot;</code></pre>
<div class="sourceCode" id="cb137"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb137-1"><a href="data-wrangling.html#cb137-1"></a><span class="co">#create list to remove phrase</span></span>
<span id="cb137-2"><a href="data-wrangling.html#cb137-2"></a>redunc1=<span class="kw">c</span>(<span class="st">&quot;respiratory tract infection&quot;</span>, <span class="st">&quot;tracheobronchitis&quot;</span>, <span class="st">&quot;aspiration&quot;</span>)</span>
<span id="cb137-3"><a href="data-wrangling.html#cb137-3"></a>pat=<span class="kw">paste0</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">b(&quot;</span>,<span class="kw">paste0</span>(redunc1,<span class="dt">collapse =</span> <span class="st">&quot;|&quot;</span>),<span class="st">&quot;)</span><span class="ch">\\</span><span class="st">b&quot;</span>)</span>
<span id="cb137-4"><a href="data-wrangling.html#cb137-4"></a>df<span class="op">$</span>infection1&lt;-<span class="kw">gsub</span>(pat,<span class="st">&quot;&quot;</span>,df<span class="op">$</span>infection)</span>
<span id="cb137-5"><a href="data-wrangling.html#cb137-5"></a>df<span class="op">$</span>infection1</span></code></pre></div>
<pre><code>## [1] &quot;pneumonia&quot;                   &quot; pneumonia&quot;                 
## [3] &quot;&quot;                            &quot;&quot;                           
## [5] &quot;respiratory.tract.infection&quot;</code></pre>
<p>This section deals with meta-characterers. Examples of meta-characters include
$ . + * ? ^ () {} []. These meta-characters requires the double back slashes \.</p>
<div class="sourceCode" id="cb139"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb139-1"><a href="data-wrangling.html#cb139-1"></a><span class="co">#create list to remove phrase</span></span>
<span id="cb139-2"><a href="data-wrangling.html#cb139-2"></a>redun=<span class="kw">c</span>(<span class="st">&quot;1mg&quot;</span>, <span class="st">&quot;zoc&quot;</span>, <span class="st">&quot;sr&quot;</span>)</span>
<span id="cb139-3"><a href="data-wrangling.html#cb139-3"></a>pat=<span class="kw">paste0</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">b(&quot;</span>,<span class="kw">paste0</span>(redun, <span class="dt">collapse =</span> <span class="st">&quot;|&quot;</span>),<span class="st">&quot;)</span><span class="ch">\\</span><span class="st">b&quot;</span>)   </span>
<span id="cb139-4"><a href="data-wrangling.html#cb139-4"></a>df<span class="op">$</span>drug2&lt;-<span class="kw">gsub</span>(pat,<span class="st">&quot;&quot;</span>,df<span class="op">$</span>drug)</span>
<span id="cb139-5"><a href="data-wrangling.html#cb139-5"></a></span>
<span id="cb139-6"><a href="data-wrangling.html#cb139-6"></a><span class="co">#[a-z] indicates any letter</span></span>
<span id="cb139-7"><a href="data-wrangling.html#cb139-7"></a><span class="co">#[a-z]+ indicates any letter and those that follow the intial letter</span></span>
<span id="cb139-8"><a href="data-wrangling.html#cb139-8"></a>df<span class="op">$</span>drug2&lt;-<span class="kw">gsub</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">(|[a-z]+</span><span class="ch">\\</span><span class="st">)&quot;</span>,<span class="st">&quot;&quot;</span>,df<span class="op">$</span>drug2)</span>
<span id="cb139-9"><a href="data-wrangling.html#cb139-9"></a>df<span class="op">$</span>drug2</span></code></pre></div>
<pre><code>## [1] &quot;valium &quot;    &quot;verapamil &quot; &quot;betaloc &quot;   &quot;tramadol&quot;   &quot;valium &quot;</code></pre>
<p>Back to our data frame df, we want to remove or change the different words accounting for pneumonia.</p>
<div class="sourceCode" id="cb141"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb141-1"><a href="data-wrangling.html#cb141-1"></a>redunc=<span class="kw">c</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">.&quot;</span>)</span>
<span id="cb141-2"><a href="data-wrangling.html#cb141-2"></a>redunc1=<span class="kw">c</span>(<span class="st">&quot;respiratory tract infection&quot;</span>, <span class="st">&quot;tracheobronchitis&quot;</span>, <span class="st">&quot;aspiration&quot;</span>)</span>
<span id="cb141-3"><a href="data-wrangling.html#cb141-3"></a>pat=<span class="kw">paste0</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">b(&quot;</span>,<span class="kw">paste0</span>(redunc,<span class="dt">collapse =</span> <span class="st">&quot;|&quot;</span>),<span class="st">&quot;)</span><span class="ch">\\</span><span class="st">b&quot;</span>)</span>
<span id="cb141-4"><a href="data-wrangling.html#cb141-4"></a>df<span class="op">$</span>infection2&lt;-<span class="kw">gsub</span>(pat,<span class="st">&quot; &quot;</span>,df<span class="op">$</span>infection)</span>
<span id="cb141-5"><a href="data-wrangling.html#cb141-5"></a>pat=<span class="kw">paste0</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">b(&quot;</span>,<span class="kw">paste0</span>(redunc1,<span class="dt">collapse =</span> <span class="st">&quot;|&quot;</span>),<span class="st">&quot;)</span><span class="ch">\\</span><span class="st">b&quot;</span>)</span>
<span id="cb141-6"><a href="data-wrangling.html#cb141-6"></a>df<span class="op">$</span>infection2&lt;-<span class="kw">gsub</span>(pat,<span class="st">&quot; &quot;</span>,df<span class="op">$</span>infection2)</span>
<span id="cb141-7"><a href="data-wrangling.html#cb141-7"></a>df<span class="op">$</span>infection2</span></code></pre></div>
<pre><code>## [1] &quot;pneumonia&quot;   &quot;  pneumonia&quot; &quot; &quot;           &quot; &quot;           &quot; &quot;</code></pre>
</div>
<div id="stringr" class="section level3">
<h3><span class="header-section-number">3.4.2</span> stringr</h3>
<p>The following examples are taken from excel after conversion from pdf. In the process of conversion errors were introduced in the conversion from pdf to excel. A full list of the options available can be found at <a href="https://stringr.tidyverse.org/articles/regular-expressions.html" class="uri">https://stringr.tidyverse.org/articles/regular-expressions.html</a></p>
<div class="sourceCode" id="cb143"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb143-1"><a href="data-wrangling.html#cb143-1"></a><span class="kw">library</span>(stringr)</span>
<span id="cb143-2"><a href="data-wrangling.html#cb143-2"></a><span class="co">#error introduced by double space </span></span>
<span id="cb143-3"><a href="data-wrangling.html#cb143-3"></a>a&lt;-<span class="kw">c</span>(<span class="st">&quot;8396 (7890 to 8920)&quot;</span>,<span class="st">&quot;6 301 113(6 085 757 to 6 517 308)&quot;</span>,</span>
<span id="cb143-4"><a href="data-wrangling.html#cb143-4"></a>     <span class="st">&quot;4 841 208 (4 533 619 to 5 141 654)&quot;</span>,</span>
<span id="cb143-5"><a href="data-wrangling.html#cb143-5"></a>     <span class="st">&quot;1 407 701 (127 445 922 to 138 273 863)&quot;</span>,</span>
<span id="cb143-6"><a href="data-wrangling.html#cb143-6"></a>     <span class="st">&quot;4 841 208</span><span class="ch">\n</span><span class="st">(4 533 619 to</span><span class="ch">\n</span><span class="st">5 141 654)&quot;</span>)</span>
<span id="cb143-7"><a href="data-wrangling.html#cb143-7"></a></span>
<span id="cb143-8"><a href="data-wrangling.html#cb143-8"></a>b&lt;-<span class="kw">str_replace</span> (a, <span class="st">&quot;</span><span class="ch">\\</span><span class="st">(c.*</span><span class="ch">\\</span><span class="st">)&quot;</span>,<span class="st">&quot;&quot;</span>)</span>
<span id="cb143-9"><a href="data-wrangling.html#cb143-9"></a></span>
<span id="cb143-10"><a href="data-wrangling.html#cb143-10"></a><span class="co">#this is a complex example to clean and requires several steps. Note that the original data in the list a is now assigned to b. </span></span>
<span id="cb143-11"><a href="data-wrangling.html#cb143-11"></a>b&lt;-<span class="kw">str_replace</span>(a,<span class="st">&quot;</span><span class="ch">\n</span><span class="st">&quot;</span>,<span class="st">&quot;&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb143-12"><a href="data-wrangling.html#cb143-12"></a><span class="st">  </span><span class="co">#remove (</span></span>
<span id="cb143-13"><a href="data-wrangling.html#cb143-13"></a><span class="st">  </span><span class="kw">str_replace</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">(.*&quot;</span>,<span class="st">&quot;&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb143-14"><a href="data-wrangling.html#cb143-14"></a><span class="st">  </span><span class="kw">str_replace</span>(<span class="st">&quot;</span><span class="ch">\n</span><span class="st">.*&quot;</span>,<span class="st">&quot;&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb143-15"><a href="data-wrangling.html#cb143-15"></a><span class="st">  </span><span class="co">#remove )</span></span>
<span id="cb143-16"><a href="data-wrangling.html#cb143-16"></a><span class="st">  </span><span class="kw">str_replace</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">)&quot;</span>,<span class="st">&quot;&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb143-17"><a href="data-wrangling.html#cb143-17"></a><span class="st">  </span><span class="co">#remove empty space</span></span>
<span id="cb143-18"><a href="data-wrangling.html#cb143-18"></a><span class="st">  </span><span class="kw">str_replace</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">s&quot;</span>,<span class="st">&quot;&quot;</span>) <span class="op">%&gt;%</span></span>
<span id="cb143-19"><a href="data-wrangling.html#cb143-19"></a><span class="st">  </span><span class="kw">str_replace</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">s&quot;</span>,<span class="st">&quot;&quot;</span>)<span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as.numeric</span>()</span>
<span id="cb143-20"><a href="data-wrangling.html#cb143-20"></a>b</span></code></pre></div>
<pre><code>## [1]    8396 6301113 4841208 1407701 4841208</code></pre>
<p>Another example. This time the 2 numbers in the column are separated by a slash sign. Supposed you want to keep the denominator. The first remove the number before the slash sign. The _*_ metacharacter denotes the action occurs at the
end.</p>
<div class="sourceCode" id="cb145"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb145-1"><a href="data-wrangling.html#cb145-1"></a>df.d&lt;-<span class="kw">data.frame</span>(<span class="dt">seizure.rate=</span><span class="kw">c</span>(<span class="st">&quot;59/90&quot;</span>, <span class="st">&quot;90/100&quot;</span>, <span class="st">&quot;3/23&quot;</span>))</span>
<span id="cb145-2"><a href="data-wrangling.html#cb145-2"></a>df.d<span class="op">$</span>seizure.number&lt;-<span class="kw">str_replace</span>(df.d<span class="op">$</span>seizure.rate,<span class="st">&quot;[0-9]*&quot;</span>,<span class="st">&quot;&quot;</span>) </span>
<span id="cb145-3"><a href="data-wrangling.html#cb145-3"></a>df.d<span class="op">$</span>seizure.number</span></code></pre></div>
<pre><code>## [1] &quot;/90&quot;  &quot;/100&quot; &quot;/23&quot;</code></pre>
<p>Now combine with the next step to remove the slash sign.</p>
<div class="sourceCode" id="cb147"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb147-1"><a href="data-wrangling.html#cb147-1"></a><span class="co">#We used [0-9] to denote any number from 0 to 9. For text, one can use [A-Z].</span></span>
<span id="cb147-2"><a href="data-wrangling.html#cb147-2"></a>df.d<span class="op">$</span>seizure.number&lt;-<span class="kw">str_replace</span>(df.d<span class="op">$</span>seizure.rate,<span class="st">&quot;^[0-9]*&quot;</span>,<span class="st">&quot;&quot;</span>)<span class="op">%&gt;%</span></span>
<span id="cb147-3"><a href="data-wrangling.html#cb147-3"></a><span class="st">  </span><span class="kw">str_replace</span>(<span class="st">&quot;/&quot;</span>,<span class="st">&quot;</span><span class="ch">\\</span><span class="st">&quot;</span>)</span>
<span id="cb147-4"><a href="data-wrangling.html#cb147-4"></a>df.d<span class="op">$</span>seizure.number</span></code></pre></div>
<pre><code>## [1] &quot;90&quot;  &quot;100&quot; &quot;23&quot;</code></pre>
<p>Removing the denominator requires a different approach. First remove the last number then the slash sign.</p>
<div class="sourceCode" id="cb149"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb149-1"><a href="data-wrangling.html#cb149-1"></a>df.d<span class="op">$</span>case&lt;-<span class="kw">str_replace</span>(df.d<span class="op">$</span>seizure.rate,<span class="st">&quot;/[0-9]*&quot;</span>,<span class="st">&quot; &quot;</span>)</span>
<span id="cb149-2"><a href="data-wrangling.html#cb149-2"></a>df.d<span class="op">$</span>case</span></code></pre></div>
<pre><code>## [1] &quot;59 &quot; &quot;90 &quot; &quot;3 &quot;</code></pre>
<p>The example below has several words mixed in numeric vector columns. The words
are a mixture of upper and lower cases. Note that “NA” is treated as a word character while <em>NA</em> is treated as Not Available by R. This recognition is important as removing them requires different actions. Character “NA” can be removed by <em>str_replace</em> while <em>NA</em> requires <em>is.na</em> operator.</p>
<div class="sourceCode" id="cb151"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb151-1"><a href="data-wrangling.html#cb151-1"></a>A&lt;-<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>,<span class="st">&quot;NA&quot;</span>,<span class="dv">4</span>,<span class="st">&quot;no COW now&quot;</span>)</span>
<span id="cb151-2"><a href="data-wrangling.html#cb151-2"></a>B&lt;-<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="ot">NA</span>,<span class="dv">4</span>,<span class="st">&quot;NA&quot;</span>,<span class="st">&quot;check&quot;</span>)</span>
<span id="cb151-3"><a href="data-wrangling.html#cb151-3"></a>C&lt;-<span class="kw">c</span>(<span class="dv">1</span>,<span class="st">&quot;not now&quot;</span>,<span class="dv">2</span>,<span class="dv">3</span>, <span class="ot">NA</span> ,<span class="dv">5</span>)</span>
<span id="cb151-4"><a href="data-wrangling.html#cb151-4"></a></span>
<span id="cb151-5"><a href="data-wrangling.html#cb151-5"></a>D&lt;-<span class="kw">data.frame</span>(A,B,C) </span>
<span id="cb151-6"><a href="data-wrangling.html#cb151-6"></a></span>
<span id="cb151-7"><a href="data-wrangling.html#cb151-7"></a><span class="co">#str_replace on one column</span></span>
<span id="cb151-8"><a href="data-wrangling.html#cb151-8"></a>D<span class="op">$</span>A1&lt;-<span class="kw">str_replace</span>(D<span class="op">$</span>A,<span class="st">&quot;[A-Z]+&quot;</span>,<span class="st">&quot;&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_replace</span>(<span class="st">&quot;[a-z]+&quot;</span>,<span class="st">&quot;&quot;</span>)</span>
<span id="cb151-9"><a href="data-wrangling.html#cb151-9"></a></span>
<span id="cb151-10"><a href="data-wrangling.html#cb151-10"></a><span class="co">#change to lower case</span></span>
<span id="cb151-11"><a href="data-wrangling.html#cb151-11"></a>D<span class="op">$</span>A2&lt;-<span class="kw">str_to_lower</span>(D<span class="op">$</span>A) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_replace</span>(<span class="st">&quot;[a-z]+&quot;</span>,<span class="st">&quot;&quot;</span>)</span>
<span id="cb151-12"><a href="data-wrangling.html#cb151-12"></a></span>
<span id="cb151-13"><a href="data-wrangling.html#cb151-13"></a><span class="co">#remove space before replacement</span></span>
<span id="cb151-14"><a href="data-wrangling.html#cb151-14"></a>D<span class="op">$</span>A3&lt;-<span class="kw">str_to_lower</span>(D<span class="op">$</span>A) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_replace</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">s+&quot;</span>,<span class="st">&quot;&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_replace</span>(<span class="st">&quot;[a-z]+&quot;</span>,<span class="st">&quot;&quot;</span>)</span>
<span id="cb151-15"><a href="data-wrangling.html#cb151-15"></a></span>
<span id="cb151-16"><a href="data-wrangling.html#cb151-16"></a><span class="co">#note that this action does not remove the third word</span></span>
<span id="cb151-17"><a href="data-wrangling.html#cb151-17"></a>D<span class="op">$</span>A4&lt;-<span class="kw">str_to_lower</span>(D<span class="op">$</span>A) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_replace</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">s&quot;</span>,<span class="st">&quot;&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_replace</span>(<span class="st">&quot;[a-z]+&quot;</span>,<span class="st">&quot;&quot;</span>)</span>
<span id="cb151-18"><a href="data-wrangling.html#cb151-18"></a></span>
<span id="cb151-19"><a href="data-wrangling.html#cb151-19"></a><span class="co">#repeat removal of empty space</span></span>
<span id="cb151-20"><a href="data-wrangling.html#cb151-20"></a>D<span class="op">$</span>A5&lt;-<span class="kw">str_to_lower</span>(D<span class="op">$</span>A) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_replace</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">s&quot;</span>,<span class="st">&quot;&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb151-21"><a href="data-wrangling.html#cb151-21"></a><span class="st">  </span><span class="kw">str_replace</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">s&quot;</span>,<span class="st">&quot;&quot;</span>) <span class="op">%&gt;%</span><span class="st">  </span><span class="kw">str_replace</span>(<span class="st">&quot;[a-z]+&quot;</span>,<span class="st">&quot;&quot;</span>)</span>
<span id="cb151-22"><a href="data-wrangling.html#cb151-22"></a></span>
<span id="cb151-23"><a href="data-wrangling.html#cb151-23"></a><span class="co">#apply str_replace_all rather than repeat</span></span>
<span id="cb151-24"><a href="data-wrangling.html#cb151-24"></a>D<span class="op">$</span>A6&lt;-<span class="kw">str_to_lower</span>(D<span class="op">$</span>A) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_replace_all</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">s&quot;</span>,<span class="st">&quot;&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb151-25"><a href="data-wrangling.html#cb151-25"></a><span class="st">    </span><span class="kw">str_replace</span>(<span class="st">&quot;[a-z]+&quot;</span>,<span class="st">&quot;&quot;</span>)</span>
<span id="cb151-26"><a href="data-wrangling.html#cb151-26"></a></span>
<span id="cb151-27"><a href="data-wrangling.html#cb151-27"></a><span class="co">#now combine into vector. Note the use of c to combine the vector and replace </span></span>
<span id="cb151-28"><a href="data-wrangling.html#cb151-28"></a><span class="co">#the comma with equal sign</span></span>
<span id="cb151-29"><a href="data-wrangling.html#cb151-29"></a>D<span class="op">$</span>A7&lt;-<span class="kw">str_to_lower</span>(D<span class="op">$</span>A) <span class="op">%&gt;%</span></span>
<span id="cb151-30"><a href="data-wrangling.html#cb151-30"></a><span class="st"> </span><span class="kw">str_replace_all</span>(<span class="kw">c</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">s&quot;</span>=<span class="st">&quot;&quot;</span>,<span class="st">&quot;[a-z]+&quot;</span>=<span class="st">&quot;&quot;</span>))</span>
<span id="cb151-31"><a href="data-wrangling.html#cb151-31"></a></span>
<span id="cb151-32"><a href="data-wrangling.html#cb151-32"></a>D</span></code></pre></div>
<pre><code>##            A     B       C    A1       A2   A3   A4 A5 A6 A7
## 1          1     1       1     1        1    1    1  1  1  1
## 2          2     2 not now     2        2    2    2  2  2  2
## 3          3  &lt;NA&gt;       2     3        3    3    3  3  3  3
## 4         NA     4       3                                  
## 5          4    NA    &lt;NA&gt;     4        4    4    4  4  4  4
## 6 no COW now check       5   now  cow now  now  now</code></pre>
<p>The lessons from above can be combine in when creating data frame. The
<em>mutate_if</em> function enable multiple columns to be changed. One problem to
handle adding multiple columns which contain <em>NA</em> is the use of <em>rowSums</em> and <em>dplyr::select</em>. These examples are illustrated below.</p>
<div class="sourceCode" id="cb153"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb153-1"><a href="data-wrangling.html#cb153-1"></a><span class="co">#use the mutate function </span></span>
<span id="cb153-2"><a href="data-wrangling.html#cb153-2"></a></span>
<span id="cb153-3"><a href="data-wrangling.html#cb153-3"></a>E&lt;-<span class="kw">data.frame</span>(A,B,C) <span class="op">%&gt;%</span></span>
<span id="cb153-4"><a href="data-wrangling.html#cb153-4"></a><span class="st">  </span><span class="kw">mutate</span> (<span class="dt">A=</span><span class="kw">str_to_lower</span>(A) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">str_replace_all</span>(<span class="kw">c</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">s&quot;</span>=<span class="st">&quot;&quot;</span>,<span class="st">&quot;[a-z]+&quot;</span>=<span class="st">&quot;&quot;</span>)),</span>
<span id="cb153-5"><a href="data-wrangling.html#cb153-5"></a>          <span class="dt">B=</span><span class="kw">str_to_lower</span>(B) <span class="op">%&gt;%</span><span class="kw">str_replace_all</span>(<span class="kw">c</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">s&quot;</span>=<span class="st">&quot;&quot;</span>,<span class="st">&quot;[a-z]+&quot;</span>=<span class="st">&quot;&quot;</span>)),</span>
<span id="cb153-6"><a href="data-wrangling.html#cb153-6"></a>          <span class="dt">C=</span><span class="kw">str_to_lower</span>(C) <span class="op">%&gt;%</span><span class="kw">str_replace_all</span>(<span class="kw">c</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">s&quot;</span>=<span class="st">&quot;&quot;</span>,<span class="st">&quot;[a-z]+&quot;</span>=<span class="st">&quot;&quot;</span>)))<span class="op">%&gt;%</span></span>
<span id="cb153-7"><a href="data-wrangling.html#cb153-7"></a><span class="st">  </span></span>
<span id="cb153-8"><a href="data-wrangling.html#cb153-8"></a><span class="st">  </span><span class="co">#change character columns to numeric</span></span>
<span id="cb153-9"><a href="data-wrangling.html#cb153-9"></a><span class="st">  </span><span class="kw">mutate_if</span>(is.character, as.numeric)<span class="op">%&gt;%</span></span>
<span id="cb153-10"><a href="data-wrangling.html#cb153-10"></a><span class="st">  </span></span>
<span id="cb153-11"><a href="data-wrangling.html#cb153-11"></a><span class="st">  </span><span class="co">#add across columns and avoid NA</span></span>
<span id="cb153-12"><a href="data-wrangling.html#cb153-12"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">ABC=</span><span class="kw">rowSums</span>(dplyr<span class="op">::</span><span class="kw">select</span>(.,A<span class="op">:</span>C),<span class="dt">na.rm =</span> T))</span></code></pre></div>
<p>Sometimes, you may only want to keep the number and ignore the words in the column. This can be done using the <em>str_extract</em> function.</p>
<div class="sourceCode" id="cb154"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb154-1"><a href="data-wrangling.html#cb154-1"></a>df.e&lt;-<span class="kw">data.frame</span>(<span class="dt">disability=</span><span class="kw">c</span>(<span class="st">&quot;1 - No significant disability despite symptoms; able to carry out all usual duties and activities&quot;</span>,<span class="st">&quot;5 - Severe disability, bedridden, incontinent and requiring constant nursing care and attention&quot;</span>,<span class="st">&quot;</span></span>
<span id="cb154-2"><a href="data-wrangling.html#cb154-2"></a><span class="st">1 - No significant disability despite symptoms; able to carry out all usual </span></span>
<span id="cb154-3"><a href="data-wrangling.html#cb154-3"></a><span class="st">duties and activities&quot;</span>))</span>
<span id="cb154-4"><a href="data-wrangling.html#cb154-4"></a>df.e<span class="op">$</span>disability2&lt;-<span class="kw">str_extract</span>(df.e<span class="op">$</span>disability,<span class="st">&quot;</span><span class="ch">\\</span><span class="st">w&quot;</span>) <span class="co">#extract number</span></span></code></pre></div>
</div>
</div>
<div id="pdf-to-xcel" class="section level2">
<h2><span class="header-section-number">3.5</span> PDF to xcel</h2>
<p>Sometimes data from public government sites come in PDF form instead of excel. Conversion from pdf to excel or text can be difficult especially with special character eg Danish. There are several libraries for doing this: <em>pdftables</em> (require API key) and <em>pdftools</em>. The example below uses <em>pdftools</em>.
available at <a href="https://docs.ropensci.org/pdftools/" class="uri">https://docs.ropensci.org/pdftools/</a>. The document is the 2018
Danish Stroke Registry report. The <em>tabulizer</em> package is excellent for
converting table data. However, <em>tabulizer</em> package depends on <em>rJava</em> and requires deft handling.</p>
<div class="sourceCode" id="cb155"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb155-1"><a href="data-wrangling.html#cb155-1"></a><span class="kw">library</span>(pdftools)</span></code></pre></div>
<pre><code>## Using poppler version 0.73.0</code></pre>
<div class="sourceCode" id="cb157"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb157-1"><a href="data-wrangling.html#cb157-1"></a>txt&lt;-<span class="kw">pdf_text</span>(<span class="st">&quot;./Data-Use/4669_dap_aarsrapport-2018_24062019final.pdf&quot;</span>)</span>
<span id="cb157-2"><a href="data-wrangling.html#cb157-2"></a><span class="kw">cat</span>(txt[<span class="dv">17</span>]) <span class="co">#browse data page 13+4 filler pages</span></span></code></pre></div>
<pre><code>## 3. Indikatorresultater på lands-, regions- og afdelingsniveau
## Indikator 1a: Andel af patienter med akut apopleksi som indlægges inden for 3 timer
## efter symptomdebut. Standard: = 30%
## Indikator 1b: Andel af patienter med akut apopleksi som indlægges inden for 4,5
## timer efter symptomdebut. Standard: = 40%
##                                      Inden for 3 timer
##                                                   Uoplyst   Aktuelle år         Tidligere år
##                            Standard   Tæller/      antal      2018          2017            2016
##                             opfyldt   nævner        (%)    %    95% CI   % (95% CI)     % (95% CI)
##   Danmark                     ja    4730 / 11794   49 (0)  40  (39 - 41) 39 (38-40)     37 (36-38)
##   Hovedstaden                 ja     1502 / 3439   49 (1)  44  (42 - 45) 40 (38-42)     40 (39-42)
##   Sjælland                    ja      760 / 1917     0 (0) 40  (37 - 42) 39 (36-41)     40 (38-43)
##   Syddanmark                  ja      942 / 2433     0 (0) 39  (37 - 41) 39 (37-41)     35 (33-37)
##   Midtjylland                 ja      918 / 2590     0 (0) 35  (34 - 37) 36 (34-38)     35 (33-37)
##   Nordjylland                 ja      577 / 1341     0 (0) 43  (40 - 46) 41 (39-44)     35 (32-37)
##   Bopæl uden for Danmark      ja          31 / 74    0 (0) 42  (31 - 54) 51 (36-66)     39 (26-53)
##   Hovedstaden                 ja     1502 / 3439   49 (1)  44  (42 - 45) 40 (38-42)     40 (39-42)
##   Albertslund                 ja          17 / 52    0 (0) 33  (20 - 47) 30 (18-44)     43 (27-59)
##   Allerød                     ja          22 / 53    0 (0) 42  (28 - 56) 32 (20-46)     35 (22-50)
##   Ballerup                    ja        43 / 106     0 (0) 41  (31 - 51) 48 (38-58)     40 (31-51)
##   Bornholms Regionskommune    ja          38 / 98    0 (0) 39  (29 - 49) 28 (19-38)     32 (22-43)
##   Brøndby                     ja        45 / 113     0 (0) 40  (31 - 49) 31 (22-42)     34 (23-47)
##   Dragør                      ja          18 / 43    0 (0) 42  (27 - 58) 47 (30-65)     33 (17-54)
##   Egedal                      ja          45 / 95    0 (0) 47  (37 - 58) 40 (30-52)     48 (37-59)
##   Fredensborg                 ja          36 / 99    0 (0) 36  (27 - 47) 37 (27-47)     43 (33-53)
##   Frederiksberg               ja        74 / 141   13 (8)  52  (44 - 61) 42 (35-50)     52 (44-61)
##   Frederikssund               ja        53 / 141     0 (0) 38  (30 - 46) 39 (31-48)     42 (33-51)
##   Furesø                      ja        55 / 110     0 (0) 50  (40 - 60) 56 (44-67)     50 (38-62)
##   Gentofte                    ja        65 / 123     1 (1) 53  (44 - 62) 46 (37-56)     38 (30-47)
##   Gladsaxe                    ja        68 / 129     0 (0) 53  (44 - 62) 48 (38-57)     36 (27-44)
##   Glostrup                    ja          38 / 72    0 (0) 53  (41 - 65) 40 (27-54)     48 (33-63)
##   Gribskov                    ja        46 / 129     0 (0) 36  (27 - 45) 35 (26-44)     36 (28-46)
##   Halsnæs                     ja          45 / 92    0 (0) 49  (38 - 60) 34 (25-45)     34 (25-44)
##   Helsingør                   ja        50 / 129     0 (0) 39  (30 - 48) 32 (24-40)     38 (30-45)
##   Herlev                      ja          25 / 50    0 (0) 50  (36 - 64) 52 (38-65)     33 (22-46)
##   Hillerød                    ja        47 / 121     0 (0) 39  (30 - 48) 39 (30-49)     40 (30-50)
##   Hvidovre                    ja        57 / 135     0 (0) 42  (34 - 51) 38 (29-48)     47 (37-57)
##                                                                                                    13</code></pre>
<div class="sourceCode" id="cb159"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb159-1"><a href="data-wrangling.html#cb159-1"></a>screenshot13&lt;-</span>
<span id="cb159-2"><a href="data-wrangling.html#cb159-2"></a><span class="st">  </span><span class="kw">pdf_render_page</span>(<span class="st">&quot;./Data-Use/4669_dap_aarsrapport-2018_24062019final.pdf&quot;</span>, </span>
<span id="cb159-3"><a href="data-wrangling.html#cb159-3"></a>                  <span class="dt">page =</span><span class="dv">17</span>)</span>
<span id="cb159-4"><a href="data-wrangling.html#cb159-4"></a>png<span class="op">::</span><span class="kw">writePNG</span>(screenshot13, <span class="st">&quot;./Data-Use/Danish-Stroke-page13.png&quot;</span>)</span>
<span id="cb159-5"><a href="data-wrangling.html#cb159-5"></a></span>
<span id="cb159-6"><a href="data-wrangling.html#cb159-6"></a>knitr<span class="op">::</span><span class="kw">include_graphics</span>(<span class="st">&quot;./Data-Use/Danish-Stroke-page13.png&quot;</span>)</span></code></pre></div>
<p><img src="Data-Use/Danish-Stroke-page13.png" width="298" /></p>
<div id="scanned-text-or-picture" class="section level3">
<h3><span class="header-section-number">3.5.1</span> Scanned text or picture</h3>
<p>Importing data from scanned text will require use of Optical Character
Recognition (OCR). The <em>tesseract</em> library provides an R interface for OCR. In
the example below, a picture is taken from same CDC website containing mortality data (<a href="https://www.cdc.gov/coronavirus/2019-ncov/covid-data/covidview/04102020/" class="uri">https://www.cdc.gov/coronavirus/2019-ncov/covid-data/covidview/04102020/</a>
nchs-data.html). The screenshot of this website was then cleaned in <em>paint</em>.
The data is available in the Data-Use folder.</p>
<div class="sourceCode" id="cb160"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb160-1"><a href="data-wrangling.html#cb160-1"></a><span class="kw">library</span>(tesseract)</span>
<span id="cb160-2"><a href="data-wrangling.html#cb160-2"></a>eng &lt;-<span class="st"> </span><span class="kw">tesseract</span>(<span class="st">&quot;eng&quot;</span>) <span class="co">#english</span></span>
<span id="cb160-3"><a href="data-wrangling.html#cb160-3"></a>text &lt;-<span class="st"> </span>tesseract<span class="op">::</span><span class="kw">ocr</span>(<span class="st">&quot;./Data-Use/Covid_PNG100420.png&quot;</span>, <span class="dt">engine =</span> eng)</span>
<span id="cb160-4"><a href="data-wrangling.html#cb160-4"></a><span class="kw">cat</span>(text)</span></code></pre></div>
<pre><code>## NCHS Mortality Surveillance Data
## Data as of April 9, 2020
## For the Week Ending April 4, 2020 (Week 14)
## 
## COVID-19 Deaths Pneumonia Deaths* Influenza Deaths
## Year Week TotalDeaths Number %ofTotal Number %ofTotal Number  %of Total
## 2019 40 52,452 0 0 2,703 5.15 16 0.03
## 2019 4) 52,860 0 0 2,770 5.24 16 0.03
## 2019 42 54,129 0 0 2,977 5.50 18 0.03
## 2019 43 53,914 0 0 2,985 5.54 30 0.06
## 2019 44 53,980 0 0 2,908 5.39 31 0.06
## 2019 AS 55,468 0 0 3,063 5.52 31 0.06
## 2019 46 55,684 0 0 3,096 5.56 39 0.07
## 2019 47 55,986 0 0 2,993 5.35 50 0.09
## 2019 48 55,238 0 0 2,976 5.38 65 0.12
## 2019 49 56,990 0 0 3,305 5.80 99 0.17
## 2019 50 57,276 0 0 3,448 6.02 111 0.19
## 2019 51 56,999 0 0 3,345 5.87 125 0.22
## 2019 52 57,956 0 0 3,478 5.99 198 0.34
## 2020 1 58,961 0 0 3,998 6.77 416 0.71
## 2020 2 58,962 0 0 3,995 6.76 450 0.76
## 2020 3 57,371 0 0 3,903 6.78 441 0.77
## 2020 4 56,666 0 0 3,742 6.56 468 0.83
## 2020 5 56,381 0 0 3,617 6.42 452 0.80
## 2020 6 56,713 0 0 3,599 6.35 482 0.85
## 2020 7 55,237 0 0 3,577 6.48 487 0.88</code></pre>
</div>
</div>
<div id="web-scraping" class="section level2">
<h2><span class="header-section-number">3.6</span> Web scraping</h2>
<p>The readers may ask why web scraping for healthcare. A pertinent example related to COVID-19 data is provided below. The library <em>rvest</em> is helpful at scraping data from an internet page. The <em>rvest</em> library assumes that web contents have
xml document-tree representation. The different options available for web
scraping with <em>rvest</em> are available at the website <a href="https://rvest.tidyverse.org/reference/" class="uri">https://rvest.tidyverse.org/reference/</a>. The user can use CSS selectors to scrape content. The library <em>Rselenium</em> is also useful for web scraping. For dynamic
web page, the library <em>CasperJS</em> library does a better job especially if the
data contain embedded java script.</p>
<p>The library <em>cdccovidview</em> provides access to the CDC website on COVID-19. In
the example below, we will try to this manually. Data from CDC website on
COVID-19 is downloaded, cleaned and saved in csv format. It is important to pay attention to the data. The first row contains header and is removed. There are several columns with commas. These commas can be removed using the exercises above. Further the data is updated on weekly basis. As such the data needs to be converted into a date time format using <em>lubridate</em>.</p>
<div class="sourceCode" id="cb162"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb162-1"><a href="data-wrangling.html#cb162-1"></a><span class="kw">library</span>(rvest)</span></code></pre></div>
<pre><code>## Loading required package: xml2</code></pre>
<pre><code>## 
## Attaching package: &#39;rvest&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:purrr&#39;:
## 
##     pluck</code></pre>
<pre><code>## The following object is masked from &#39;package:readr&#39;:
## 
##     guess_encoding</code></pre>
<div class="sourceCode" id="cb167"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb167-1"><a href="data-wrangling.html#cb167-1"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb167-2"><a href="data-wrangling.html#cb167-2"></a><span class="co">#assign handle to web page accessed 12/4/20</span></span>
<span id="cb167-3"><a href="data-wrangling.html#cb167-3"></a><span class="co">#cdc&lt;-read_html(&quot;https://www.cdc.gov/coronavirus/2019-ncov/covid-data/</span></span>
<span id="cb167-4"><a href="data-wrangling.html#cb167-4"></a><span class="co">#               covidview/04102020/nchs-data.html&quot;)</span></span>
<span id="cb167-5"><a href="data-wrangling.html#cb167-5"></a><span class="co"># scrape all div tags</span></span>
<span id="cb167-6"><a href="data-wrangling.html#cb167-6"></a><span class="co">#html_tag &lt;- cdc %&gt;% html_nodes(&quot;div&quot;)</span></span>
<span id="cb167-7"><a href="data-wrangling.html#cb167-7"></a><span class="co"># scrape header h1 tags</span></span>
<span id="cb167-8"><a href="data-wrangling.html#cb167-8"></a><span class="co">#html_list&lt;-html_tag %&gt;% html_nodes(&quot;h1&quot;) %&gt;% html_text()</span></span>
<span id="cb167-9"><a href="data-wrangling.html#cb167-9"></a><span class="co">#there is only one table on this web page</span></span>
<span id="cb167-10"><a href="data-wrangling.html#cb167-10"></a><span class="co">#Table1&lt;- cdc %&gt;% html_node(&quot;table&quot;) %&gt;% html_table(fill = TRUE)</span></span>
<span id="cb167-11"><a href="data-wrangling.html#cb167-11"></a><span class="co">#Table1 has a header row</span></span>
<span id="cb167-12"><a href="data-wrangling.html#cb167-12"></a><span class="co">#Table1&lt;-Table1[-1,]</span></span>
<span id="cb167-13"><a href="data-wrangling.html#cb167-13"></a><span class="co">#The data in the Total Deaths column has a comma </span></span>
<span id="cb167-14"><a href="data-wrangling.html#cb167-14"></a><span class="co">#Table1$Total.Deaths&lt;-as.numeric(gsub(&quot;,&quot;,&quot;&quot;,Table1$`Total Deaths`))</span></span>
<span id="cb167-15"><a href="data-wrangling.html#cb167-15"></a><span class="co">#now combine the year and week column to Date</span></span>
<span id="cb167-16"><a href="data-wrangling.html#cb167-16"></a><span class="co">#Table1$Date&lt;-lubridate::parse_date_time(paste(Table1$Year, Table1$Week, &#39;Mon&#39;, sep=&quot;/&quot;),&#39;Y/W/a&#39;)</span></span>
<span id="cb167-17"><a href="data-wrangling.html#cb167-17"></a><span class="co">#there are still commas remaining in some columns. This is a useful exercise for the reader. A solution is provided in the next example.  </span></span>
<span id="cb167-18"><a href="data-wrangling.html#cb167-18"></a><span class="co">#write.csv(Table1,file=&quot;./Data-Use/Covid_Table100420.csv&quot;)</span></span></code></pre></div>
<p>The next example is from the CDC COVID-19 website. It poses a different
challenges as there are several columns with the same names. In this case we
will rename the column by index. There are several columns containing commas. Rather than removing column by column we will write a function with lapply to do it over the table. the apply function returns a matrix whereas lapply returns a dataframe. There is one column containing percentage enclosed in a bracket. This can be removed using the example above on metacharacter ie using doule back slash in front of bracket and again at close of bracket.</p>
<div class="sourceCode" id="cb168"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb168-1"><a href="data-wrangling.html#cb168-1"></a><span class="kw">library</span>(rvest)</span>
<span id="cb168-2"><a href="data-wrangling.html#cb168-2"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb168-3"><a href="data-wrangling.html#cb168-3"></a>cdc&lt;-</span>
<span id="cb168-4"><a href="data-wrangling.html#cb168-4"></a><span class="kw">read_html</span>(<span class="st">&quot;https://www.cdc.gov/mmwr/volumes/69/wr/mm6915e4.htm?s_cid=mm6915e4_w&quot;</span>)</span>
<span id="cb168-5"><a href="data-wrangling.html#cb168-5"></a></span>
<span id="cb168-6"><a href="data-wrangling.html#cb168-6"></a><span class="co"># scrape all div tags</span></span>
<span id="cb168-7"><a href="data-wrangling.html#cb168-7"></a>html_tag &lt;-<span class="st"> </span>cdc <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">html_nodes</span>(<span class="st">&quot;div&quot;</span>)</span>
<span id="cb168-8"><a href="data-wrangling.html#cb168-8"></a><span class="co"># scrape header h1 tags</span></span>
<span id="cb168-9"><a href="data-wrangling.html#cb168-9"></a>html_list&lt;-html_tag <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">html_nodes</span>(<span class="st">&quot;h1&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">html_text</span>()</span>
<span id="cb168-10"><a href="data-wrangling.html#cb168-10"></a><span class="co">#there is only one table on this web page</span></span>
<span id="cb168-11"><a href="data-wrangling.html#cb168-11"></a>Table2&lt;-<span class="st"> </span>cdc <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">html_node</span>(<span class="st">&quot;table&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">html_table</span>(<span class="dt">fill =</span> <span class="ot">TRUE</span>)</span>
<span id="cb168-12"><a href="data-wrangling.html#cb168-12"></a><span class="co">#first row is header</span></span>
<span id="cb168-13"><a href="data-wrangling.html#cb168-13"></a><span class="kw">names</span>(Table2) &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(Table2[<span class="dv">1</span>, ])</span>
<span id="cb168-14"><a href="data-wrangling.html#cb168-14"></a>Table2&lt;-Table2[<span class="op">-</span><span class="kw">c</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>,<span class="dv">55</span>),]<span class="co">#rows 1 and 2 are redundant</span></span>
<span id="cb168-15"><a href="data-wrangling.html#cb168-15"></a><span class="co">#rename the columns by index </span></span>
<span id="cb168-16"><a href="data-wrangling.html#cb168-16"></a><span class="kw">names</span>(Table2)[<span class="dv">2</span>] &lt;-<span class="st">&quot;NumberCases31.03.20&quot;</span></span>
<span id="cb168-17"><a href="data-wrangling.html#cb168-17"></a><span class="kw">names</span>(Table2)[<span class="dv">3</span>]&lt;-<span class="st">&quot;CumulativeIncidence31.03.20&quot;</span></span>
<span id="cb168-18"><a href="data-wrangling.html#cb168-18"></a><span class="kw">names</span>(Table2)[<span class="dv">4</span>]&lt;-<span class="st">&quot;NumberCases07.04.20&quot;</span></span>
<span id="cb168-19"><a href="data-wrangling.html#cb168-19"></a><span class="kw">names</span>(Table2)[<span class="dv">5</span>]&lt;-<span class="st">&quot;NumberDeath07.04.20&quot;</span></span>
<span id="cb168-20"><a href="data-wrangling.html#cb168-20"></a><span class="kw">names</span>(Table2)[<span class="dv">6</span>]&lt;-<span class="st">&quot;CumulativeIncidence07.04.20&quot;</span></span>
<span id="cb168-21"><a href="data-wrangling.html#cb168-21"></a><span class="co">#rather than removing column by column we will write a function with lapply to remove commas over the table. the apply function returns a matrix whereas lapply returns a dataframe.</span></span>
<span id="cb168-22"><a href="data-wrangling.html#cb168-22"></a>Table2&lt;-<span class="kw">as.data.frame</span>(<span class="kw">lapply</span>(Table2, <span class="cf">function</span>(y) <span class="kw">gsub</span>(<span class="st">&quot;,&quot;</span>, <span class="st">&quot;&quot;</span>, y))) </span>
<span id="cb168-23"><a href="data-wrangling.html#cb168-23"></a>Table2&lt;-<span class="kw">as.data.frame</span>(<span class="kw">lapply</span>(Table2, <span class="cf">function</span>(x)</span>
<span id="cb168-24"><a href="data-wrangling.html#cb168-24"></a>  <span class="kw">gsub</span>(<span class="st">&quot;</span><span class="ch">\\</span><span class="st">(|[0-9]+</span><span class="ch">\\</span><span class="st">)&quot;</span>,<span class="st">&quot;&quot;</span>,x)))</span>
<span id="cb168-25"><a href="data-wrangling.html#cb168-25"></a><span class="co">#write.csv(Table2,file=&quot;./Data-Use/Covid_bystate_Table130420.csv&quot;)</span></span></code></pre></div>
</div>
<div id="manipulating-medical-images" class="section level2">
<h2><span class="header-section-number">3.7</span> Manipulating Medical Images</h2>
<div id="dicom-and-nifti-format" class="section level3">
<h3><span class="header-section-number">3.7.1</span> DICOM and nifti format</h3>
<p>R can handle a variety of different data format. Medical images are stored as <em>DICOM</em> files for handling and converted to <em>nifti</em> files for analysis. The workhorses are the <em>oro.dicom</em> and <em>oro.nifti</em> libraries. Nifti is an S4 class object with multiple slots for data type. These slots can be accessed by typing the @ after the handle of the file. The slots also contain information on whether the data has been scaled. This can be checked by accessing the <em>scl_slope</em> and <em>scl_inter</em> slots. These data on slope and intercept provide a mean of returning an image to its correct value.</p>
<div class="sourceCode" id="cb169"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb169-1"><a href="data-wrangling.html#cb169-1"></a><span class="kw">library</span>(oro.nifti)</span></code></pre></div>
<pre><code>## oro.nifti 0.11.0</code></pre>
<pre><code>## 
## Attaching package: &#39;oro.nifti&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     slice</code></pre>
<div class="sourceCode" id="cb173"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb173-1"><a href="data-wrangling.html#cb173-1"></a>mca&lt;-<span class="kw">readNIfTI</span>(<span class="st">&quot;./Data-Use/mca_notpa.nii.gz&quot;</span>, <span class="dt">reorient =</span> <span class="ot">FALSE</span>) </span>
<span id="cb173-2"><a href="data-wrangling.html#cb173-2"></a></span>
<span id="cb173-3"><a href="data-wrangling.html#cb173-3"></a>mca<span class="op">@</span>scl_slope</span></code></pre></div>
<pre><code>## [1] 1</code></pre>
<p>To find available slots</p>
<div class="sourceCode" id="cb175"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb175-1"><a href="data-wrangling.html#cb175-1"></a><span class="co">#find available of slots</span></span>
<span id="cb175-2"><a href="data-wrangling.html#cb175-2"></a><span class="kw">slotNames</span>(mca)</span></code></pre></div>
<pre><code>##  [1] &quot;.Data&quot;          &quot;sizeof_hdr&quot;     &quot;data_type&quot;      &quot;db_name&quot;       
##  [5] &quot;extents&quot;        &quot;session_error&quot;  &quot;regular&quot;        &quot;dim_info&quot;      
##  [9] &quot;dim_&quot;           &quot;intent_p1&quot;      &quot;intent_p2&quot;      &quot;intent_p3&quot;     
## [13] &quot;intent_code&quot;    &quot;datatype&quot;       &quot;bitpix&quot;         &quot;slice_start&quot;   
## [17] &quot;pixdim&quot;         &quot;vox_offset&quot;     &quot;scl_slope&quot;      &quot;scl_inter&quot;     
## [21] &quot;slice_end&quot;      &quot;slice_code&quot;     &quot;xyzt_units&quot;     &quot;cal_max&quot;       
## [25] &quot;cal_min&quot;        &quot;slice_duration&quot; &quot;toffset&quot;        &quot;glmax&quot;         
## [29] &quot;glmin&quot;          &quot;descrip&quot;        &quot;aux_file&quot;       &quot;qform_code&quot;    
## [33] &quot;sform_code&quot;     &quot;quatern_b&quot;      &quot;quatern_c&quot;      &quot;quatern_d&quot;     
## [37] &quot;qoffset_x&quot;      &quot;qoffset_y&quot;      &quot;qoffset_z&quot;      &quot;srow_x&quot;        
## [41] &quot;srow_y&quot;         &quot;srow_z&quot;         &quot;intent_name&quot;    &quot;magic&quot;         
## [45] &quot;extender&quot;       &quot;reoriented&quot;</code></pre>
<p>In this example below we will simulated an image of dimensions 5 by 5 by 5.</p>
<div class="sourceCode" id="cb177"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb177-1"><a href="data-wrangling.html#cb177-1"></a><span class="kw">library</span>(oro.nifti)</span>
<span id="cb177-2"><a href="data-wrangling.html#cb177-2"></a><span class="kw">set.seed</span>(<span class="dv">1234</span>)</span>
<span id="cb177-3"><a href="data-wrangling.html#cb177-3"></a>dims =<span class="st"> </span><span class="kw">rep</span>(<span class="dv">5</span>, <span class="dv">3</span>)</span>
<span id="cb177-4"><a href="data-wrangling.html#cb177-4"></a>SimArr =<span class="st"> </span><span class="kw">array</span>(<span class="kw">rnorm</span>(<span class="dv">5</span><span class="op">*</span><span class="dv">5</span><span class="op">*</span><span class="dv">5</span>), <span class="dt">dim =</span> dims)</span>
<span id="cb177-5"><a href="data-wrangling.html#cb177-5"></a>SimIm =<span class="st"> </span>oro.nifti<span class="op">::</span><span class="kw">nifti</span>(SimArr)</span>
<span id="cb177-6"><a href="data-wrangling.html#cb177-6"></a><span class="kw">print</span>(SimIm)</span></code></pre></div>
<pre><code>## NIfTI-1 format
##   Type            : nifti
##   Data Type       : 2 (UINT8)
##   Bits per Pixel  : 8
##   Slice Code      : 0 (Unknown)
##   Intent Code     : 0 (None)
##   Qform Code      : 0 (Unknown)
##   Sform Code      : 0 (Unknown)
##   Dimension       : 5 x 5 x 5
##   Pixel Dimension : 1 x 1 x 1
##   Voxel Units     : Unknown
##   Time Units      : Unknown</code></pre>
<p>View the simulated image.</p>
<div class="sourceCode" id="cb179"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb179-1"><a href="data-wrangling.html#cb179-1"></a>neurobase<span class="op">::</span><span class="kw">ortho2</span>(SimIm)</span></code></pre></div>
<p><img src="Applications-of-R-in-Healthcare_files/figure-html/02-Data-Wrangling-25-3-1.png" width="672" /></p>
<p>This section provides a brief introduction to viewing nifti files. Data are stored as rows, columns and slices. To view sagital image then assign a number to the row data.</p>
<div class="sourceCode" id="cb180"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb180-1"><a href="data-wrangling.html#cb180-1"></a><span class="co">#plot mca #sagittal</span></span>
<span id="cb180-2"><a href="data-wrangling.html#cb180-2"></a><span class="kw">image</span>(mca[<span class="dv">50</span>,,]) </span></code></pre></div>
<p><img src="Applications-of-R-in-Healthcare_files/figure-html/02-Data-Wrangling-26-1.png" width="672" /></p>
<p>To see coronal image, assign a number to the column data.</p>
<div class="sourceCode" id="cb181"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb181-1"><a href="data-wrangling.html#cb181-1"></a><span class="co">#plot mca #coronal</span></span>
<span id="cb181-2"><a href="data-wrangling.html#cb181-2"></a><span class="kw">image</span>(mca[,<span class="dv">70</span>,]) </span></code></pre></div>
<p><img src="Applications-of-R-in-Healthcare_files/figure-html/02-Data-Wrangling-26-1-1.png" width="672" /></p>
<p>To see axial image, assign a number to the slice data.</p>
<div class="sourceCode" id="cb182"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb182-1"><a href="data-wrangling.html#cb182-1"></a><span class="co">#plot mca #axial</span></span>
<span id="cb182-2"><a href="data-wrangling.html#cb182-2"></a><span class="kw">image</span>(mca[,,<span class="dv">35</span>]) </span></code></pre></div>
<p><img src="Applications-of-R-in-Healthcare_files/figure-html/02-Data-Wrangling-26-2-1.png" width="672" /></p>
</div>
<div id="manipulating-array-of-medical-images" class="section level3">
<h3><span class="header-section-number">3.7.2</span> Manipulating array of medical images</h3>
<p>These arrays of medical images should be treated no differently from any other arrays. The imaging data are stored as arrays within the <em>.Data</em> slot in nifti.</p>
<p>Data can be subset using the square bracket. The image is referred to x (right to left), y (front to back), z (superior to inferior).</p>
<div class="sourceCode" id="cb183"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb183-1"><a href="data-wrangling.html#cb183-1"></a><span class="kw">library</span>(oro.nifti)</span>
<span id="cb183-2"><a href="data-wrangling.html#cb183-2"></a><span class="co">#extract data as array using @ function</span></span>
<span id="cb183-3"><a href="data-wrangling.html#cb183-3"></a>img&lt;-<span class="kw">readNIfTI</span>(<span class="st">&quot;./Data-Use/mca_notpa.nii.gz&quot;</span>, <span class="dt">reorient =</span> <span class="ot">FALSE</span>) </span>
<span id="cb183-4"><a href="data-wrangling.html#cb183-4"></a>k&lt;-img<span class="op">@</span>.Data</span>
<span id="cb183-5"><a href="data-wrangling.html#cb183-5"></a></span>
<span id="cb183-6"><a href="data-wrangling.html#cb183-6"></a><span class="co">#change x orientation to right to left 91*109*91</span></span>
<span id="cb183-7"><a href="data-wrangling.html#cb183-7"></a>k1&lt;-k[<span class="dv">91</span><span class="op">:</span><span class="dv">1</span>,,] </span>
<span id="cb183-8"><a href="data-wrangling.html#cb183-8"></a><span class="co">#access slice 35 to verify that the image orientation has been switched.</span></span>
<span id="cb183-9"><a href="data-wrangling.html#cb183-9"></a><span class="kw">image</span>(k1[,,<span class="dv">35</span>])</span></code></pre></div>
<p><img src="Applications-of-R-in-Healthcare_files/figure-html/02-Data-Wrangling-27-1.png" width="672" /></p>
<p>With the image now flipped to the other side, we can create an image by returning the array into a data slot.</p>
<div class="sourceCode" id="cb184"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb184-1"><a href="data-wrangling.html#cb184-1"></a>img2&lt;-img</span>
<span id="cb184-2"><a href="data-wrangling.html#cb184-2"></a>img2<span class="op">@</span>.Data &lt;-<span class="st"> </span>k1</span>
<span id="cb184-3"><a href="data-wrangling.html#cb184-3"></a>img2</span></code></pre></div>
<pre><code>## NIfTI-1 format
##   Type            : nifti
##   Data Type       : 16 (FLOAT32)
##   Bits per Pixel  : 32
##   Slice Code      : 0 (Unknown)
##   Intent Code     : 0 (None)
##   Qform Code      : 0 (Unknown)
##   Sform Code      : 1 (Scanner_Anat)
##   Dimension       : 91 x 109 x 91
##   Pixel Dimension : 2 x 2 x 2
##   Voxel Units     : mm
##   Time Units      : sec</code></pre>
<p>Arrays can be manipulated to split an image into 2 images. Below is a function to split B0 and B1000 images from diffusion series.</p>
<div class="sourceCode" id="cb186"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb186-1"><a href="data-wrangling.html#cb186-1"></a><span class="co">#split dwi file into b0 and b1000</span></span>
<span id="cb186-2"><a href="data-wrangling.html#cb186-2"></a><span class="co">#assume that b1000 is the second volume</span></span>
<span id="cb186-3"><a href="data-wrangling.html#cb186-3"></a><span class="co">#dwi&lt;-readNIfTI(&quot;....nii.gz&quot;,reorient = F)</span></span>
<span id="cb186-4"><a href="data-wrangling.html#cb186-4"></a></span>
<span id="cb186-5"><a href="data-wrangling.html#cb186-5"></a>DWIsplit&lt;-<span class="cf">function</span>(D) {</span>
<span id="cb186-6"><a href="data-wrangling.html#cb186-6"></a>  DWI&lt;-<span class="kw">readNIfTI</span>(D,<span class="dt">reorient =</span> F)</span>
<span id="cb186-7"><a href="data-wrangling.html#cb186-7"></a>  b1000&lt;-dwi <span class="co">#dim (b1000) [1] 384 384  32   2</span></span>
<span id="cb186-8"><a href="data-wrangling.html#cb186-8"></a>  k&lt;-dwi<span class="op">@</span>.Data</span>
<span id="cb186-9"><a href="data-wrangling.html#cb186-9"></a>  b1000k&lt;-k[,,,<span class="dv">2</span>] <span class="co">#dim(b1000k) [1] 384 384  32</span></span>
<span id="cb186-10"><a href="data-wrangling.html#cb186-10"></a>  <span class="kw">writeNIfTI</span>(b1000k,<span class="st">&quot;b1000&quot;</span>)</span>
<span id="cb186-11"><a href="data-wrangling.html#cb186-11"></a>}</span></code></pre></div>
<p>Measurement of volume requires information on the dimensions of voxel.</p>
<div class="sourceCode" id="cb187"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb187-1"><a href="data-wrangling.html#cb187-1"></a><span class="co">#measure volume</span></span>
<span id="cb187-2"><a href="data-wrangling.html#cb187-2"></a>A=<span class="st">&quot;./Ext-Data/3000F_mca_blur.nii&quot;</span></span>
<span id="cb187-3"><a href="data-wrangling.html#cb187-3"></a></span>
<span id="cb187-4"><a href="data-wrangling.html#cb187-4"></a>VoxelDim&lt;-<span class="cf">function</span>(A){</span>
<span id="cb187-5"><a href="data-wrangling.html#cb187-5"></a>  <span class="kw">library</span>(oro.nifti)</span>
<span id="cb187-6"><a href="data-wrangling.html#cb187-6"></a>  img&lt;-<span class="kw">readNIfTI</span>(A,<span class="dt">reorient =</span> F)</span>
<span id="cb187-7"><a href="data-wrangling.html#cb187-7"></a>  VoxDim&lt;-<span class="kw">pixdim</span>(img)</span>
<span id="cb187-8"><a href="data-wrangling.html#cb187-8"></a>Volume&lt;-<span class="kw">sum</span>(img<span class="op">&gt;</span>.<span class="dv">5</span>)<span class="op">*</span>VoxDim[<span class="dv">2</span>]<span class="op">*</span>VoxDim[<span class="dv">3</span>]<span class="op">*</span>VoxDim[<span class="dv">4</span>]<span class="op">/</span><span class="dv">1000</span></span>
<span id="cb187-9"><a href="data-wrangling.html#cb187-9"></a>Volume</span>
<span id="cb187-10"><a href="data-wrangling.html#cb187-10"></a>}</span>
<span id="cb187-11"><a href="data-wrangling.html#cb187-11"></a></span>
<span id="cb187-12"><a href="data-wrangling.html#cb187-12"></a><span class="kw">VoxelDim</span>(A)</span></code></pre></div>
<pre><code>## Malformed NIfTI - not reading NIfTI extension, use at own risk!</code></pre>
<pre><code>## [1] 69</code></pre>
<p>Find unsigned angle between 2 vectors k and k1</p>
<div class="sourceCode" id="cb190"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb190-1"><a href="data-wrangling.html#cb190-1"></a>Morpho<span class="op">::</span><span class="kw">angle.calc</span>(k,k1)</span></code></pre></div>
<pre><code>## [1] 1.570771</code></pre>
<p>Determine centre of gravity of an object.</p>
<div class="sourceCode" id="cb192"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb192-1"><a href="data-wrangling.html#cb192-1"></a><span class="co">#centre of gravity</span></span>
<span id="cb192-2"><a href="data-wrangling.html#cb192-2"></a>neurobase<span class="op">::</span><span class="kw">cog</span>(img)</span></code></pre></div>
<pre><code>##     dim1     dim2     dim3 
## 65.74983 53.09204 46.67048</code></pre>
</div>
<div id="combining-arrays" class="section level3">
<h3><span class="header-section-number">3.7.3</span> Combining arrays</h3>
<p>This is an illustration of combining array using <em>cbind</em>.</p>
<div class="sourceCode" id="cb194"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb194-1"><a href="data-wrangling.html#cb194-1"></a><span class="co">#stack arrays</span></span>
<span id="cb194-2"><a href="data-wrangling.html#cb194-2"></a>k2&lt;-<span class="kw">cbind</span>(k,k1)</span>
<span id="cb194-3"><a href="data-wrangling.html#cb194-3"></a><span class="kw">dim</span>(k2) <span class="co">### [1] 902629      2</span></span></code></pre></div>
<pre><code>## [1] 902629      2</code></pre>
<p>The <em>abind</em> function produces a different array output. Later we will repeat the same exercise using <em>list</em> function.</p>
<div class="sourceCode" id="cb196"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb196-1"><a href="data-wrangling.html#cb196-1"></a><span class="co">#combine multi-dimensional arrays </span></span>
<span id="cb196-2"><a href="data-wrangling.html#cb196-2"></a>ab&lt;-abind<span class="op">::</span><span class="kw">abind</span>(k,k1)</span>
<span id="cb196-3"><a href="data-wrangling.html#cb196-3"></a><span class="kw">dim</span>(ab) <span class="co">### [1]  91 109 182</span></span></code></pre></div>
<pre><code>## [1]  91 109 182</code></pre>
<p>This example uses 25 files. Rather than open one file at a time create a list from pattern matching.</p>
<div class="sourceCode" id="cb198"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb198-1"><a href="data-wrangling.html#cb198-1"></a><span class="kw">library</span>(oro.nifti)</span>
<span id="cb198-2"><a href="data-wrangling.html#cb198-2"></a><span class="kw">library</span>(abind)</span>
<span id="cb198-3"><a href="data-wrangling.html#cb198-3"></a><span class="kw">library</span>(CHNOSZ) <span class="co"># for working with arrays</span></span></code></pre></div>
<pre><code>## CHNOSZ version 1.4.0 (2020-11-11)</code></pre>
<pre><code>## reset: creating &quot;thermo&quot; object</code></pre>
<pre><code>## OBIGT: loading default database with 1880 aqueous, 3418 total species</code></pre>
<pre><code>## 
## Attaching package: &#39;CHNOSZ&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:oro.nifti&#39;:
## 
##     slice</code></pre>
<pre><code>## The following object is masked from &#39;package:dplyr&#39;:
## 
##     slice</code></pre>
<div class="sourceCode" id="cb205"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb205-1"><a href="data-wrangling.html#cb205-1"></a><span class="kw">library</span>(RNiftyReg)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;RNiftyReg&#39;</code></pre>
<pre><code>## The following objects are masked from &#39;package:oro.nifti&#39;:
## 
##     pixdim, pixdim&lt;-</code></pre>
<div class="sourceCode" id="cb208"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb208-1"><a href="data-wrangling.html#cb208-1"></a><span class="co">#create a list using pattern matching</span></span>
<span id="cb208-2"><a href="data-wrangling.html#cb208-2"></a>mca.list&lt;-<span class="kw">list.files</span>(<span class="dt">path=</span><span class="st">&quot;./Ext-Data/&quot;</span>,<span class="dt">pattern =</span> <span class="st">&quot;*.nii&quot;</span>, <span class="dt">full.names =</span> <span class="ot">TRUE</span>)</span>
<span id="cb208-3"><a href="data-wrangling.html#cb208-3"></a></span>
<span id="cb208-4"><a href="data-wrangling.html#cb208-4"></a><span class="co">#length of list</span></span>
<span id="cb208-5"><a href="data-wrangling.html#cb208-5"></a><span class="kw">length</span>(mca.list)</span></code></pre></div>
<pre><code>## [1] 29</code></pre>
<div class="sourceCode" id="cb210"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb210-1"><a href="data-wrangling.html#cb210-1"></a><span class="co">#read multiple files using lapply function</span></span>
<span id="cb210-2"><a href="data-wrangling.html#cb210-2"></a></span>
<span id="cb210-3"><a href="data-wrangling.html#cb210-3"></a><span class="co">#use lappy to read in the nifti files</span></span>
<span id="cb210-4"><a href="data-wrangling.html#cb210-4"></a><span class="co">#note lapply returns a list</span></span>
<span id="cb210-5"><a href="data-wrangling.html#cb210-5"></a>mca.list.nii &lt;-<span class="st"> </span><span class="kw">lapply</span>(mca.list, readNIfTI)</span></code></pre></div>
<pre><code>## Malformed NIfTI - not reading NIfTI extension, use at own risk!</code></pre>
<pre><code>## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!
## Malformed NIfTI - not reading NIfTI extension, use at own risk!</code></pre>
<div class="sourceCode" id="cb213"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb213-1"><a href="data-wrangling.html#cb213-1"></a><span class="kw">class</span>(mca.list.nii)</span></code></pre></div>
<pre><code>## [1] &quot;list&quot;</code></pre>
<p>This example illustrates how to view the first image in the list.</p>
<div class="sourceCode" id="cb215"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb215-1"><a href="data-wrangling.html#cb215-1"></a><span class="co">#view each image in the list</span></span>
<span id="cb215-2"><a href="data-wrangling.html#cb215-2"></a>neurobase<span class="op">::</span><span class="kw">ortho2</span>(mca.list.nii[[<span class="dv">1</span>]])</span></code></pre></div>
<p><img src="Applications-of-R-in-Healthcare_files/figure-html/02-Data-Wrangling-28-2-1-1.png" width="672" /></p>
<p>In this example, the first 3 segmented images from the list are averaged and viewed.</p>
<div class="sourceCode" id="cb216"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb216-1"><a href="data-wrangling.html#cb216-1"></a><span class="co">#view average image</span></span>
<span id="cb216-2"><a href="data-wrangling.html#cb216-2"></a>mca_ave3&lt;-(mca.list.nii[[<span class="dv">5</span>]]<span class="op">+</span>mca.list.nii[[<span class="dv">6</span>]]<span class="op">+</span>mca.list.nii[[<span class="dv">7</span>]])<span class="op">/</span><span class="dv">3</span></span>
<span id="cb216-3"><a href="data-wrangling.html#cb216-3"></a>neurobase<span class="op">::</span><span class="kw">ortho2</span>(mca_ave3)</span></code></pre></div>
<p><img src="Applications-of-R-in-Healthcare_files/figure-html/02-Data-Wrangling-28-2-2-1.png" width="672" /></p>
<p>The output below is the same as above but is performed on arrays. The <em>image_data</em> function from <em>oro.nifti</em> library extracts the image attribute from the slot .Data.</p>
<div class="sourceCode" id="cb217"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb217-1"><a href="data-wrangling.html#cb217-1"></a><span class="co">#extract multiple arrays using lapply</span></span>
<span id="cb217-2"><a href="data-wrangling.html#cb217-2"></a>mca.list.array&lt;-<span class="kw">lapply</span>(mca.list.nii, img_data)</span>
<span id="cb217-3"><a href="data-wrangling.html#cb217-3"></a></span>
<span id="cb217-4"><a href="data-wrangling.html#cb217-4"></a>m3&lt;-(mca.list.array[[<span class="dv">5</span>]]<span class="op">+</span>mca.list.array[[<span class="dv">6</span>]]<span class="op">+</span>mca.list.array[[<span class="dv">7</span>]])<span class="op">/</span><span class="dv">3</span></span>
<span id="cb217-5"><a href="data-wrangling.html#cb217-5"></a></span>
<span id="cb217-6"><a href="data-wrangling.html#cb217-6"></a><span class="co">#compare this with the output from above</span></span>
<span id="cb217-7"><a href="data-wrangling.html#cb217-7"></a>neurobase<span class="op">::</span><span class="kw">double_ortho</span>(m3, mca_ave3)</span></code></pre></div>
<p><img src="Applications-of-R-in-Healthcare_files/figure-html/02-Data-Wrangling-28-2-3-1.png" width="672" /></p>
<p>Arrays can be extracted from list using <em>list2array</em> function.</p>
<div class="sourceCode" id="cb218"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb218-1"><a href="data-wrangling.html#cb218-1"></a><span class="kw">class</span>(mca.list.array)</span></code></pre></div>
<pre><code>## [1] &quot;list&quot;</code></pre>
<div class="sourceCode" id="cb220"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb220-1"><a href="data-wrangling.html#cb220-1"></a><span class="co">#convert list to array</span></span>
<span id="cb220-2"><a href="data-wrangling.html#cb220-2"></a><span class="co">#CHNOSZ function</span></span>
<span id="cb220-3"><a href="data-wrangling.html#cb220-3"></a>m.listarray&lt;-<span class="kw">list2array</span>(mca.list.array)<span class="co">#91 109  91  25</span></span>
<span id="cb220-4"><a href="data-wrangling.html#cb220-4"></a></span>
<span id="cb220-5"><a href="data-wrangling.html#cb220-5"></a><span class="kw">class</span>(m.listarray)</span></code></pre></div>
<pre><code>## [1] &quot;array&quot;</code></pre>
</div>
<div id="math-operation-on-multidimensional-array" class="section level3">
<h3><span class="header-section-number">3.7.4</span> Math operation on multidimensional array</h3>
<p>Here we use <em>apply</em> function to average over every element of the multidimensional array. The first argument of <em>apply</em> is the array, the second argument is the margin or the component for analysis and the last argument is the function. If the idea is to analyse the row then the margin argument is <em>c(1)</em>, column then the margin argumen is <em>c(2)</em> and so on.</p>
<div class="sourceCode" id="cb222"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb222-1"><a href="data-wrangling.html#cb222-1"></a>ma&lt;-<span class="kw">apply</span>(m.listarray,<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), mean)</span>
<span id="cb222-2"><a href="data-wrangling.html#cb222-2"></a>neurobase<span class="op">::</span><span class="kw">ortho2</span>(ma)</span></code></pre></div>
<p><img src="Applications-of-R-in-Healthcare_files/figure-html/02-Data-Wrangling-28-3-1.png" width="672" /></p>
<p>Thresholding can be performed using <em>mask_img</em> function. This function can also be used to create a mask.</p>
<div class="sourceCode" id="cb223"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb223-1"><a href="data-wrangling.html#cb223-1"></a>ma_mask=neurobase<span class="op">::</span><span class="kw">mask_img</span>(ma, ma<span class="op">&gt;</span><span class="fl">0.1</span>)</span>
<span id="cb223-2"><a href="data-wrangling.html#cb223-2"></a></span>
<span id="cb223-3"><a href="data-wrangling.html#cb223-3"></a>neurobase<span class="op">::</span><span class="kw">double_ortho</span>(ma_mask, ma)</span></code></pre></div>
<p><img src="Applications-of-R-in-Healthcare_files/figure-html/02-Data-Wrangling-28-3-1-1.png" width="672" /></p>
</div>
<div id="math-operation-on-list" class="section level3">
<h3><span class="header-section-number">3.7.5</span> Math operation on list</h3>
<p>In this example we us <em>lapply</em> to a function within a list. This is an example of a functional or a function which takes a function as an input and return a vector as an output. In this example, a functional operates on one element of the list at a time. This example of functional is the same as the function <em>VoxelDim</em> describes above.</p>
<div class="sourceCode" id="cb224"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb224-1"><a href="data-wrangling.html#cb224-1"></a>vox=<span class="kw">unlist</span>(<span class="kw">lapply</span>(mca.list.nii, </span>
<span id="cb224-2"><a href="data-wrangling.html#cb224-2"></a>           <span class="cf">function</span>(A) <span class="kw">sum</span>(A<span class="op">&gt;</span>.<span class="dv">5</span>)<span class="op">*</span></span>
<span id="cb224-3"><a href="data-wrangling.html#cb224-3"></a><span class="st">             </span><span class="co">#obtain voxel dimensions</span></span>
<span id="cb224-4"><a href="data-wrangling.html#cb224-4"></a><span class="st">             </span>oro.nifti<span class="op">::</span><span class="kw">pixdim</span>(A)[<span class="dv">2</span>]<span class="op">*</span></span>
<span id="cb224-5"><a href="data-wrangling.html#cb224-5"></a><span class="st">             </span>oro.nifti<span class="op">::</span><span class="kw">pixdim</span>(A)[<span class="dv">3</span>]<span class="op">*</span></span>
<span id="cb224-6"><a href="data-wrangling.html#cb224-6"></a><span class="st">             </span>oro.nifti<span class="op">::</span><span class="kw">pixdim</span>(A)[<span class="dv">4</span>]<span class="op">/</span><span class="dv">1000</span></span>
<span id="cb224-7"><a href="data-wrangling.html#cb224-7"></a>           ))</span>
<span id="cb224-8"><a href="data-wrangling.html#cb224-8"></a></span>
<span id="cb224-9"><a href="data-wrangling.html#cb224-9"></a>vox</span></code></pre></div>
<pre><code>##  [1]   0.000   0.000   0.000   0.000  69.000   3.624   8.768  58.200 326.760
## [10]   1.608   0.000   3.904   3.328  66.776 106.976  70.928  26.912  11.768
## [19]  17.592  14.720 151.872  87.528   0.000  24.192  80.248 152.792  69.552
## [28]  25.576   0.000</code></pre>
</div>
<div id="vectorising-nifti-object" class="section level3">
<h3><span class="header-section-number">3.7.6</span> Vectorising nifti object</h3>
<p>One way of handling imaging data for analysis is to flatten the image, then create an empty array of the same size to return the image.</p>
<div class="sourceCode" id="cb226"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb226-1"><a href="data-wrangling.html#cb226-1"></a><span class="co">#flatten 3D</span></span>
<span id="cb226-2"><a href="data-wrangling.html#cb226-2"></a>niivector =<span class="st"> </span><span class="kw">as.vector</span>(img[,,]) <span class="co">#902629</span></span>
<span id="cb226-3"><a href="data-wrangling.html#cb226-3"></a></span>
<span id="cb226-4"><a href="data-wrangling.html#cb226-4"></a><span class="co">#Create empty array of same size to fill up</span></span>
<span id="cb226-5"><a href="data-wrangling.html#cb226-5"></a>niinew =<span class="st"> </span><span class="kw">array</span>(<span class="dv">0</span>, <span class="dt">dim=</span><span class="kw">dim</span>(img))</span>
<span id="cb226-6"><a href="data-wrangling.html#cb226-6"></a></span>
<span id="cb226-7"><a href="data-wrangling.html#cb226-7"></a><span class="co">#return to 3D</span></span>
<span id="cb226-8"><a href="data-wrangling.html#cb226-8"></a>niinew =<span class="st"> </span><span class="kw">array</span>(niivector, <span class="dt">dim=</span><span class="kw">dim</span>(img))</span>
<span id="cb226-9"><a href="data-wrangling.html#cb226-9"></a></span>
<span id="cb226-10"><a href="data-wrangling.html#cb226-10"></a><span class="co">#confirm</span></span>
<span id="cb226-11"><a href="data-wrangling.html#cb226-11"></a>neurobase<span class="op">::</span><span class="kw">ortho2</span>(niinew)</span></code></pre></div>
<p><img src="Applications-of-R-in-Healthcare_files/figure-html/02-Data-Wrangling-28-5-1.png" width="672" /></p>
<p>Another way of creating vector from the image is to use <em>c</em> function.</p>
<div class="sourceCode" id="cb227"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb227-1"><a href="data-wrangling.html#cb227-1"></a><span class="co">#vector can also be created using c</span></span>
<span id="cb227-2"><a href="data-wrangling.html#cb227-2"></a>niivector2 =<span class="st"> </span><span class="kw">c</span>(img[,,]) <span class="co">#902629</span></span></code></pre></div>
</div>
<div id="tar-file" class="section level3">
<h3><span class="header-section-number">3.7.7</span> tar file</h3>
<p>Image files can be large and are often stored as tar files. The <em>tar</em>
(tgz file), <em>untar</em>, <em>zip</em> (gz file) and <em>unzip</em> function are from the <em>utils</em> library.</p>
<div class="sourceCode" id="cb228"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb228-1"><a href="data-wrangling.html#cb228-1"></a>colin_1mm&lt;-<span class="kw">untar</span>(<span class="st">&quot;./Data-Use/colin_1mm.tgz&quot;</span>)</span>
<span id="cb228-2"><a href="data-wrangling.html#cb228-2"></a>colinIm&lt;-<span class="kw">readNIfTI</span>(<span class="st">&quot;colin_1mm&quot;</span>) <span class="co">#1 x 1 x 1</span></span>
<span id="cb228-3"><a href="data-wrangling.html#cb228-3"></a><span class="kw">class</span>(colinIm)</span></code></pre></div>
<pre><code>## [1] &quot;nifti&quot;
## attr(,&quot;package&quot;)
## [1] &quot;oro.nifti&quot;</code></pre>
<div class="sourceCode" id="cb230"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb230-1"><a href="data-wrangling.html#cb230-1"></a>neurobase<span class="op">::</span><span class="kw">ortho2</span>(colinIm)</span></code></pre></div>
<p><img src="Applications-of-R-in-Healthcare_files/figure-html/02-Data-Wrangling-28-6-1.png" width="672" /></p>
<p>The <em>readNIfTI</em> call can open gz file without the need to call <em>unzip</em> function.</p>
<div class="sourceCode" id="cb231"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb231-1"><a href="data-wrangling.html#cb231-1"></a><span class="kw">library</span>(RNiftyReg)</span>
<span id="cb231-2"><a href="data-wrangling.html#cb231-2"></a>epi_t2&lt;-<span class="st"> </span><span class="kw">readNIfTI</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;epi_t2.nii.gz&quot;</span>, <span class="dt">package=</span><span class="st">&quot;RNiftyReg&quot;</span>))</span>
<span id="cb231-3"><a href="data-wrangling.html#cb231-3"></a><span class="kw">class</span>(epi_t2)</span></code></pre></div>
<pre><code>## [1] &quot;nifti&quot;
## attr(,&quot;package&quot;)
## [1] &quot;oro.nifti&quot;</code></pre>
<div class="sourceCode" id="cb233"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb233-1"><a href="data-wrangling.html#cb233-1"></a>neurobase<span class="op">::</span><span class="kw">ortho2</span>(epi_t2)</span></code></pre></div>
<p><img src="Applications-of-R-in-Healthcare_files/figure-html/02-Data-Wrangling-28-6-1-1.png" width="672" /></p>
</div>
<div id="image-registration" class="section level3">
<h3><span class="header-section-number">3.7.8</span> Image registration</h3>
<p>There are many different libraries for performing registration.</p>
<div class="sourceCode" id="cb234"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb234-1"><a href="data-wrangling.html#cb234-1"></a><span class="kw">library</span>(RNiftyReg)</span>
<span id="cb234-2"><a href="data-wrangling.html#cb234-2"></a></span>
<span id="cb234-3"><a href="data-wrangling.html#cb234-3"></a><span class="co">#example from data</span></span>
<span id="cb234-4"><a href="data-wrangling.html#cb234-4"></a>source&lt;-<span class="kw">readNifti</span>(<span class="st">&quot;./Data-Use/mca10_pca10_border10.nii.gz&quot;</span>)</span>
<span id="cb234-5"><a href="data-wrangling.html#cb234-5"></a><span class="kw">pixdim</span>(source)</span></code></pre></div>
<pre><code>## [1] 2 2 2</code></pre>
<div class="sourceCode" id="cb236"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb236-1"><a href="data-wrangling.html#cb236-1"></a>colin_1mm&lt;-<span class="kw">untar</span>(<span class="st">&quot;./Data-Use/colin_1mm.tgz&quot;</span>)</span>
<span id="cb236-2"><a href="data-wrangling.html#cb236-2"></a>target&lt;-<span class="kw">readNifti</span>(<span class="st">&quot;colin_1mm&quot;</span>)</span>
<span id="cb236-3"><a href="data-wrangling.html#cb236-3"></a><span class="kw">pixdim</span>(target)</span></code></pre></div>
<pre><code>## [1] 1 1 1 1</code></pre>
<div class="sourceCode" id="cb238"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb238-1"><a href="data-wrangling.html#cb238-1"></a>target</span></code></pre></div>
<pre><code>## Image array of mode &quot;double&quot; (54.2 Mb)
## - 181 x 217 x 181 x 1 voxels
## - 1 x 1 x 1 mm x 1 s per voxel</code></pre>
<div class="sourceCode" id="cb240"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb240-1"><a href="data-wrangling.html#cb240-1"></a><span class="co">#register source to target </span></span>
<span id="cb240-2"><a href="data-wrangling.html#cb240-2"></a>result &lt;-<span class="st"> </span><span class="kw">niftyreg</span>(source, target)</span>
<span id="cb240-3"><a href="data-wrangling.html#cb240-3"></a><span class="co">#affine transformation</span></span>
<span id="cb240-4"><a href="data-wrangling.html#cb240-4"></a>result<span class="op">$</span>forwardTransforms</span></code></pre></div>
<pre><code>## [[1]]
## NiftyReg affine matrix:
##   0.77338767    0.56943178    0.47119385    4.53472805
##  -0.59504575    0.72087985   -0.80796480  -19.90197945
##  -0.56219202   -0.06933179    0.55241060   21.51253319
##   0.00000000    0.00000000    0.00000000    1.00000000</code></pre>
<div class="sourceCode" id="cb242"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb242-1"><a href="data-wrangling.html#cb242-1"></a><span class="co">#image in target space</span></span>
<span id="cb242-2"><a href="data-wrangling.html#cb242-2"></a>result<span class="op">$</span>image</span></code></pre></div>
<pre><code>## Image array of mode &quot;double&quot; (54.2 Mb)
## - 181 x 217 x 181 voxels
## - 1 x 1 x 1 mm per voxel</code></pre>
<div class="sourceCode" id="cb244"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb244-1"><a href="data-wrangling.html#cb244-1"></a>neurobase<span class="op">::</span><span class="kw">ortho2</span>(result<span class="op">$</span>image)</span></code></pre></div>
<p><img src="Applications-of-R-in-Healthcare_files/figure-html/02-Data-Wrangling-29-1.png" width="672" /></p>
<p>Output from <em>RNiftyReg</em> are <em>niftiImage</em> object. They can be converted to <em>oro.nifti</em> object using <em>nii2oro</em> function.</p>
<div class="sourceCode" id="cb245"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb245-1"><a href="data-wrangling.html#cb245-1"></a>otarget&lt;-<span class="kw">nii2oro</span>(target)</span>
<span id="cb245-2"><a href="data-wrangling.html#cb245-2"></a>oimage&lt;-<span class="kw">nii2oro</span>(result<span class="op">$</span>image)</span>
<span id="cb245-3"><a href="data-wrangling.html#cb245-3"></a></span>
<span id="cb245-4"><a href="data-wrangling.html#cb245-4"></a><span class="kw">overlay</span>(otarget, <span class="dt">y=</span>oimage,<span class="dt">z =</span> <span class="dv">90</span>, <span class="dt">plot.type =</span> <span class="st">&quot;single&quot;</span> )</span></code></pre></div>
<p><img src="Applications-of-R-in-Healthcare_files/figure-html/02-Data-Wrangling-29-1-1.png" width="672" /></p>
</div>
<div id="rescaling" class="section level3">
<h3><span class="header-section-number">3.7.9</span> Rescaling</h3>
<p>Perform affine registration and resampling of image using the transformation file.</p>
<div class="sourceCode" id="cb246"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb246-1"><a href="data-wrangling.html#cb246-1"></a><span class="co">#affine</span></span>
<span id="cb246-2"><a href="data-wrangling.html#cb246-2"></a>ica1000&lt;-<span class="kw">readNifti</span>(<span class="st">&quot;./Ext-Data/1000M_ica.nii&quot;</span>)</span>
<span id="cb246-3"><a href="data-wrangling.html#cb246-3"></a>ica1000</span></code></pre></div>
<pre><code>## Image array of mode &quot;double&quot; (6.9 Mb)
## - 91 x 109 x 91 voxels
## - 2 x 2 x 2 mm per voxel</code></pre>
<div class="sourceCode" id="cb248"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb248-1"><a href="data-wrangling.html#cb248-1"></a>colin_affine&lt;-<span class="kw">buildAffine</span>(<span class="dt">source=</span>ica1000, <span class="dt">target=</span>target)</span>
<span id="cb248-2"><a href="data-wrangling.html#cb248-2"></a><span class="co">#apply transformation from above</span></span>
<span id="cb248-3"><a href="data-wrangling.html#cb248-3"></a><span class="co">#assume 1000M_ica.nii and source are in the same space</span></span>
<span id="cb248-4"><a href="data-wrangling.html#cb248-4"></a>colin_like&lt;-<span class="kw">applyTransform</span>(colin_affine,ica1000)</span>
<span id="cb248-5"><a href="data-wrangling.html#cb248-5"></a>neurobase<span class="op">::</span><span class="kw">ortho2</span>(colin_like)</span></code></pre></div>
<p><img src="Applications-of-R-in-Healthcare_files/figure-html/02-Data-Wrangling-30-1.png" width="672" /></p>
<p>Resampling an image to different dimensions. This example is different from above in which rescaling is performed as part of registration to higher resolution image. Here the <em>rescale</em> function from <em>RNiftyReg</em> library is used to change the dimensions from 1x1x1 mm to 2x2x2 mm.</p>
<div class="sourceCode" id="cb249"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb249-1"><a href="data-wrangling.html#cb249-1"></a>ica1000.rescale&lt;-RNiftyReg<span class="op">::</span><span class="kw">rescale</span>(ica1000,<span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>))</span>
<span id="cb249-2"><a href="data-wrangling.html#cb249-2"></a></span>
<span id="cb249-3"><a href="data-wrangling.html#cb249-3"></a>ica1000</span></code></pre></div>
<pre><code>## Image array of mode &quot;double&quot; (6.9 Mb)
## - 91 x 109 x 91 voxels
## - 2 x 2 x 2 mm per voxel</code></pre>
<div class="sourceCode" id="cb251"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb251-1"><a href="data-wrangling.html#cb251-1"></a><span class="co">#compare with rescale</span></span>
<span id="cb251-2"><a href="data-wrangling.html#cb251-2"></a></span>
<span id="cb251-3"><a href="data-wrangling.html#cb251-3"></a>ica1000.rescale</span></code></pre></div>
<pre><code>## Image array of mode &quot;double&quot; (55.1 Mb)
## - 182 x 218 x 182 voxels
## - 1 x 1 x 1 mm per voxel</code></pre>
</div>
<div id="mni-template" class="section level3">
<h3><span class="header-section-number">3.7.10</span> MNI template</h3>
<p>There are several different MRI templates. The well known one is the MNI 152 template <span class="citation">(Mazziotta J <a href="#ref-pmid11545704" role="doc-biblioref">2001</a>)</span>. This was developed from male right-handed medical students.</p>
<div id="aal-atlas" class="section level4">
<h4><span class="header-section-number">3.7.10.1</span> AAL atlas</h4>
<p>The automated anatomical labeling (AAL) atlas contains 45 volume of interest <span class="citation">(Tzourio-Mazoyer N <a href="#ref-pmid11771995" role="doc-biblioref">2002</a>)</span>. The atlas is aligned to MNI 152 coordinates.</p>
<div class="sourceCode" id="cb253"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb253-1"><a href="data-wrangling.html#cb253-1"></a><span class="kw">library</span>(rgl)</span>
<span id="cb253-2"><a href="data-wrangling.html#cb253-2"></a><span class="kw">library</span>(misc3d)</span>
<span id="cb253-3"><a href="data-wrangling.html#cb253-3"></a><span class="kw">library</span>(MNITemplate)</span>
<span id="cb253-4"><a href="data-wrangling.html#cb253-4"></a><span class="kw">library</span>(aal)</span>
<span id="cb253-5"><a href="data-wrangling.html#cb253-5"></a><span class="kw">library</span>(neurobase)</span>
<span id="cb253-6"><a href="data-wrangling.html#cb253-6"></a><span class="kw">library</span>(magick)</span></code></pre></div>
<pre><code>## Linking to ImageMagick 6.9.11.34
## Enabled features: cairo, freetype, fftw, ghostscript, lcms, pango, rsvg, webp
## Disabled features: fontconfig, x11</code></pre>
<div class="sourceCode" id="cb255"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb255-1"><a href="data-wrangling.html#cb255-1"></a><span class="kw">library</span>(oro.nifti)</span>
<span id="cb255-2"><a href="data-wrangling.html#cb255-2"></a></span>
<span id="cb255-3"><a href="data-wrangling.html#cb255-3"></a>img =<span class="st"> </span><span class="kw">aal_image</span>()</span>
<span id="cb255-4"><a href="data-wrangling.html#cb255-4"></a>template =<span class="st"> </span><span class="kw">readMNI</span>(<span class="dt">res =</span> <span class="st">&quot;2mm&quot;</span>)</span>
<span id="cb255-5"><a href="data-wrangling.html#cb255-5"></a>cut &lt;-<span class="st"> </span><span class="dv">4500</span></span>
<span id="cb255-6"><a href="data-wrangling.html#cb255-6"></a>dtemp &lt;-<span class="st"> </span><span class="kw">dim</span>(template)</span>
<span id="cb255-7"><a href="data-wrangling.html#cb255-7"></a></span>
<span id="cb255-8"><a href="data-wrangling.html#cb255-8"></a><span class="co"># All of the sections you can label</span></span>
<span id="cb255-9"><a href="data-wrangling.html#cb255-9"></a>labs =<span class="st"> </span><span class="kw">aal_get_labels</span>()</span>
<span id="cb255-10"><a href="data-wrangling.html#cb255-10"></a></span>
<span id="cb255-11"><a href="data-wrangling.html#cb255-11"></a><span class="co"># highlight - in this case the Cingulate_Post_L</span></span>
<span id="cb255-12"><a href="data-wrangling.html#cb255-12"></a>cingulate =<span class="st"> </span>labs<span class="op">$</span>index[<span class="kw">grep</span>(<span class="st">&quot;Cingulate_Post_R&quot;</span>, labs<span class="op">$</span>name)]</span>
<span id="cb255-13"><a href="data-wrangling.html#cb255-13"></a></span>
<span id="cb255-14"><a href="data-wrangling.html#cb255-14"></a><span class="co">#mask of object for rendering</span></span>
<span id="cb255-15"><a href="data-wrangling.html#cb255-15"></a>mask =<span class="st"> </span><span class="kw">remake_img</span>(<span class="dt">vec =</span> img <span class="op">%in%</span><span class="st"> </span>cingulate, <span class="dt">img =</span> img)</span>
<span id="cb255-16"><a href="data-wrangling.html#cb255-16"></a></span>
<span id="cb255-17"><a href="data-wrangling.html#cb255-17"></a><span class="co">#contour for MNI template</span></span>
<span id="cb255-18"><a href="data-wrangling.html#cb255-18"></a><span class="kw">contour3d</span>(template, <span class="dt">x=</span><span class="dv">1</span><span class="op">:</span>dtemp[<span class="dv">1</span>], <span class="dt">y=</span><span class="dv">1</span><span class="op">:</span>dtemp[<span class="dv">2</span>], <span class="dt">z=</span><span class="dv">1</span><span class="op">:</span>dtemp[<span class="dv">3</span>], <span class="dt">level =</span> cut, <span class="dt">alpha =</span> <span class="fl">0.1</span>, <span class="dt">draw =</span> <span class="ot">TRUE</span>)</span>
<span id="cb255-19"><a href="data-wrangling.html#cb255-19"></a></span>
<span id="cb255-20"><a href="data-wrangling.html#cb255-20"></a><span class="co">#contour for mask</span></span>
<span id="cb255-21"><a href="data-wrangling.html#cb255-21"></a><span class="kw">contour3d</span>(mask, <span class="dt">level =</span> <span class="kw">c</span>(<span class="fl">0.5</span>), <span class="dt">alpha =</span> <span class="kw">c</span>(<span class="fl">0.5</span>), <span class="dt">add =</span> <span class="ot">TRUE</span>, <span class="dt">color=</span><span class="kw">c</span>(<span class="st">&quot;red&quot;</span>) )</span>
<span id="cb255-22"><a href="data-wrangling.html#cb255-22"></a></span>
<span id="cb255-23"><a href="data-wrangling.html#cb255-23"></a><span class="co">### add text</span></span>
<span id="cb255-24"><a href="data-wrangling.html#cb255-24"></a><span class="kw">text3d</span>(<span class="dt">x=</span>dtemp[<span class="dv">1</span>]<span class="op">/</span><span class="dv">2</span>, <span class="dt">y=</span>dtemp[<span class="dv">2</span>]<span class="op">/</span><span class="dv">2</span>, <span class="dt">z =</span> dtemp[<span class="dv">3</span>]<span class="op">*</span><span class="fl">0.98</span>, <span class="dt">text=</span><span class="st">&quot;Top&quot;</span>)</span>
<span id="cb255-25"><a href="data-wrangling.html#cb255-25"></a><span class="kw">text3d</span>(<span class="dt">x=</span><span class="op">-</span><span class="fl">0.98</span>, <span class="dt">y=</span>dtemp[<span class="dv">2</span>]<span class="op">/</span><span class="dv">2</span>, <span class="dt">z =</span> dtemp[<span class="dv">3</span>]<span class="op">/</span><span class="dv">2</span>, <span class="dt">text=</span><span class="st">&quot;Right&quot;</span>)</span>
<span id="cb255-26"><a href="data-wrangling.html#cb255-26"></a></span>
<span id="cb255-27"><a href="data-wrangling.html#cb255-27"></a><span class="co">#create movie</span></span>
<span id="cb255-28"><a href="data-wrangling.html#cb255-28"></a><span class="co">#movie file is saved to temporary folder</span></span>
<span id="cb255-29"><a href="data-wrangling.html#cb255-29"></a><span class="co">#movie3d(spin3d(),duration=30)</span></span>
<span id="cb255-30"><a href="data-wrangling.html#cb255-30"></a></span>
<span id="cb255-31"><a href="data-wrangling.html#cb255-31"></a></span>
<span id="cb255-32"><a href="data-wrangling.html#cb255-32"></a><span class="co">#add digital map of mca territory</span></span>
<span id="cb255-33"><a href="data-wrangling.html#cb255-33"></a>MCA&lt;-<span class="kw">readNIfTI</span>(<span class="st">&quot;./Data-Use/MCA_average28_MAP_100.nii&quot;</span>)</span></code></pre></div>
<pre><code>## Malformed NIfTI - not reading NIfTI extension, use at own risk!</code></pre>
<div class="sourceCode" id="cb257"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb257-1"><a href="data-wrangling.html#cb257-1"></a><span class="co">#check range </span></span>
<span id="cb257-2"><a href="data-wrangling.html#cb257-2"></a><span class="kw">range</span>(MCA)</span></code></pre></div>
<pre><code>## [1] -2.531219e-05  2.320454e+01</code></pre>
<div class="sourceCode" id="cb259"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb259-1"><a href="data-wrangling.html#cb259-1"></a><span class="co">#mask2 = remake_img(vec = img %in% source, img = img)</span></span>
<span id="cb259-2"><a href="data-wrangling.html#cb259-2"></a><span class="kw">contour3d</span>(template, <span class="dt">x=</span><span class="dv">1</span><span class="op">:</span>dtemp[<span class="dv">1</span>], <span class="dt">y=</span><span class="dv">1</span><span class="op">:</span>dtemp[<span class="dv">2</span>], <span class="dt">z=</span><span class="dv">1</span><span class="op">:</span>dtemp[<span class="dv">3</span>], <span class="dt">level =</span> cut, <span class="dt">alpha =</span> <span class="fl">0.1</span>, <span class="dt">draw =</span> <span class="ot">TRUE</span>)</span>
<span id="cb259-3"><a href="data-wrangling.html#cb259-3"></a></span>
<span id="cb259-4"><a href="data-wrangling.html#cb259-4"></a><span class="co">#contour for mask</span></span>
<span id="cb259-5"><a href="data-wrangling.html#cb259-5"></a><span class="kw">contour3d</span>(MCA, <span class="dt">level =</span> <span class="kw">c</span>(<span class="fl">0.5</span>), <span class="dt">alpha =</span> <span class="kw">c</span>(<span class="fl">0.5</span>), <span class="dt">add =</span> <span class="ot">TRUE</span>, <span class="dt">color=</span><span class="kw">c</span>(<span class="st">&quot;Yellow&quot;</span>))</span>
<span id="cb259-6"><a href="data-wrangling.html#cb259-6"></a><span class="co">#add frontal</span></span>
<span id="cb259-7"><a href="data-wrangling.html#cb259-7"></a>angular =<span class="st"> </span>labs<span class="op">$</span>index[<span class="kw">grep</span>(<span class="st">&quot;Angular_R&quot;</span>, labs<span class="op">$</span>name)]</span>
<span id="cb259-8"><a href="data-wrangling.html#cb259-8"></a><span class="co">#mask of object for rendering</span></span>
<span id="cb259-9"><a href="data-wrangling.html#cb259-9"></a>mask2 =<span class="st"> </span><span class="kw">remake_img</span>(<span class="dt">vec =</span> img <span class="op">%in%</span><span class="st"> </span>angular, <span class="dt">img =</span> img)</span>
<span id="cb259-10"><a href="data-wrangling.html#cb259-10"></a><span class="kw">contour3d</span>(mask2, <span class="dt">level =</span> <span class="kw">c</span>(<span class="fl">0.5</span>), <span class="dt">alpha =</span> <span class="kw">c</span>(<span class="fl">0.5</span>), <span class="dt">add =</span> <span class="ot">TRUE</span>, <span class="dt">color=</span><span class="kw">c</span>(<span class="st">&quot;red&quot;</span>))</span>
<span id="cb259-11"><a href="data-wrangling.html#cb259-11"></a><span class="co">#add cingulate</span></span>
<span id="cb259-12"><a href="data-wrangling.html#cb259-12"></a><span class="kw">contour3d</span>(mask, <span class="dt">level =</span> <span class="kw">c</span>(<span class="fl">0.5</span>), <span class="dt">alpha =</span> <span class="kw">c</span>(<span class="fl">0.5</span>), <span class="dt">add =</span> <span class="ot">TRUE</span>, <span class="dt">color=</span><span class="kw">c</span>(<span class="st">&quot;blue&quot;</span>))</span>
<span id="cb259-13"><a href="data-wrangling.html#cb259-13"></a><span class="co">### add text</span></span>
<span id="cb259-14"><a href="data-wrangling.html#cb259-14"></a><span class="kw">text3d</span>(<span class="dt">x=</span>dtemp[<span class="dv">1</span>]<span class="op">/</span><span class="dv">2</span>, <span class="dt">y=</span>dtemp[<span class="dv">2</span>]<span class="op">/</span><span class="dv">2</span>, <span class="dt">z =</span> dtemp[<span class="dv">3</span>]<span class="op">*</span><span class="fl">0.98</span>, <span class="dt">text=</span><span class="st">&quot;Top&quot;</span>)</span>
<span id="cb259-15"><a href="data-wrangling.html#cb259-15"></a><span class="kw">text3d</span>(<span class="dt">x=</span><span class="op">-</span><span class="fl">0.98</span>, <span class="dt">y=</span>dtemp[<span class="dv">2</span>]<span class="op">/</span><span class="dv">2</span>, <span class="dt">z =</span> dtemp[<span class="dv">3</span>]<span class="op">/</span><span class="dv">2</span>, <span class="dt">text=</span><span class="st">&quot;Right&quot;</span>)</span>
<span id="cb259-16"><a href="data-wrangling.html#cb259-16"></a></span>
<span id="cb259-17"><a href="data-wrangling.html#cb259-17"></a><span class="co">#create movie</span></span>
<span id="cb259-18"><a href="data-wrangling.html#cb259-18"></a><span class="co">#movie file is saved to temporary folder</span></span>
<span id="cb259-19"><a href="data-wrangling.html#cb259-19"></a><span class="co">#movie3d(spin3d(),duration=5)</span></span>
<span id="cb259-20"><a href="data-wrangling.html#cb259-20"></a><span class="co">#rglwidget()</span></span></code></pre></div>
</div>
<div id="eve-template" class="section level4">
<h4><span class="header-section-number">3.7.10.2</span> Eve template</h4>
<p>The Eve template is from John Hopkins <span class="citation">(“Atlas-Based Whole Brain White Matter Analysis Using Large Deformation Diffeomorphic Metric Mapping: Application to Normal Elderly and Alzheimer’s Disease Participants.” <a href="#ref-pmid19385016" role="doc-biblioref">2009</a>)</span>. It is a single subject high resolution white matter atlas that has been morphed into MNI 152 coordinates. The atlas is parcellated into 176 regions based on ICBM-DTI-81 atlas.</p>
<div class="sourceCode" id="cb260"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb260-1"><a href="data-wrangling.html#cb260-1"></a><span class="kw">library</span>(EveTemplate)</span>
<span id="cb260-2"><a href="data-wrangling.html#cb260-2"></a></span>
<span id="cb260-3"><a href="data-wrangling.html#cb260-3"></a>eve_labels =<span class="st"> </span><span class="kw">readEveMap</span>(<span class="dt">type =</span> <span class="st">&quot;II&quot;</span>)</span>
<span id="cb260-4"><a href="data-wrangling.html#cb260-4"></a>eve_labels</span></code></pre></div>
<pre><code>## NIfTI-1 format
##   Type            : nifti
##   Data Type       : 4 (INT16)
##   Bits per Pixel  : 16
##   Slice Code      : 0 (Unknown)
##   Intent Code     : 0 (None)
##   Qform Code      : 2 (Aligned_Anat)
##   Sform Code      : 1 (Scanner_Anat)
##   Dimension       : 181 x 217 x 181
##   Pixel Dimension : 1 x 1 x 1
##   Voxel Units     : mm
##   Time Units      : Unknown</code></pre>
<div class="sourceCode" id="cb262"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb262-1"><a href="data-wrangling.html#cb262-1"></a>neurobase<span class="op">::</span><span class="kw">ortho2</span>(eve_labels)</span></code></pre></div>
<p><img src="Applications-of-R-in-Healthcare_files/figure-html/02-Data-Wrangling-32-1.png" width="672" /></p>
<div class="sourceCode" id="cb263"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb263-1"><a href="data-wrangling.html#cb263-1"></a><span class="kw">library</span>(RColorBrewer)</span>
<span id="cb263-2"><a href="data-wrangling.html#cb263-2"></a><span class="kw">library</span>(tidyverse)</span>
<span id="cb263-3"><a href="data-wrangling.html#cb263-3"></a>unique_labs =<span class="st"> </span>eve_labels <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb263-4"><a href="data-wrangling.html#cb263-4"></a><span class="st">  </span>c <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb263-5"><a href="data-wrangling.html#cb263-5"></a><span class="st">  </span>unique <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb263-6"><a href="data-wrangling.html#cb263-6"></a><span class="st">  </span>sort </span>
<span id="cb263-7"><a href="data-wrangling.html#cb263-7"></a>breaks =<span class="st"> </span>unique_labs</span>
<span id="cb263-8"><a href="data-wrangling.html#cb263-8"></a>rf &lt;-<span class="st"> </span><span class="kw">colorRampPalette</span>(<span class="kw">rev</span>(<span class="kw">brewer.pal</span>(<span class="dv">11</span>,<span class="st">&#39;Spectral&#39;</span>)))</span>
<span id="cb263-9"><a href="data-wrangling.html#cb263-9"></a>cols &lt;-<span class="st"> </span><span class="kw">rf</span>(<span class="kw">length</span>(unique_labs))</span>
<span id="cb263-10"><a href="data-wrangling.html#cb263-10"></a></span>
<span id="cb263-11"><a href="data-wrangling.html#cb263-11"></a>neurobase<span class="op">::</span><span class="kw">ortho2</span>(eve_labels, <span class="dt">col =</span> cols, <span class="dt">breaks =</span> <span class="kw">c</span>(<span class="op">-</span><span class="dv">1</span>, breaks))</span></code></pre></div>
<p><img src="Applications-of-R-in-Healthcare_files/figure-html/02-Data-Wrangling-32-1-1.png" width="672" /></p>
<p>Reference a figure by its code chunk label with the <code>fig:</code> prefix, e.g., see Figure <a href="#fig:nice-fig"><strong>??</strong></a>. Similarly, you can reference tables generated from <code>knitr::kable()</code>, e.g., see Table <a href="data-wrangling.html#tab:nice-tab">3.1</a>.</p>
<div class="sourceCode" id="cb264"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb264-1"><a href="data-wrangling.html#cb264-1"></a>knitr<span class="op">::</span><span class="kw">kable</span>(</span>
<span id="cb264-2"><a href="data-wrangling.html#cb264-2"></a>  df, <span class="dt">caption =</span> <span class="st">&#39;Table containing uncleaned data above&#39;</span>,</span>
<span id="cb264-3"><a href="data-wrangling.html#cb264-3"></a>  <span class="dt">booktabs =</span> <span class="ot">TRUE</span></span>
<span id="cb264-4"><a href="data-wrangling.html#cb264-4"></a>)</span></code></pre></div>
<table>
<caption><span id="tab:nice-tab">Table 3.1: </span>Table containing uncleaned data above</caption>
<colgroup>
<col width="15%" />
<col width="24%" />
<col width="15%" />
<col width="24%" />
<col width="8%" />
<col width="9%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">drug</th>
<th align="left">infection</th>
<th align="left">drug1</th>
<th align="left">infection1</th>
<th align="left">drug2</th>
<th align="left">infection2</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">valium 1mg</td>
<td align="left">pneumonia</td>
<td align="left">valium</td>
<td align="left">pneumonia</td>
<td align="left">valium</td>
<td align="left">pneumonia</td>
</tr>
<tr class="even">
<td align="left">verapamil sr</td>
<td align="left">aspiration pneumonia</td>
<td align="left">verapamil</td>
<td align="left">pneumonia</td>
<td align="left">verapamil</td>
<td align="left">pneumonia</td>
</tr>
<tr class="odd">
<td align="left">betaloc zoc</td>
<td align="left">tracheobronchitis</td>
<td align="left">betaloc</td>
<td align="left"></td>
<td align="left">betaloc</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">tramadol</td>
<td align="left">respiratory tract infection</td>
<td align="left">tramadol</td>
<td align="left"></td>
<td align="left">tramadol</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">valium (diazepam)</td>
<td align="left">respiratory.tract.infection</td>
<td align="left">valium (diazepam)</td>
<td align="left">respiratory.tract.infection</td>
<td align="left">valium</td>
<td align="left"></td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb265"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb265-1"><a href="data-wrangling.html#cb265-1"></a>knitr<span class="op">::</span><span class="kw">kable</span>(</span>
<span id="cb265-2"><a href="data-wrangling.html#cb265-2"></a>  Table2, <span class="dt">caption =</span> <span class="st">&#39;Table containing cleaned data above&#39;</span>,</span>
<span id="cb265-3"><a href="data-wrangling.html#cb265-3"></a>  <span class="dt">booktabs =</span> <span class="ot">TRUE</span></span>
<span id="cb265-4"><a href="data-wrangling.html#cb265-4"></a>)</span></code></pre></div>
<table>
<caption><span id="tab:nice-tab2">Table 3.2: </span>Table containing cleaned data above</caption>
<colgroup>
<col width="16%" />
<col width="10%" />
<col width="14%" />
<col width="10%" />
<col width="10%" />
<col width="14%" />
<col width="21%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Jurisdiction</th>
<th align="left">NumberCases31.03.20</th>
<th align="left">CumulativeIncidence31.03.20</th>
<th align="left">NumberCases07.04.20</th>
<th align="left">NumberDeath07.04.20</th>
<th align="left">CumulativeIncidence07.04.20</th>
<th align="left">Absolute.change.in.cumulative.incidence.</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Alabama</td>
<td align="left">999</td>
<td align="left">20.4</td>
<td align="left">2197</td>
<td align="left">39 1.</td>
<td align="left">44.9</td>
<td align="left">24.5</td>
</tr>
<tr class="even">
<td align="left">Alaska</td>
<td align="left">133</td>
<td align="left">18.0</td>
<td align="left">213</td>
<td align="left">6 2.</td>
<td align="left">28.9</td>
<td align="left">10.8</td>
</tr>
<tr class="odd">
<td align="left">Arizona</td>
<td align="left">1289</td>
<td align="left">18.0</td>
<td align="left">2575</td>
<td align="left">73 2.</td>
<td align="left">35.9</td>
<td align="left">17.9</td>
</tr>
<tr class="even">
<td align="left">Arkansas</td>
<td align="left">560</td>
<td align="left">18.6</td>
<td align="left">993</td>
<td align="left">18 1.</td>
<td align="left">32.9</td>
<td align="left">14.4</td>
</tr>
<tr class="odd">
<td align="left">California</td>
<td align="left">8131</td>
<td align="left">20.6</td>
<td align="left">15865</td>
<td align="left">374 2.</td>
<td align="left">40.1</td>
<td align="left">19.6</td>
</tr>
<tr class="even">
<td align="left">Colorado</td>
<td align="left">2966</td>
<td align="left">52.1</td>
<td align="left">5429</td>
<td align="left">179 3.</td>
<td align="left">95.3</td>
<td align="left">43.2</td>
</tr>
<tr class="odd">
<td align="left">Connecticut</td>
<td align="left">3128</td>
<td align="left">87.6</td>
<td align="left">7781</td>
<td align="left">277 3.</td>
<td align="left">217.8</td>
<td align="left">130.2</td>
</tr>
<tr class="even">
<td align="left">Delaware</td>
<td align="left">319</td>
<td align="left">33.0</td>
<td align="left">928</td>
<td align="left">16 1.</td>
<td align="left">95.9</td>
<td align="left">63.0</td>
</tr>
<tr class="odd">
<td align="left">District of Columbia</td>
<td align="left">495</td>
<td align="left">70.5</td>
<td align="left">1211</td>
<td align="left">24 2.</td>
<td align="left">172.4</td>
<td align="left">101.9</td>
</tr>
<tr class="even">
<td align="left">Florida</td>
<td align="left">6490</td>
<td align="left">30.5</td>
<td align="left">14302</td>
<td align="left">296 2.</td>
<td align="left">67.1</td>
<td align="left">36.7</td>
</tr>
<tr class="odd">
<td align="left">Georgia</td>
<td align="left">4585</td>
<td align="left">43.6</td>
<td align="left">9713</td>
<td align="left">351 3.</td>
<td align="left">92.3</td>
<td align="left">48.7</td>
</tr>
<tr class="even">
<td align="left">Hawaii</td>
<td align="left">185</td>
<td align="left">13.0</td>
<td align="left">362</td>
<td align="left">5 1.</td>
<td align="left">25.5</td>
<td align="left">12.5</td>
</tr>
<tr class="odd">
<td align="left">Idaho</td>
<td align="left">525</td>
<td align="left">29.9</td>
<td align="left">1210</td>
<td align="left">15 1.</td>
<td align="left">69.0</td>
<td align="left">39.0</td>
</tr>
<tr class="even">
<td align="left">Illinois</td>
<td align="left">5994</td>
<td align="left">47.0</td>
<td align="left">13549</td>
<td align="left">380 2.</td>
<td align="left">106.3</td>
<td align="left">59.3</td>
</tr>
<tr class="odd">
<td align="left">Indiana</td>
<td align="left">2159</td>
<td align="left">32.3</td>
<td align="left">5507</td>
<td align="left">173 3.</td>
<td align="left">82.3</td>
<td align="left">50.0</td>
</tr>
<tr class="even">
<td align="left">Iowa</td>
<td align="left">497</td>
<td align="left">15.7</td>
<td align="left">1048</td>
<td align="left">26 2.</td>
<td align="left">33.2</td>
<td align="left">17.5</td>
</tr>
<tr class="odd">
<td align="left">Kansas</td>
<td align="left">428</td>
<td align="left">14.7</td>
<td align="left">900</td>
<td align="left">27 3.</td>
<td align="left">30.9</td>
<td align="left">16.2</td>
</tr>
<tr class="even">
<td align="left">Kentucky</td>
<td align="left">591</td>
<td align="left">13.2</td>
<td align="left">1149</td>
<td align="left">65 5.</td>
<td align="left">25.7</td>
<td align="left">12.5</td>
</tr>
<tr class="odd">
<td align="left">Louisiana</td>
<td align="left">5237</td>
<td align="left">112.4</td>
<td align="left">16284</td>
<td align="left">582 3.</td>
<td align="left">349.4</td>
<td align="left">237.1</td>
</tr>
<tr class="even">
<td align="left">Maine</td>
<td align="left">303</td>
<td align="left">22.6</td>
<td align="left">519</td>
<td align="left">12 2.</td>
<td align="left">38.8</td>
<td align="left">16.1</td>
</tr>
<tr class="odd">
<td align="left">Maryland</td>
<td align="left">1660</td>
<td align="left">27.5</td>
<td align="left">5529</td>
<td align="left">124 2.</td>
<td align="left">91.5</td>
<td align="left">64.0</td>
</tr>
<tr class="even">
<td align="left">Massachusetts</td>
<td align="left">6620</td>
<td align="left">95.9</td>
<td align="left">15202</td>
<td align="left">356 2.</td>
<td align="left">220.3</td>
<td align="left">124.3</td>
</tr>
<tr class="odd">
<td align="left">Michigan</td>
<td align="left">7615</td>
<td align="left">76.2</td>
<td align="left">18970</td>
<td align="left">845 4.</td>
<td align="left">189.8</td>
<td align="left">113.6</td>
</tr>
<tr class="even">
<td align="left">Minnesota</td>
<td align="left">689</td>
<td align="left">12.3</td>
<td align="left">1154</td>
<td align="left">39 3.</td>
<td align="left">20.6</td>
<td align="left">8.3</td>
</tr>
<tr class="odd">
<td align="left">Mississippi</td>
<td align="left">1073</td>
<td align="left">35.9</td>
<td align="left">2003</td>
<td align="left">67 3.</td>
<td align="left">67.1</td>
<td align="left">31.1</td>
</tr>
<tr class="even">
<td align="left">Missouri</td>
<td align="left">1327</td>
<td align="left">21.7</td>
<td align="left">3037</td>
<td align="left">53 1.</td>
<td align="left">49.6</td>
<td align="left">27.9</td>
</tr>
<tr class="odd">
<td align="left">Montana</td>
<td align="left">203</td>
<td align="left">19.1</td>
<td align="left">332</td>
<td align="left">6 1.</td>
<td align="left">31.3</td>
<td align="left">12.1</td>
</tr>
<tr class="even">
<td align="left">Nebraska</td>
<td align="left">177</td>
<td align="left">9.2</td>
<td align="left">478</td>
<td align="left">10 2.</td>
<td align="left">24.8</td>
<td align="left">15.6</td>
</tr>
<tr class="odd">
<td align="left">Nevada</td>
<td align="left">1113</td>
<td align="left">36.7</td>
<td align="left">2087</td>
<td align="left">71 3.</td>
<td align="left">68.8</td>
<td align="left">32.1</td>
</tr>
<tr class="even">
<td align="left">New Hampshire</td>
<td align="left">367</td>
<td align="left">27.1</td>
<td align="left">747</td>
<td align="left">13 1.</td>
<td align="left">55.1</td>
<td align="left">28.0</td>
</tr>
<tr class="odd">
<td align="left">New Jersey</td>
<td align="left">18696</td>
<td align="left">209.9</td>
<td align="left">44416</td>
<td align="left">1232 2.</td>
<td align="left">498.6</td>
<td align="left">288.7</td>
</tr>
<tr class="even">
<td align="left">New Mexico</td>
<td align="left">315</td>
<td align="left">15.0</td>
<td align="left">794</td>
<td align="left">13 1.</td>
<td align="left">37.9</td>
<td align="left">22.9</td>
</tr>
<tr class="odd">
<td align="left">New York†</td>
<td align="left">32656</td>
<td align="left">293.1</td>
<td align="left">61897</td>
<td align="left">1378 2.</td>
<td align="left">555.5</td>
<td align="left">262.4</td>
</tr>
<tr class="even">
<td align="left">New York City</td>
<td align="left">41771</td>
<td align="left">497.3</td>
<td align="left">76876</td>
<td align="left">4111 5.</td>
<td align="left">915.3</td>
<td align="left">418.0</td>
</tr>
<tr class="odd">
<td align="left">North Carolina</td>
<td align="left">1584</td>
<td align="left">15.3</td>
<td align="left">3221</td>
<td align="left">46 1.</td>
<td align="left">31.0</td>
<td align="left">15.8</td>
</tr>
<tr class="even">
<td align="left">North Dakota</td>
<td align="left">126</td>
<td align="left">16.6</td>
<td align="left">237</td>
<td align="left">4 1.</td>
<td align="left">31.2</td>
<td align="left">14.6</td>
</tr>
<tr class="odd">
<td align="left">Ohio</td>
<td align="left">2199</td>
<td align="left">18.8</td>
<td align="left">4782</td>
<td align="left">167 3.</td>
<td align="left">40.9</td>
<td align="left">22.1</td>
</tr>
<tr class="even">
<td align="left">Oklahoma</td>
<td align="left">565</td>
<td align="left">14.3</td>
<td align="left">1472</td>
<td align="left">67 4.</td>
<td align="left">37.3</td>
<td align="left">23.0</td>
</tr>
<tr class="odd">
<td align="left">Oregon</td>
<td align="left">690</td>
<td align="left">16.5</td>
<td align="left">1181</td>
<td align="left">33 2.</td>
<td align="left">28.2</td>
<td align="left">11.7</td>
</tr>
<tr class="even">
<td align="left">Pennsylvania</td>
<td align="left">4843</td>
<td align="left">37.8</td>
<td align="left">14559</td>
<td align="left">240 1.</td>
<td align="left">113.7</td>
<td align="left">75.9</td>
</tr>
<tr class="odd">
<td align="left">Rhode Island</td>
<td align="left">520</td>
<td align="left">49.2</td>
<td align="left">1414</td>
<td align="left">30 2.</td>
<td align="left">133.7</td>
<td align="left">84.6</td>
</tr>
<tr class="even">
<td align="left">South Carolina</td>
<td align="left">1083</td>
<td align="left">21.3</td>
<td align="left">2417</td>
<td align="left">51 2.</td>
<td align="left">47.5</td>
<td align="left">26.2</td>
</tr>
<tr class="odd">
<td align="left">South Dakota</td>
<td align="left">108</td>
<td align="left">12.2</td>
<td align="left">320</td>
<td align="left">6 1.</td>
<td align="left">36.3</td>
<td align="left">24.0</td>
</tr>
<tr class="even">
<td align="left">Tennessee</td>
<td align="left">2239</td>
<td align="left">33.1</td>
<td align="left">4139</td>
<td align="left">72 1.</td>
<td align="left">61.1</td>
<td align="left">28.1</td>
</tr>
<tr class="odd">
<td align="left">Texas</td>
<td align="left">3266</td>
<td align="left">11.4</td>
<td align="left">8262</td>
<td align="left">154 1.</td>
<td align="left">28.8</td>
<td align="left">17.4</td>
</tr>
<tr class="even">
<td align="left">Utah</td>
<td align="left">934</td>
<td align="left">29.5</td>
<td align="left">1804</td>
<td align="left">13 0.</td>
<td align="left">57.1</td>
<td align="left">27.5</td>
</tr>
<tr class="odd">
<td align="left">Vermont</td>
<td align="left">293</td>
<td align="left">46.8</td>
<td align="left">575</td>
<td align="left">23 4.</td>
<td align="left">91.8</td>
<td align="left">45.0</td>
</tr>
<tr class="even">
<td align="left">Virginia</td>
<td align="left">1484</td>
<td align="left">17.4</td>
<td align="left">3645</td>
<td align="left">75 2.</td>
<td align="left">42.8</td>
<td align="left">25.4</td>
</tr>
<tr class="odd">
<td align="left">Washington</td>
<td align="left">4896</td>
<td align="left">65.0</td>
<td align="left">8682</td>
<td align="left">394 4.</td>
<td align="left">115.2</td>
<td align="left">50.2</td>
</tr>
<tr class="even">
<td align="left">West Virginia</td>
<td align="left">162</td>
<td align="left">9.0</td>
<td align="left">412</td>
<td align="left">4 1.</td>
<td align="left">22.8</td>
<td align="left">13.8</td>
</tr>
<tr class="odd">
<td align="left">Wisconsin</td>
<td align="left">1351</td>
<td align="left">23.2</td>
<td align="left">2578</td>
<td align="left">92 3.</td>
<td align="left">44.3</td>
<td align="left">21.1</td>
</tr>
<tr class="even">
<td align="left">Wyoming</td>
<td align="left">120</td>
<td align="left">20.8</td>
<td align="left">221</td>
<td align="left">0 —)</td>
<td align="left">38.3</td>
<td align="left">17.5</td>
</tr>
<tr class="odd">
<td align="left">American Samoa</td>
<td align="left">0</td>
<td align="left">0.0</td>
<td align="left">0</td>
<td align="left">0 —)</td>
<td align="left">0.0</td>
<td align="left">0.0</td>
</tr>
<tr class="even">
<td align="left">Federated States of Micronesia</td>
<td align="left">0</td>
<td align="left">0.0</td>
<td align="left">0</td>
<td align="left">0 —)</td>
<td align="left">0.0</td>
<td align="left">0.0</td>
</tr>
<tr class="odd">
<td align="left">Guam</td>
<td align="left">71</td>
<td align="left">42.8</td>
<td align="left">122</td>
<td align="left">4 3.</td>
<td align="left">73.6</td>
<td align="left">30.8</td>
</tr>
<tr class="even">
<td align="left">Marshall Islands</td>
<td align="left">0</td>
<td align="left">0.0</td>
<td align="left">0</td>
<td align="left">0 —)</td>
<td align="left">0.0</td>
<td align="left">0.0</td>
</tr>
<tr class="odd">
<td align="left">Northern Mariana Islands</td>
<td align="left">2</td>
<td align="left">3.5</td>
<td align="left">8</td>
<td align="left">2 25.</td>
<td align="left">14.1</td>
<td align="left">10.5</td>
</tr>
<tr class="even">
<td align="left">Palau</td>
<td align="left">0</td>
<td align="left">0.0</td>
<td align="left">0</td>
<td align="left">0 —)</td>
<td align="left">0.0</td>
<td align="left">0.0</td>
</tr>
<tr class="odd">
<td align="left">Puerto Rico</td>
<td align="left">239</td>
<td align="left">7.5</td>
<td align="left">573</td>
<td align="left">23 4.</td>
<td align="left">17.9</td>
<td align="left">10.5</td>
</tr>
<tr class="even">
<td align="left">U.S. Virgin Islands</td>
<td align="left">30</td>
<td align="left">28.0</td>
<td align="left">45</td>
<td align="left">1 2.</td>
<td align="left">42.1</td>
<td align="left">14.0</td>
</tr>
<tr class="odd">
<td align="left">U.S. Total</td>
<td align="left">186101</td>
<td align="left">56.2</td>
<td align="left">395926</td>
<td align="left">12757 3.</td>
<td align="left">119.6</td>
<td align="left">63.4</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
</div>
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-pmid31593277">
<p>Aboumatar, H., and R. A. Wise. 2019. “Notice of Retraction. Aboumatar et al. Effect of a Program Combining Transitional Care and Long-term Self-management Support on Outcomes of Hospitalized Patients With Chronic Obstructive Pulmonary Disease: A Randomized Clinical Trial. JAMA. 2018;320(22):2335-2343.” <em>JAMA</em> 322 (14): 1417–8.</p>
</div>
<div id="ref-pmid19385016">
<p>“Atlas-Based Whole Brain White Matter Analysis Using Large Deformation Diffeomorphic Metric Mapping: Application to Normal Elderly and Alzheimer’s Disease Participants.” 2009. <em>Neuroimage</em> 46 (2): 486–99. <a href="https://doi.org/10.1016/j.neuroimage.2009.01.002">https://doi.org/10.1016/j.neuroimage.2009.01.002</a>.</p>
</div>
<div id="ref-pmid11545704">
<p>Mazziotta J, Evans A, Toga A. 2001. “A Probabilistic Atlas and Reference System for the Human Brain: International Consortium for Brain Mapping (Icbm).” <em>Philos Trans R Soc Lond B Biol Sci</em> 356 (1412): 1293–1322. <a href="https://doi.org/10.1098/rstb.2001.0915">https://doi.org/10.1098/rstb.2001.0915</a>.</p>
</div>
<div id="ref-pmid11771995">
<p>Tzourio-Mazoyer N, Papathanassiou D, Landeau B. 2002. “Automated Anatomical Labeling of Activations in Spm Using a Macroscopic Anatomical Parcellation of the Mni Mri Single-Subject Brain.” <em>Neuroimage</em> 15 (1): 273–89. <a href="https://doi.org/10.1006/nimg.2001.0978">https://doi.org/10.1006/nimg.2001.0978</a>.</p>
</div>
<div id="ref-Wickham2019">
<p>Wickham, Hadley. 2019. 2nd ed. Chapman &amp; Hall/Crc the R Series. Chapman; Hall/CRC; 2 edition (May 30, 2019).</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="intro.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="statistics.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/02-Data-Wrangling.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["Applications-of-R-in-Healthcare.pdf", "Applications-of-R-in-Healthcare.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
